diff --git a/Model/Transport.php b/Model/Transport.php
index d8b12da..22d172f 100644
--- a/Model/Transport.php
+++ b/Model/Transport.php
@@ -80,7 +80,7 @@ class Transport implements TransportInterface

         $magVersion = $scopeConfig->getValue('sendinblue/mag_version_sib', 'default');

-
+        $this->transport = new \Zend\Mail\Transport\Sendmail($parameters);

         if (empty($magVersion)) {

@@ -234,7 +234,7 @@ class Transport implements TransportInterface


                     $zend_mail = Message::fromString($this->_message->getRawMessage());
-
+                    $zend_mail->setEncoding('UTF-8');
                     $subject = $zend_mail->getSubject();

                     $zend_mail->setSubject(htmlspecialchars_decode((string)$subject, ENT_QUOTES));
@@ -248,8 +248,8 @@ class Transport implements TransportInterface

                 } elseif ((version_compare($this->_magVersion, "2.2.10", "==") || version_compare($this->_magVersion, "2.2.11", "==") || version_compare($this->_magVersion, "2.3.0", "==") || version_compare($this->_magVersion, "2.3.1", "==") || version_compare($this->_magVersion, "2.3.2", "==")) && $this->_message instanceof MessageInterface) {

-                    $zend_mail = Message::fromString($this->_message->getRawMessage());
-
+                    $zend_mail = \Zend\Mail\Message::fromString($this->_message->getRawMessage());
+                    $zend_mail->setEncoding('UTF-8');
                     $subject = $zend_mail->getSubject();

                     $zend_mail->setSubject(htmlspecialchars_decode((string)$subject, ENT_QUOTES));
