diff --git a/Controller/Account/ForgotPasswordPost.php b/Controller/Account/ForgotPasswordPost.php
index 79c8d75..095afdf 100644
--- a/Controller/Account/ForgotPasswordPost.php
+++ b/Controller/Account/ForgotPasswordPost.php
@@ -93,7 +93,9 @@ class ForgotPasswordPost extends \Magento\Customer\Controller\AbstractAccount im
                 );
                 return $resultRedirect->setPath('*/*/forgotpassword');
             }
-            $this->session->destroy(['send_expire_cookie']);
+            //BEGIN PATCH
+//            $this->session->destroy(['send_expire_cookie']);
+            //END PATCH
             $this->messageManager->addSuccessMessage($this->getSuccessMessage($email));
             return $resultRedirect->setPath('*/*/');
         } else {
