Add events for custom scripts (kratki only)

diff --git a/view/frontend/templates/js.phtml b/view/frontend/templates/js.phtml
index 9301ab6..634229d 100644
--- a/view/frontend/templates/js.phtml
+++ b/view/frontend/templates/js.phtml
@@ -12,10 +12,11 @@

 $gtmScript = <<<SCRIPT
     window.{$dataLayerName} = window.{$dataLayerName} || [];
+    {$block->getDataLayerJs()}
     SCRIPT;

 if (!$block->isGdprEnabled() && $block->addJsInHead() && !$block->isAdvancedSettingsEnabled()) {
-    $dataLayerJs = $block->getDataLayerJs();
+    $dataLayerJs = '';
     $gtmScript .= <<<SCRIPT
             {$dataLayerJs}
             (function(w,d,s,l,i){
