diff --git a/view/frontend/templates/js.phtml b/view/frontend/templates/js.phtml
index 4569b4b..6f1bc48 100644
--- a/view/frontend/templates/js.phtml
+++ b/view/frontend/templates/js.phtml
@@ -20,7 +20,7 @@ $containerCode = $block->getEmbeddedCode();
     (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
             new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
         j=d.createElement(s),dl=l!='<?= /* @noEscape */ $dataLayerName ?>'?'&l='+l:'';j.async=true;j.src=
-        'https://www.googletagmanager.com/gtm.js?id='+i+dl<?= /* @noEscape */ $containerCode ? "+'{$containerCode}'" : '' ?>;f.parentNode.insertBefore(j,f);
+        'https://msr.kratki.com.com/gtm.js?id='+i+dl<?= /* @noEscape */ $containerCode ? "+'{$containerCode}'" : '' ?>;f.parentNode.insertBefore(j,f);
     })(window,document,'script','<?= /* @noEscape */ $dataLayerName ?>','<?= /* @noEscape */ $accountId ?>');
 <?php endif; ?>
 </script>
diff --git a/view/frontend/web/js/datalayer.js b/view/frontend/web/js/datalayer.js
index 24bcb24..44d43b9 100644
--- a/view/frontend/web/js/datalayer.js
+++ b/view/frontend/web/js/datalayer.js
@@ -89,7 +89,7 @@ define([
             var f = d.getElementsByTagName(s)[0],
                 j = d.createElement(s), dl = l != dataLayerName ? '&l=' + l : '';
             j.async = true;
-            j.src = '//www.googletagmanager.com/gtm.js?id=' + i + dl + containerCode;
+            j.src = '//msr.kratki.com.com/gtm.js?id=' + i + dl + containerCode;
             f.parentNode.insertBefore(j, f);
         })(window, document, 'script', dataLayerName, accountId);

diff --git a/view/frontend/templates/iframe.phtml b/view/frontend/templates/iframe.phtml
index ec06c3d..58c9860 100644
--- a/view/frontend/templates/iframe.phtml
+++ b/view/frontend/templates/iframe.phtml
@@ -9,6 +9,6 @@
 ?>

 <!-- Google Tag Manager by MagePal -->
-<noscript><iframe src="<?= '/' . '/' ?>www.googletagmanager.com/ns.html?id=<?= /* @noEscape */ $block->getEmbeddedAccountId() ?>"
+<noscript><iframe src="<?= '/' . '/' ?>msr.kratki.com/ns.html?id=<?= /* @noEscape */ $block->getEmbeddedAccountId() ?>"
 height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
 <!-- End Google Tag Manager by MagePal -->
