diff --git a/Model/Import/Product/LinkProcessor.php b/Model/Import/Product/LinkProcessor.php
index f8633a8..a147efb 100644
--- a/Model/Import/Product/LinkProcessor.php
+++ b/Model/Import/Product/LinkProcessor.php
@@ -329,10 +329,12 @@ class LinkProcessor
     ) {
         $mainTable = $resource->getMainTable();
         if (Import::BEHAVIOR_APPEND != $importEntity->getBehavior() && $productIds) {
+            //PATCH BEGIN - do not remove related products
             $importEntity->getConnection()->delete(
                 $mainTable,
-                $importEntity->getConnection()->quoteInto('product_id IN (?)', array_unique($productIds))
+                ['product_id IN (?)' => array_unique($productIds), 'link_type_id NOT IN (?)' => array_values($this->linkNameToId)]
             );
+            //PATCH END
         }
         if ($linkRows) {
             $importEntity->getConnection()->insertOnDuplicate($mainTable, $linkRows, ['link_id']);
