diff --git a/Model/Import/Product.php b/Model/Import/Product.php
index bec6517..24e96a8 100644
--- a/Model/Import/Product.php
+++ b/Model/Import/Product.php
@@ -2053,7 +2053,7 @@ class Product extends AbstractEntity
                 }
             }
             foreach ($storeIds as $storeId) {
-                if (!isset($attributes[$attrTable][$rowSku][$attrId][$storeId])) {
+                if (!isset($attributes[$attrTable][$rowSku][$attrId][$storeId]) && !($attrCode == 'url_key' && empty($attrValue))) {
                     $attributes[$attrTable][$rowSku][$attrId][$storeId] = $attrValue;
                 }
             }
