diff --git a/Model/Import/Product.php b/Model/Import/Product.php
index 24e96a8..038c48d 100644
--- a/Model/Import/Product.php
+++ b/Model/Import/Product.php
@@ -1672,6 +1672,17 @@ class Product extends AbstractEntity
                     // Product is skipped.  Go on to the next one.
                 }
             }
+            // PATCH BEGIN
+            if (!empty($attributes) && !\Creativestyle\CustomizationRiekerStaging\Model\AreaLocator::isStagingArea()) {
+                $stagingTableNameResolver = ObjectManager::getInstance()->get(\Creativestyle\CustomizationRiekerStaging\Model\Db\TableNameResolver::class);
+
+                foreach ($attributes as $tableName => $changes) {
+                    $stagingTableName = $stagingTableNameResolver->getNameWithPrefix($tableName);
+                    $attributes[$stagingTableName] = $changes;
+                }
+            }
+            // PATCH END
+
             foreach ($bunch as $rowNum => $rowData) {
                 if ($this->getErrorAggregator()->isRowInvalid($rowNum)) {
                     unset($bunch[$rowNum]);
