diff --git a/Block/Sidebar/Categories.php b/Block/Sidebar/Categories.php
index 054284c..bc048d5 100755
--- a/Block/Sidebar/Categories.php
+++ b/Block/Sidebar/Categories.php
@@ -56,7 +56,10 @@ class Categories extends Template implements IdentityInterface
                 ->addActiveFilter()
                 ->addStoreFilter($this->_storeManager->getStore()->getId())
                 ->addFieldToFilter('include_in_sidebar_tree', 1)
-                ->setOrder('position')
+                // PATCH BEGIN
+                // ->setOrder('position')
+                ->setOrder('position', \Magento\Framework\Data\Collection::SORT_ORDER_ASC)
+                // PATCH END
                 ->getTreeOrderedArray();
             foreach ($array as $key => $item) {
                 $maxDepth = $this->maxDepth();