diff --git a/Block/Widget.php b/Block/Widget.php
index 07fcf4a..607bc10 100644
--- a/Block/Widget.php
+++ b/Block/Widget.php
@@ -71,6 +71,10 @@ class Widget extends Base
 
     public function getParentProductForWidget($product)
     {
+        // PATCH START TOYS-847
+        $this->setData('product', $product);
+        return;
+        // PATCH END
         $productType = $product->getTypeId();
 
         switch ($productType) {
