Github issue: https://github.com/magento/magento2/issues/23618

diff --git a/Model/QuoteAddressValidator.php b/Model/QuoteAddressValidator.php
index f0bc12f..ebe9f12 100644
--- a/Model/QuoteAddressValidator.php
+++ b/Model/QuoteAddressValidator.php
@@ -127,6 +127,9 @@ class QuoteAddressValidator
      */
     public function validateForCart(CartInterface $cart, AddressInterface $address): void
     {
-        $this->doValidate($address, $cart->getCustomerIsGuest() ? null : $cart->getCustomer()->getId());
+        // PATCH BEGIN
+        // $this->doValidate($address, $cart->getCustomerIsGuest() ? null : $cart->getCustomer()->getId());
+        $this->doValidate($address, $cart->getCustomerId() ? $cart->getCustomer()->getId() : null);
+        // PATCH END
     }
 }
