diff --git a/src/Rewards/Helper/Balance/Spend.php b/src/Rewards/Helper/Balance/Spend.php
index 9587082..204c8a5 100644
--- a/src/Rewards/Helper/Balance/Spend.php
+++ b/src/Rewards/Helper/Balance/Spend.php
@@ -256,6 +256,14 @@ class Spend extends \Magento\Framework\App\Helper\AbstractHelper
     {
         if ($this->isRewardsIncludeTax()) {
             $subtotal = $quote->getBaseGrandTotal();
+
+            // BEGIN PATCH
+            $rules = $this->spendRulesHelper->getRules($quote);
+            foreach ($rules as $rule) {
+                $subtotal = $this->getLimitedSubtotal($quote, $rule);
+            }
+            //END PATCH
+
         } else {
             $subtotal = $quote->getBaseSubtotalWithDiscount();
             if ($this->config->getGeneralIsSpendShipping() && !$quote->isVirtual()) {
@@ -306,12 +314,17 @@ class Spend extends \Magento\Framework\App\Helper\AbstractHelper
 
             if ($rule->getActions()->validate($item)) {
                 if ($priceIncludesTax) {
-                    $itemPrice = $item->getBasePrice();
-                    if ($this->isRewardsIncludeTax()) {
-                        $itemPrice = $item->getBasePriceInclTax();
-                    }
-                    $itemPrice += (float)$item->getWeeeTaxAppliedAmountInclTax();
-                    $itemPrice = $itemPrice * $item->getQty() - $item->getBaseDiscountAmount();
+                    // BEGIN PATCH
+                    // Always use full item price, do not add discount to it
+
+                    // $itemPrice = $item->getBasePrice();
+                    // if ($this->isRewardsIncludeTax()) {
+                    //     $itemPrice = $item->getBasePriceInclTax();
+                    // }
+                    // $itemPrice += (float)$item->getWeeeTaxAppliedAmountInclTax();
+                    // $itemPrice = $itemPrice * $item->getQty() - $item->getBaseDiscountAmount();
+                    $itemPrice = $item->getRowTotalInclTax();
+                    // END PATCH
                 } else {
                     $itemPrice = $item->getBasePrice() * $item->getQty() - $item->getBaseDiscountAmount();
                 }
diff --git a/src/Rewards/Plugin/SalesRuleValidator.php b/src/Rewards/Plugin/SalesRuleValidator.php
index e6930fb..3a3eea6 100644
--- a/src/Rewards/Plugin/SalesRuleValidator.php
+++ b/src/Rewards/Plugin/SalesRuleValidator.php
@@ -60,6 +60,10 @@ class SalesRuleValidator
 
     private $taxData;
 
+    // BEGIN PATCH
+    protected $ruleCollectionFactory;
+    // END PATCH
+
     public function __construct(
         \Mirasvit\Rewards\Helper\Balance\Spend $spendHelper,
         \Mirasvit\Rewards\Helper\Purchase $rewardsPurchase,
@@ -70,6 +74,9 @@ class SalesRuleValidator
         \Magento\Tax\Model\TaxCalculation $taxCalculation,
         \Magento\Tax\Model\Config $taxConfig,
         \Magento\Tax\Helper\Data $taxData
+        // BEGIN PATCH
+        , ?\Magento\SalesRule\Model\ResourceModel\Rule\CollectionFactory $ruleCollectionFactory = null
+        // END PATCH
     ) {
         $this->spendHelper      = $spendHelper;
         $this->rewardsPurchase  = $rewardsPurchase;
@@ -80,6 +87,10 @@ class SalesRuleValidator
         $this->taxCalculation   = $taxCalculation;
         $this->taxConfig        = $taxConfig;
         $this->taxData          = $taxData;
+
+        // BEGIN PATCH
+        $this->ruleCollectionFactory = $ruleCollectionFactory ?? \Magento\Framework\App\ObjectManager::getInstance()->get(\Magento\SalesRule\Model\ResourceModel\Rule\CollectionFactory::class);
+        // END PATCH
     }
 
     /**
@@ -126,6 +137,9 @@ class SalesRuleValidator
         return $returnValue;
     }
 
+// BEGIN PATCH
+// Disabling set reward point name in discount_description in quote/order.
+// This description is confused for customer when "Spending discount calculation method = Apply to totals", because discount from reward points has separate position in totals.
     /**
      * @param Validator $validator
      * @param \callable $proceed
@@ -135,26 +149,27 @@ class SalesRuleValidator
      * @return Validator
      * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
-    public function aroundPrepareDescription(Validator $validator, $proceed, $address, $separator = ', ')
-    {
-        \Magento\Framework\Profiler::start(__CLASS__ . ':' . __METHOD__);
-        $descriptions = (array)$address->getDiscountDescriptionArray();
-
-        $quote    = $address->getQuote();
-        $purchase = $this->rewardsPurchase->getByQuote($quote);
-
-        if ($purchase && $purchase->getSpendAmount() > 0) {
-            $descriptions[] = $this->rewardsData->getPointsName();
-        }
-
-        $address->setDiscountDescriptionArray($descriptions);
-
-        $returnValue = $proceed($address, $separator);
-
-        \Magento\Framework\Profiler::stop(__CLASS__ . ':' . __METHOD__);
-
-        return $returnValue;
-    }
+//    public function aroundPrepareDescription(Validator $validator, $proceed, $address, $separator = ', ')
+//    {
+//        \Magento\Framework\Profiler::start(__CLASS__ . ':' . __METHOD__);
+//        $descriptions = (array)$address->getDiscountDescriptionArray();
+//
+//        $quote    = $address->getQuote();
+//        $purchase = $this->rewardsPurchase->getByQuote($quote);
+//
+//        if ($purchase && $purchase->getSpendAmount() > 0) {
+//            $descriptions[] = $this->rewardsData->getPointsName();
+//        }
+//
+//        $address->setDiscountDescriptionArray($descriptions);
+//
+//        $returnValue = $proceed($address, $separator);
+//
+//        \Magento\Framework\Profiler::stop(__CLASS__ . ':' . __METHOD__);
+//
+//        return $returnValue;
+//    }
+// END PATCH
 
     /**
      * @return $this
@@ -321,6 +336,12 @@ class SalesRuleValidator
         $purchase        = $this->rewardsPurchase->getByQuote($this->item->getQuote());
         $baseSpendAmount = $purchase->getSpendAmount();
 
+        // BEGIN PATCH
+        // In some cases, module returns value with applied reward point discount and in some cases without (when additional discount is added and grandTotal == 0)
+        $quote = $this->item->getQuote();
+        $baseSpendAmount = $this->adjustBaseSpendAmount($quote, $baseSpendAmount);
+        // END PATCH
+
         if (!$baseTotal) {
             $baseTotal = $total;
         }
@@ -347,6 +368,29 @@ class SalesRuleValidator
         ];
     }
 
+    // BEGIN PATCH
+    // Method to adjust reward points variable - baseSpendAmount. This value contain grand total with all discounts without reward point discount
+    protected function adjustBaseSpendAmount($quote, $baseSpendAmount)
+    {
+        $cartFixedRules = $quote->getCartFixedRules();
+
+        if (empty($cartFixedRules)) {
+            return $baseSpendAmount;
+        }
+
+        $grandTotal = $quote->getGrandTotal();
+        $subtotalWithoutDiscount = $grandTotal + ($quote->getSubtotal() - $quote->getSubtotalWithDiscount());
+
+        $ruleCollection = $this->ruleCollectionFactory->create();
+        $ruleCollection->addFieldToFilter('rule_id', ['in' => array_keys($cartFixedRules)]);
+
+        foreach ($ruleCollection as $rule) {
+            $subtotalWithoutDiscount -= $rule->getDiscountAmount();
+        }
+
+        return min($baseSpendAmount, $subtotalWithoutDiscount);
+    }
+    // END PATCH
 
     /**
      * We need this because Faonni_Price changes total without basetotal
