diff --git a/src/Rewards/Helper/Balance/Spend.php b/src/Rewards/Helper/Balance/Spend.php
index 262edda..ac7e8e1 100644
--- a/src/Rewards/Helper/Balance/Spend.php
+++ b/src/Rewards/Helper/Balance/Spend.php
@@ -256,6 +256,13 @@ class Spend extends \Magento\Framework\App\Helper\AbstractHelper
     {
         if ($this->isRewardsIncludeTax()) {
             $subtotal = $quote->getBaseGrandTotal();
+
+            // BEGIN PATCH
+            $rules = $this->spendRulesHelper->getRules($quote);
+            foreach ($rules as $rule) {
+                $subtotal = $this->getLimitedSubtotal($quote, $rule);
+            }
+            //END PATCH
         } else {
             $subtotal = $quote->getBaseSubtotalWithDiscount();
             if ($this->config->getGeneralIsSpendShipping() && !$quote->isVirtual()) {
@@ -306,12 +313,17 @@ class Spend extends \Magento\Framework\App\Helper\AbstractHelper
 
             if ($rule->getActions()->validate($item)) {
                 if ($priceIncludesTax) {
-                    $itemPrice = $item->getBasePrice();
-                    if ($this->isRewardsIncludeTax()) {
-                        $itemPrice = $item->getBasePriceInclTax();
-                    }
-                    $itemPrice += (float)$item->getWeeeTaxAppliedAmountInclTax();
-                    $itemPrice = $itemPrice * $item->getQty() - $item->getBaseDiscountAmount();
+                    // BEGIN PATCH
+                    // Always use full item price, do not add discount to it
+
+                    // $itemPrice = $item->getBasePrice();
+                    // if ($this->isRewardsIncludeTax()) {
+                    //     $itemPrice = $item->getBasePriceInclTax();
+                    // }
+                    // $itemPrice += (float)$item->getWeeeTaxAppliedAmountInclTax();
+                    // $itemPrice = $itemPrice * $item->getQty() - $item->getBaseDiscountAmount();
+                    $itemPrice = $item->getRowTotalInclTax();
+                    // END PATCH
                 } else {
                     $itemPrice = $item->getBasePrice() * $item->getQty() - $item->getBaseDiscountAmount();
                 }
diff --git a/src/Rewards/Plugin/SalesRuleValidator.php b/src/Rewards/Plugin/SalesRuleValidator.php
index 52d59bf..58e30b9 100644
--- a/src/Rewards/Plugin/SalesRuleValidator.php
+++ b/src/Rewards/Plugin/SalesRuleValidator.php
@@ -30,10 +30,21 @@ class SalesRuleValidator
 {
     private $itemDiscountService;
 
+    // BEGIN PATCH
+    protected $ruleCollectionFactory;
+    // END PATCH
+
     public function __construct(
         ItemDiscountService $itemDiscountService
+        // BEGIN PATCH
+        , ?\Magento\SalesRule\Model\ResourceModel\Rule\CollectionFactory $ruleCollectionFactory = null
+        // END PATCH
     ) {
         $this->itemDiscountService = $itemDiscountService;
+
+        // BEGIN PATCH
+        $this->ruleCollectionFactory = $ruleCollectionFactory ?? \Magento\Framework\App\ObjectManager::getInstance()->get(\Magento\SalesRule\Model\ResourceModel\Rule\CollectionFactory::class);
+        // END PATCH
     }
 
     /**
@@ -58,10 +69,14 @@ class SalesRuleValidator
      * @return array
      * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
-    public function beforePrepareDescription(Validator $validator, $address, $separator = ', ')
-    {
-        $this->itemDiscountService->applyDiscountDescription($address);
-
-        return [$address, $separator];
-    }
+// BEGIN PATCH
+// Disabling set reward point name in discount_description in quote/order.
+// This description is confused for customer when "Spending discount calculation method = Apply to totals", because discount from reward points has separate position in totals.
+//    public function beforePrepareDescription(Validator $validator, $address, $separator = ', ')
+//    {
+//        $this->itemDiscountService->applyDiscountDescription($address);
+//
+//        return [$address, $separator];
+//    }
+// END PATCH
 }
diff --git a/src/Rewards/Service/Discount/ItemDiscountService.php b/src/Rewards/Service/Discount/ItemDiscountService.php
index 8168184..126d030 100644
--- a/src/Rewards/Service/Discount/ItemDiscountService.php
+++ b/src/Rewards/Service/Discount/ItemDiscountService.php
@@ -270,6 +270,12 @@ class ItemDiscountService
         $purchase        = $this->rewardsPurchase->getByQuote($this->item->getQuote());
         $baseSpendAmount = $purchase->getSpendAmount();
 
+        // BEGIN PATCH
+        // In some cases, module returns value with applied reward point discount and in some cases without (when additional discount is added and grandTotal == 0)
+        $quote = $this->item->getQuote();
+        $baseSpendAmount = $this->adjustBaseSpendAmount($quote, $baseSpendAmount);
+        // END PATCH
+
         if (!$baseTotal) {
             $baseTotal = $total;
         }
@@ -296,6 +302,30 @@ class ItemDiscountService
         ];
     }
 
+    // BEGIN PATCH
+    // Method to adjust reward points variable - baseSpendAmount. This value contain grand total with all discounts without reward point discount
+    protected function adjustBaseSpendAmount($quote, $baseSpendAmount)
+    {
+        $cartFixedRules = $quote->getCartFixedRules();
+
+        if (empty($cartFixedRules)) {
+            return $baseSpendAmount;
+        }
+
+        $grandTotal = $quote->getGrandTotal();
+        $subtotalWithoutDiscount = $grandTotal + ($quote->getSubtotal() - $quote->getSubtotalWithDiscount());
+
+        $ruleCollection = $this->ruleCollectionFactory->create();
+        $ruleCollection->addFieldToFilter('rule_id', ['in' => array_keys($cartFixedRules)]);
+
+        foreach ($ruleCollection as $rule) {
+            $subtotalWithoutDiscount -= $rule->getDiscountAmount();
+        }
+
+        return min($baseSpendAmount, $subtotalWithoutDiscount);
+    }
+    // END PATCH
+
 
     /**
      * We need this because Faonni_Price changes total without basetotal
