diff --git a/view/frontend/web/js/form/element/suggestions-mixin.js b/view/frontend/web/js/form/element/suggestions-mixin.js
index 80422a7..51bb6a1 100644
--- a/view/frontend/web/js/form/element/suggestions-mixin.js
+++ b/view/frontend/web/js/form/element/suggestions-mixin.js
@@ -47,7 +47,7 @@ define([
 
             return this;
         },
-        
+
         initObservable: function () {
             this._super();
 
@@ -58,6 +58,11 @@ define([
 
         focusSuggestionsInput: function() {
             this.showSuggestionsDropdown(true);
+
+            if (this.value() === undefined) {
+                return;
+            }
+
             if (this.selectedCountry() === this.suggestionsCountry && this.value().length >= this.minValueLength && this.suggestions().length == 0) { // load suggestions on focus, when empty
                 clearTimeout(this.suggestionsTimeout);
                 this.suggestionsTimeout = setTimeout(this.getSuggestions.bind(this, this.value()), this.suggestionsDelay);
@@ -121,7 +126,7 @@ define([
                     zipCity: val
                 }),
                 self = this;
-                
+
             storage.post(
                 serviceUrl,
                 payload,
