diff --git a/Model/ResourceModel/Rating.php b/Model/ResourceModel/Rating.php
index 81f732f..836753a 100644
--- a/Model/ResourceModel/Rating.php
+++ b/Model/ResourceModel/Rating.php
@@ -460,7 +460,10 @@ class Rating extends \Magento\Framework\Model\ResourceModel\Db\AbstractDb
 
         if ($onlyForCurrentStore) {
             foreach ($data as $row) {
-                if ($row['store_id'] !== $currentStore) {
+                // BEGIN PATCH
+                //if ($row['store_id'] !== $currentStore) {
+                if ($row['store_id'] == $currentStore) {
+                // END PATCH
                     $object->addData($row);
                 }
             }
