diff --git a/Helper/Data.php b/Helper/Data.php
index cd105eb..a82d765 100644
--- a/Helper/Data.php
+++ b/Helper/Data.php
@@ -75,7 +75,11 @@ class Data extends \Magento\Framework\App\Helper\AbstractHelper
         }
         while ($dateStart->diff($dateEnd)->invert == 0) {
             $intervals[] = $dateStart->format($dateFormat);
-            $dateStart->add($dateInterval);
+            if (($period === self::REPORT_PERIOD_TYPE_MONTH) && ((int) $dateStart->format('j') === (int) $dateStart->format('t'))) {
+                $dateStart->modify('last day of next month');
+            } else {
+                $dateStart->add($dateInterval);
+            }
         }
 
         if (!in_array($dateEnd->format($dateFormat), $intervals)) {
