diff --git a/Model/ResourceModel/Post/Fulltext/Collection.php b/Model/ResourceModel/Post/Fulltext/Collection.php
index 4c0bf5a..34cf3d5 100755
--- a/Model/ResourceModel/Post/Fulltext/Collection.php
+++ b/Model/ResourceModel/Post/Fulltext/Collection.php
@@ -101,7 +101,7 @@ class Collection extends \Magefan\Blog\Model\ResourceModel\Post\Collection
     /**
      * {@inheritDoc}
      */
-    public function setOrder($attribute, $dir = Select::SQL_DESC)
+    public function setOrder($field, $direction = self::SORT_ORDER_DESC)
     {
         throw new \LogicException("Sorting on multiple stores is not allowed in search engine collections.");
     }
