diff --git a/Plugin/InventoryIndexer/Indexer/SourceItem/Strategy/Sync/FulltextIndexUpdater.php b/Plugin/InventoryIndexer/Indexer/SourceItem/Strategy/Sync/FulltextIndexUpdater.php
index 4ae3da0..1337bd6 100644
--- a/Plugin/InventoryIndexer/Indexer/SourceItem/Strategy/Sync/FulltextIndexUpdater.php
+++ b/Plugin/InventoryIndexer/Indexer/SourceItem/Strategy/Sync/FulltextIndexUpdater.php
@@ -74,7 +74,10 @@ class FulltextIndexUpdater
         $afterSalableList = $this->getSalableStatuses->execute($sourceItemIds);
         $productsIdsToProcess = $this->getProductsIdsToProcess->execute($beforeSalableList, $afterSalableList);
         if (!empty($productsIdsToProcess)) {
-            $this->fulltextIndexProcessor->reindexList($productsIdsToProcess, true);
+            // $this->fulltextIndexProcessor->reindexList($productsIdsToProcess, true);
+            // BEGIN PATCH
+            $this->fulltextIndexProcessor->reindexList($productsIdsToProcess, false);
+            // END PATCH
         }
     }
 
@@ -96,7 +99,10 @@ class FulltextIndexUpdater
         $afterSalableList = $this->getSalableStatuses->execute([$sourceItemId]);
         $productsIdsToProcess = $this->getProductsIdsToProcess->execute($beforeSalableList, $afterSalableList);
         if (!empty($productsIdsToProcess)) {
-            $this->fulltextIndexProcessor->reindexList($productsIdsToProcess, true);
+            // $this->fulltextIndexProcessor->reindexList($productsIdsToProcess, true);
+            // BEGIN PATCH
+            $this->fulltextIndexProcessor->reindexList($productsIdsToProcess, false);
+            // END PATCH
         }
     }
 }
