diff --git a/Model/Export/Review.php b/Model/Export/Review.php
index f89d222..22a380a 100644
--- a/Model/Export/Review.php
+++ b/Model/Export/Review.php
@@ -101,7 +101,10 @@ class Review extends AbstractEntity implements EntityInterface
         'detail',
         'status',
         'created_at',
-        'product_name'
+        'product_name',
+        // BEGIN PATCH
+        'store_id'
+        // END PATCH
     ];
 
     /**
@@ -349,6 +352,13 @@ class Review extends AbstractEntity implements EntityInterface
                 if ($key == 'product_name') {
                     $key = 'cpev.value';
                 }
+
+                // BEGIN PATCH
+                if ($key == 'store_id') {
+                    $key = 'detail.store_id';
+                }
+                // END PATCH
+
                 if (in_array($key, $this->ratingFields)) {
                     $ratingId = array_search($key, $this->ratingFields);
                     $reviewIds = $this->getReviewIds($ratingId, $value);
diff --git a/Model/ResourceModel/Review/Collection.php b/Model/ResourceModel/Review/Collection.php
index 309f1b2..a728479 100644
--- a/Model/ResourceModel/Review/Collection.php
+++ b/Model/ResourceModel/Review/Collection.php
@@ -27,7 +27,16 @@ class Collection extends AbstractCollection
      */
     protected function _initSelect()
     {
-        parent::_initSelect();
+        //parent::_initSelect();
+        // BEGIN PATCH
+        \Magento\Framework\Model\ResourceModel\Db\Collection\AbstractCollection::_initSelect();
+
+        $this->getSelect()->join(
+            ['detail' => $this->getReviewDetailTable()],
+            'main_table.review_id = detail.review_id',
+            ['detail_id', 'title', 'detail', 'nickname', 'customer_id', 'store_id']
+        );
+        // END PATCH
 
         $this->getSelect()->join(
             ['p' => $this->getProductTable()],
