diff --git a/Plugin/InventoryIndexer/Indexer/SourceItem/Strategy/Sync/CacheFlush.php b/Plugin/InventoryIndexer/Indexer/SourceItem/Strategy/Sync/CacheFlush.php
index 59d420b..2114f7c 100644
--- a/Plugin/InventoryIndexer/Indexer/SourceItem/Strategy/Sync/CacheFlush.php
+++ b/Plugin/InventoryIndexer/Indexer/SourceItem/Strategy/Sync/CacheFlush.php
@@ -7,11 +7,14 @@ declare(strict_types=1);

 namespace Magento\InventoryCache\Plugin\InventoryIndexer\Indexer\SourceItem\Strategy\Sync;

+use Magento\Framework\Indexer\IndexerRegistry;
 use Magento\InventoryCache\Model\FlushCacheByCategoryIds;
 use Magento\InventoryCache\Model\FlushCacheByProductIds;
+use Magento\InventoryIndexer\Model\GetProductsIdsToProcess;
 use Magento\InventoryIndexer\Indexer\SourceItem\Strategy\Sync;
+use Magento\InventoryIndexer\Indexer\SourceItem\GetSalableStatuses;
 use Magento\InventoryIndexer\Model\ResourceModel\GetCategoryIdsByProductIds;
-use Magento\InventoryIndexer\Model\ResourceModel\GetProductIdsBySourceItemIds;
+use Magento\InventoryIndexer\Indexer\InventoryIndexer;

 /**
  * Clean cache for corresponding products after source item reindex.
@@ -24,52 +27,79 @@ class CacheFlush
     private $flushCacheByIds;

     /**
-     * @var GetProductIdsBySourceItemIds
+     * @var GetCategoryIdsByProductIds
      */
-    private $getProductIdsBySourceItemIds;
+    private $getCategoryIdsByProductIds;

     /**
-     * @var GetCategoryIdsByProductIds
+     * @var GetSalableStatuses
      */
-    private $getCategoryIdsByProductIds;
+    private $getSalableStatuses;

     /**
      * @var FlushCacheByCategoryIds
      */
     private $flushCategoryByCategoryIds;

+    /**
+     * @var GetProductsIdsToProcess
+     */
+    private $getProductsIdsToProcess;
+
+    /**
+     * @var IndexerRegistry
+     */
+    private $indexerRegistry;
+
     /**
      * @param FlushCacheByProductIds $flushCacheByIds
-     * @param GetProductIdsBySourceItemIds $getProductIdsBySourceItemIds
      * @param GetCategoryIdsByProductIds $getCategoryIdsByProductIds
      * @param FlushCacheByCategoryIds $flushCategoryByCategoryIds
+     * @param GetSalableStatuses $getSalableStatuses
+     * @param GetProductsIdsToProcess $getProductsIdsToProcess
+     * @param IndexerRegistry $indexerRegistry
      */
     public function __construct(
         FlushCacheByProductIds $flushCacheByIds,
-        GetProductIdsBySourceItemIds $getProductIdsBySourceItemIds,
         GetCategoryIdsByProductIds $getCategoryIdsByProductIds,
-        FlushCacheByCategoryIds $flushCategoryByCategoryIds
+        FlushCacheByCategoryIds $flushCategoryByCategoryIds,
+        GetSalableStatuses $getSalableStatuses,
+        GetProductsIdsToProcess $getProductsIdsToProcess,
+        IndexerRegistry $indexerRegistry
     ) {
         $this->flushCacheByIds = $flushCacheByIds;
-        $this->getProductIdsBySourceItemIds = $getProductIdsBySourceItemIds;
         $this->getCategoryIdsByProductIds = $getCategoryIdsByProductIds;
         $this->flushCategoryByCategoryIds = $flushCategoryByCategoryIds;
+        $this->getSalableStatuses = $getSalableStatuses;
+        $this->getProductsIdsToProcess = $getProductsIdsToProcess;
+        $this->indexerRegistry = $indexerRegistry;
     }

     /**
      * Clean cache for specific products after source items reindex.
      *
      * @param Sync $subject
-     * @param void $result
+     * @param callable $proceed
      * @param array $sourceItemIds
-     * @throws \Exception in case catalog product entity type hasn't been initialize.
+     * @return void
+     * @throws \Exception in case catalog product entity type hasn't been initialized.
      * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
-    public function afterExecuteList(Sync $subject, $result, array $sourceItemIds)
+    public function aroundExecuteList(Sync $subject, callable $proceed, array $sourceItemIds) : void
     {
-        $productIds = $this->getProductIdsBySourceItemIds->execute($sourceItemIds);
-        $categoryIds = $this->getCategoryIdsByProductIds->execute($productIds);
-        $this->flushCategoryByCategoryIds->execute($categoryIds);
-        $this->flushCacheByIds->execute($productIds);
+        $beforeSalableList = $this->getSalableStatuses->execute($sourceItemIds);
+        $proceed($sourceItemIds);
+        $afterSalableList = $this->getSalableStatuses->execute($sourceItemIds);
+        $forceDefaultProcessing = !$this->indexerRegistry->get(InventoryIndexer::INDEXER_ID)->isScheduled();
+        $productsIdsToFlush = $this->getProductsIdsToProcess->execute(
+            $beforeSalableList,
+            $afterSalableList,
+            $forceDefaultProcessing
+        );
+        if (!empty($productsIdsToFlush)) {
+            $categoryIds = $this->getCategoryIdsByProductIds->execute($productsIdsToFlush);
+            $this->flushCacheByIds->execute($productsIdsToFlush);
+            $this->flushCategoryByCategoryIds->execute($categoryIds);
+        }
     }
 }
--
2.30.1 (Apple Git-130)

