diff --git a/Plugin/Block/Stockqty/AbstractStockqtyPlugin.php b/Plugin/Block/Stockqty/AbstractStockqtyPlugin.php
index 561b9b1..33b634e 100644
--- a/Plugin/Block/Stockqty/AbstractStockqtyPlugin.php
+++ b/Plugin/Block/Stockqty/AbstractStockqtyPlugin.php
@@ -73,11 +73,13 @@ class AbstractStockqtyPlugin
         $websiteId = (int)$subject->getProduct()->getStore()->getWebsiteId();
         $stockId = (int)$this->stockByWebsiteId->execute($websiteId)->getStockId();
         $stockItemConfig = $this->getStockItemConfiguration->execute($sku, $stockId);
+        $productSalableQty = $this->getProductSalableQty->execute($sku, $stockId);
 
         return ($stockItemConfig->getBackorders() === StockItemConfigurationInterface::BACKORDERS_NO
             || $stockItemConfig->getBackorders() !== StockItemConfigurationInterface::BACKORDERS_NO
             && $stockItemConfig->getMinQty() < 0)
-            && $this->getProductSalableQty->execute($sku, $stockId) <= $stockItemConfig->getStockThresholdQty();
+            && $productSalableQty <= $stockItemConfig->getStockThresholdQty()
+            && $productSalableQty > 0;
     }
 
     /**
