diff --git a/Model/Webhook/Listener/Refund/SuccessfulCommand.php b/Model/Webhook/Listener/Refund/SuccessfulCommand.php
index 0cd0e36..d1b2be1 100644
--- a/Model/Webhook/Listener/Refund/SuccessfulCommand.php
+++ b/Model/Webhook/Listener/Refund/SuccessfulCommand.php
@@ -240,9 +240,22 @@ class SuccessfulCommand extends AbstractCommand
                     }
                     break;
                 case LineItemType::FEE:
+                    // BEGIN PATCH
+
+                    // Add payment fee to $creditmemoAmount. Otherwise, it will be added as adjustment_positive to refund and it will break creation of creditmemo
+                    if ($reduction->getQuantityReduction() > 0 && $order->getPaymentFee()) {
+                        $creditmemoAmount += ($order->getPaymentFee() + $order->getPaymentFeeTax());
+                        $creditmemoAmount += ($order->getPaymentFeeTax() + $order->getPaymentFeeTaxDiscount());
+                    }
+                    break;
+                // END PATCH
                 case LineItemType::DISCOUNT:
                     break;
                 case LineItemType::SHIPPING:
+                    // BEGIN PATCH
+
+                    // Use shipping price without discount
+                    /*
                     if ($reduction->getQuantityReduction() > 0) {
                         $shippingAmount = $baseLineItems[$reduction->getLineItemUniqueId()]->getAmountIncludingTax();
                     } elseif ($reduction->getUnitPriceReduction() > 0) {
@@ -263,7 +276,11 @@ class SuccessfulCommand extends AbstractCommand
                         $shippingAmount += ($shippingAmount / $order->getShippingAmount()) *
                             $order->getShippingDiscountAmount();
                     }
+                    */
+                    $shippingAmount = $order->getShippingInclTax();
                     break;
+
+                    //END PATCH
             }
         }
 
