diff --git a/Model/Service/AbstractLineItemService.php b/Model/Service/AbstractLineItemService.php
index 5510975..d00f535 100644
--- a/Model/Service/AbstractLineItemService.php
+++ b/Model/Service/AbstractLineItemService.php
@@ -86,6 +86,13 @@ abstract class AbstractLineItemService
      */
     private $productRepository;
 
+    // BEGIN PATCH
+    /**
+     * @var \Magento\Tax\Model\Config
+     */
+    protected $taxConfig;
+    // END PATCH
+
     /**
      *
      * @param Helper $helper
@@ -97,12 +104,18 @@ abstract class AbstractLineItemService
      * @param CustomerGroupRegistry $groupRegistry
      * @param EventManagerInterface $eventManager
      * @param ProductRepositoryInterface $productRepository
+     * BEGIN PATCH
+     * @param ?\Magento\Tax\Model\Config $taxConfig
+     * END PATCH
      */
     public function __construct(Helper $helper, LineItemHelper $lineItemHelper, ScopeConfigInterface $scopeConfig,
         TaxClassRepositoryInterface $taxClassRepository, TaxHelper $taxHelper, TaxCalculation $taxCalculation,
         CustomerGroupRegistry $groupRegistry, EventManagerInterface $eventManager,
-        ProductRepositoryInterface $productRepository)
-    {
+        ProductRepositoryInterface $productRepository,
+        // BEGIN PATCH
+        ?\Magento\Tax\Model\Config $taxConfig = null
+        // END PATCH
+    ) {
         $this->helper = $helper;
         $this->lineItemHelper = $lineItemHelper;
         $this->scopeConfig = $scopeConfig;
@@ -112,6 +125,10 @@ abstract class AbstractLineItemService
         $this->groupRegistry = $groupRegistry;
         $this->eventManager = $eventManager;
         $this->productRepository = $productRepository;
+
+        // BEGIN PATCH
+        $this->taxConfig = $taxConfig ?? \Magento\Framework\App\ObjectManager::getInstance()->get(\Magento\Tax\Model\Config::class);
+        // END PATCH
     }
 
     /**
@@ -393,9 +410,16 @@ abstract class AbstractLineItemService
         $customerGroup = $this->groupRegistry->retrieve($entity->getCustomerGroupId());
         $taxRateRequest = $this->taxCalculation->getRateRequest($entity->getShippingAddress(),
             $entity->getBillingAddress(), $customerGroup->getTaxClassId(), $entity->getStore());
-        $shippingTaxClassId = $this->scopeConfig->getValue(
-            \Magento\Tax\Model\Config::CONFIG_XML_PATH_SHIPPING_TAX_CLASS, ScopeInterface::SCOPE_STORE,
-            $entity->getStoreId());
+
+        // BEGIN PATCH
+        // use the highest product tax class id as shipping tax class id
+
+        // $shippingTaxClassId = $this->scopeConfig->getValue(
+        //    \Magento\Tax\Model\Config::CONFIG_XML_PATH_SHIPPING_TAX_CLASS, ScopeInterface::SCOPE_STORE,
+        //    $entity->getStoreId());
+        $shippingTaxClassId = $this->taxConfig->getShippingTaxClass($entity->getStoreId());
+        // END PATCH
+
         if ($shippingTaxClassId > 0) {
             $shippingTaxClass = $this->taxClassRepository->get($shippingTaxClassId);
             $taxRateRequest->setProductClassId($shippingTaxClassId);
