diff --git a/Model/ProductPermission/Resolver/CurrentCategory.php b/Model/ProductPermission/Resolver/CurrentCategory.php
index 66e3a55..abfb6ad 100644
--- a/Model/ProductPermission/Resolver/CurrentCategory.php
+++ b/Model/ProductPermission/Resolver/CurrentCategory.php
@@ -96,7 +96,12 @@ class CurrentCategory
         $currentCategoryId = $product->getCategoryId();
 
         try {
-            $currentCategory = $this->categoryRepository->get($currentCategoryId);
+            // BEGIN PATCH
+            // $currentCategory = $this->categoryRepository->get($currentCategoryId);
+            if ($currentCategoryId) {
+                $currentCategory = $this->categoryRepository->get($currentCategoryId);
+            }
+            // PATCH END
         } catch (NoSuchEntityException $e) {
             $currentCategory = null;
         }
