diff --git a/Model/Rate/Provider.php b/Model/Rate/Provider.php
index c81ffba..8bd4b9c 100644
--- a/Model/Rate/Provider.php
+++ b/Model/Rate/Provider.php
@@ -293,15 +293,17 @@ class Provider
     private function getSourceCodes(array $quoteItems)
     {
         $sourceCodes = [];
-        if ($this->msiModuleStatusInspector->isEnabled()) {
-            $getSourceSelectionResultFromQuote = $this->objectManager->create(
-                GetSourceSelectionResultFromQuote::class
-            );
-
-            /** @var Item|QuoteAddressItem $firstItem */
-            $firstItem = array_shift($quoteItems);
-            $sourceCodes = $getSourceSelectionResultFromQuote->execute($firstItem->getQuote());
-        }
+        // BEGIN PATCH - infinite loop in checkout triggered by reward points
+        // if ($this->msiModuleStatusInspector->isEnabled()) {
+        //    $getSourceSelectionResultFromQuote = $this->objectManager->create(
+        //        GetSourceSelectionResultFromQuote::class
+        //    );
+        //
+        //    /** @var Item|QuoteAddressItem $firstItem */
+        //    $firstItem = array_shift($quoteItems);
+        //    $sourceCodes = $getSourceSelectionResultFromQuote->execute($firstItem->getQuote());
+        // }
+        // PATCH END
 
         return $sourceCodes;
     }
