diff --git a/Helper/Data.php b/Helper/Data.php
index 0e95046..6825e9f 100644
--- a/Helper/Data.php
+++ b/Helper/Data.php
@@ -774,6 +774,29 @@ class Data extends \Magento\Framework\App\Helper\AbstractHelper
             }
         }
 
+        // BEGIN PATCH
+        // Add payment fee tax item to aggregatedTaxes
+        $paymentFeeTaxAmount = $salesItem->getPaymentFeeTax() - $salesItem->getPaymentFeeTaxDiscount();
+        $originalPaymentFeeTaxAmount = $order->getPaymentFeeTax() - $order->getPaymentFeeTaxDiscount();
+
+        if ($salesItem instanceof \Magento\Sales\Model\Order\Creditmemo) {
+            $paymentFeeTaxAmount = $originalPaymentFeeTaxAmount;
+        }
+
+        if ($originalPaymentFeeTaxAmount <= 0 || $paymentFeeTaxAmount <= 0) {
+            return $taxClassAmount;
+        }
+
+        $paymentFeeRatio = $paymentFeeTaxAmount / $originalPaymentFeeTaxAmount;
+        foreach ($orderTaxDetails->getItems() as $itemTaxDetail) {
+            if ($itemTaxDetail->getType() != 'payment_fee') {
+                continue;
+            }
+
+            $taxClassAmount = $this->_aggregateTaxes($taxClassAmount, $itemTaxDetail, $paymentFeeRatio);
+        }
+        // END PATCH
+
         return $taxClassAmount;
     }
 
