diff --git a/Controller/Adminhtml/Index/Rollback.php b/Controller/Adminhtml/Index/Rollback.php
index 25da907e4ca32..41a4e5755b413 100644
--- a/Controller/Adminhtml/Index/Rollback.php
+++ b/Controller/Adminhtml/Index/Rollback.php
@@ -16,6 +16,11 @@
  */
 class Rollback extends \Magento\Backup\Controller\Adminhtml\Index implements HttpPostActionInterface
 {
+    /**
+     * @see _isAllowed()
+     */
+    public const ADMIN_RESOURCE = 'Magento_Backup::rollback';
+
     /**
      * Rollback Action
      *
@@ -26,10 +31,6 @@ class Rollback extends \Magento\Backup\Controller\Adminhtml\Index implements Htt
      */
     public function execute()
     {
-        if (!$this->_objectManager->get(\Magento\Backup\Helper\Data::class)->isRollbackAllowed()) {
-            $this->_forward('denied');
-        }
-
         if (!$this->getRequest()->isAjax()) {
             return $this->_redirect('*/*/index');
         }
