diff --git a/Api/SortOrder.php b/Api/SortOrder.php
index b2da7d1..11a7635 100644
--- a/Api/SortOrder.php
+++ b/Api/SortOrder.php
@@ -150,7 +150,7 @@ class SortOrder extends AbstractSimpleObject
      */
     private function validateField(string $field): void
     {
-        if (preg_match('/[^a-z0-9\_]/i', $field)) {
+        if (preg_match('/[^a-z0-9\.\_]/i', $field)) {
             throw new InputException(
                 new Phrase(
                     'Sort order field %1 contains restricted symbols',
-- 