diff --git a/Model/ResourceModel/Indexer/Price.php b/Model/ResourceModel/Indexer/Price.php
index 0a098b1..a8288a7 100644
--- a/Model/ResourceModel/Indexer/Price.php
+++ b/Model/ResourceModel/Indexer/Price.php
@@ -709,12 +709,15 @@ class Price implements DimensionalIndexerInterface
                 'tier_price' => $tierExpr,
             ]
         );
-        $select->join(
-            ['si' => $this->getTable('cataloginventory_stock_status')],
-            'si.product_id = bs.product_id',
-            []
-        );
-        $select->where('si.stock_status = ?', Stock::STOCK_IN_STOCK);
+        // BEGIN PATCH
+        // Disable checking stock for bundle product during reindex.
+        // $select->join(
+        //     ['si' => $this->getTable('cataloginventory_stock_status')],
+        //     'si.product_id = bs.product_id',
+        //     []
+        // );
+        // $select->where('si.stock_status = ?', Stock::STOCK_IN_STOCK);
+        // END PATCH
         $query = str_replace('AS `idx`', 'AS `idx` USE INDEX (PRIMARY)', (string) $select);
         $insertColumns = [
             'entity_id',
