diff --git a/Helper/Data.php b/Helper/Data.php
index 28b192c..bc18a0a 100755
--- a/Helper/Data.php
+++ b/Helper/Data.php
@@ -47,6 +47,10 @@ class Data extends AbstractHelper
      */
     private $_priceHelper;

+    // BEGIN PATCH
+    protected \Magento\Customer\Model\Session $customerSession;
+    // END PATCH
+
     /**
      * Data constructor.
      * @param \Magento\Framework\App\Helper\Context $context
