diff --git a/Plugin/Cms/Block/Block/InjectLazyLoadingIntoImages.php b/Plugin/Cms/Block/Block/InjectLazyLoadingIntoImages.php
index c7b6a4a..c396d62 100644
--- a/Plugin/Cms/Block/Block/InjectLazyLoadingIntoImages.php
+++ b/Plugin/Cms/Block/Block/InjectLazyLoadingIntoImages.php
@@ -34,7 +34,18 @@ class InjectLazyLoadingIntoImages

         try {
             $dom = new \DomDocument();
-            $dom->loadHTML(mb_encode_numericentity('<html>' . $result . '</html>', [0x80, 0x10FFFF, 0, ~0], 'UTF-8'), LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD  | LIBXML_NOERROR);
+            // PATCH BEGIN
+            //$dom->loadHTML(mb_encode_numericentity('<html>' . $result . '</html>', [0x80, 0x10FFFF, 0, ~0], 'UTF-8'), LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD  | LIBXML_NOERROR);
+            $html = str_replace('&', '＆', html_entity_decode($result)); // unicode ampersand
+            $dom->loadHTML(
+                mb_encode_numericentity('<html>' . $html . '</html>', [0x80, 0x10FFFF, 0, ~0], 'UTF-8'),
+                LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD | LIBXML_NOERROR
+            );
+            // PATCH END
         } catch (\Exception $e) {

             if ($this->configuration->isLoggerEnabled()) {
