diff --git a/src/Magento/FunctionalTestingFramework/Config/FileResolver/Primary.php b/src/Magento/FunctionalTestingFramework/Config/FileResolver/Primary.php
index 3a97694..cdec04d 100644
--- a/src/Magento/FunctionalTestingFramework/Config/FileResolver/Primary.php
+++ b/src/Magento/FunctionalTestingFramework/Config/FileResolver/Primary.php
@@ -68,6 +68,9 @@ class Primary implements FileResolverInterface
         } else {
             $defaultPath = dirname(dirname(dirname(dirname(__DIR__))));
             $defaultPath = str_replace('\\', DIRECTORY_SEPARATOR, $defaultPath);
+            // PATCH BEGIN
+            $magesuiteModuleDefaultPath = MAGENTO_BP . DIRECTORY_SEPARATOR . 'vendor' . DIRECTORY_SEPARATOR . 'creativestyle' . DIRECTORY_SEPARATOR . 'magesuite-mftf';
+            // PATCH END
             $patterns = [
                 $defaultPath . DIRECTORY_SEPARATOR . $scope . DIRECTORY_SEPARATOR . $filename,
                 $defaultPath . DIRECTORY_SEPARATOR . $scope . DIRECTORY_SEPARATOR . '*' . DIRECTORY_SEPARATOR
@@ -75,6 +78,9 @@ class Primary implements FileResolverInterface
                 FilePathFormatter::format(FW_BP) . $scope . DIRECTORY_SEPARATOR . $filename,
                 FilePathFormatter::format(FW_BP)  . $scope . DIRECTORY_SEPARATOR . '*' . DIRECTORY_SEPARATOR
                 . $filename
+                // PATCH BEGIN
+                ,$magesuiteModuleDefaultPath . DIRECTORY_SEPARATOR . $scope . DIRECTORY_SEPARATOR . $filename
+                // PATCH END
             ];
         }
         return str_replace(DIRECTORY_SEPARATOR, DIRECTORY_SEPARATOR, $patterns);
