diff --git a/Model/Processor/Product.php b/Model/Processor/Product.php
index d9da804..a4afef3 100644
--- a/Model/Processor/Product.php
+++ b/Model/Processor/Product.php
@@ -181,8 +181,8 @@ class Product extends Common {
         ));

         $node->appendChild($xml->createElementNS("http://www.toppak.nl/item",'eanNo',
-            $product->getEanNo()
-                ? $this->_cleanupString($product->getEanNo(), 15)
+            $product->getEan()
+                ? $this->_cleanupString($product->getEan(), 15)
                 : $this->_cleanupString($product->getSku(), 15)
         ));
