diff --git a/Model/Processor/Common.php b/Model/Processor/Common.php
index 2bde85a..f694868 100644
--- a/Model/Processor/Common.php
+++ b/Model/Processor/Common.php
@@ -70,7 +70,9 @@ class Common {
 
         $trimmed = trim(preg_replace('#\s+#us', ' ', str_replace($this->_getBadCharacters(), '', $string)));
         if ($maxLength && mb_strlen($string, 'UTF-8') > $maxLength)
-            $trimmed = mb_substr($string, 0, $maxLength, 'UTF-8');
+            //BEGIN PATCH
+            $trimmed = mb_substr($trimmed, 0, $maxLength, 'UTF-8');
+            //END PATCH
         return $trimmed;
     }
 
