diff --git a/Model/Processor/Product.php b/Model/Processor/Product.php
index 702165a..0f6e170 100644
--- a/Model/Processor/Product.php
+++ b/Model/Processor/Product.php
@@ -136,12 +136,12 @@ class Product extends Common {
         
         $productName = preg_replace('/[^A-Za-z0-9 .]/u','', strip_tags($product->getName()));
         $node->appendChild($xml->createElementNS("http://www.toppak.nl/item",'description', 
-            $this->_cleanupString($productName, 35)
+            $this->_cleanupString($productName, 60)
         ));
         
         $productDescription = preg_replace('/[^A-Za-z0-9 .]/u','', strip_tags($product->getShortDescription()));
         $node->appendChild($xml->createElementNS("http://www.toppak.nl/item",'description2', 
-            $this->_cleanupString($productDescription, 35)
+            $this->_cleanupString($productDescription, 30)
         ));
         
         $node->appendChild($xml->createElementNS("http://www.toppak.nl/item",'unitOfMeasure', 
