diff --git a/Model/Email/Template.php b/Model/Email/Template.php
index c7e8634..6265a7d 100755
--- a/Model/Email/Template.php
+++ b/Model/Email/Template.php
@@ -129,7 +129,10 @@ class Template extends \Magento\Framework\DataObject implements SyncInterface
         'sales_email_shipment_comment_guest_template' =>
             \Magento\Sales\Model\Order\Email\Container\ShipmentCommentIdentity::XML_PATH_EMAIL_GUEST_TEMPLATE,
         'contact_email_email_template' => 'contact/email/email_template',//interface don't exist on lower versions 2.2.2
-        'sendfriend_email_template' => \Magento\SendFriend\Helper\Data::XML_PATH_EMAIL_TEMPLATE,
+        // BEGIN PATCH
+        //'sendfriend_email_template' => \Magento\SendFriend\Helper\Data::XML_PATH_EMAIL_TEMPLATE,
+        'sendfriend_email_template' => 'sendfriend/email/template',
+        // PATCH END
         'product_stock_alert_template' => \Magento\ProductAlert\Model\Email::XML_PATH_EMAIL_STOCK_TEMPLATE,
         'product_price_alert_template' => \Magento\ProductAlert\Model\Email::XML_PATH_EMAIL_PRICE_TEMPLATE,
 
