diff --git a/Model/ResourceModel/Product/Compare/Item/Collection.php b/Model/ResourceModel/Product/Compare/Item/Collection.php
index 76f566a..f3ec823 100644
--- a/Model/ResourceModel/Product/Compare/Item/Collection.php
+++ b/Model/ResourceModel/Product/Compare/Item/Collection.php
@@ -437,7 +437,7 @@ class Collection extends \Magento\Catalog\Model\ResourceModel\Product\Collection
                 )->where(
                     'main_table.attribute_id IN(?)',
                     $attributeIds
-                );
+                )->order('position ASC');
                 $attributesData = $this->getConnection()->fetchAll($select);
                 if ($attributesData) {
                     $entityType = \Magento\Catalog\Model\Product::ENTITY;
--
2.30.1 (Apple Git-130)
