diff --git a/CustomerData/LastOrderedItems.php b/CustomerData/LastOrderedItems.php
index 91164da..f687a7d 100644
--- a/CustomerData/LastOrderedItems.php
+++ b/CustomerData/LastOrderedItems.php
@@ -143,13 +143,21 @@ class LastOrderedItems implements SectionSourceInterface
             foreach ($order->getParentItemsRandomCollection($limit) as $item) {
                 /** @var Product $product */
                 try {
+                    // BEGIN PATCH
+                    if (!$item->getProduct()) {
+                        continue;
+                    }
+                    // END PATCH
                     $product = $this->productRepository->getById(
                         $item->getProductId(),
                         false,
                         $this->_storeManager->getStore()->getId()
                     );
                 } catch (NoSuchEntityException $noEntityException) {
-                    $this->logger->critical($noEntityException);
+                    //$this->logger->critical($noEntityException);
+                    // BEGIN PATCH
+                    $this->logger->info($noEntityException);
+                    // END PATCH
                     continue;
                 }
                 if (in_array($website, $product->getWebsiteIds())) {
--
2.39.5 (Apple Git-154)

