diff --git a/Block/Html/Title.php b/Block/Html/Title.php
index 1189ee0..addd3f2 100644
--- a/Block/Html/Title.php
+++ b/Block/Html/Title.php
@@ -79,7 +79,10 @@ class Title extends Template
     {
         $pageTitle = !empty($this->pageTitle) ? $this->pageTitle : $this->pageConfig->getTitle()->getShortHeading();
 
-        return $this->shouldTranslateTitle() ? __($pageTitle) : $pageTitle;
+        //BEGIN PATCH
+        //return $this->shouldTranslateTitle() ? __($pageTitle) : $pageTitle;
+        return $this->shouldTranslateTitle() ? __($pageTitle) : (string)$pageTitle;
+        //END PATCH
     }
 
     /**
