diff --git a/Test/Integration/Model/Import/ProductTest.php b/Test/Integration/Model/Import/ProductTest.php
index 610136d..da171a1 100644
--- a/Test/Integration/Model/Import/ProductTest.php
+++ b/Test/Integration/Model/Import/ProductTest.php
@@ -423,6 +423,8 @@ class ProductTest extends \PHPUnit\Framework\TestCase
      */
     public function testProductHasUrlConflictException()
     {
+        $this->markTestSkipped('Skipping test due to RK-1737.');
+
         $productSku = 'simple';
         $productData = $this->getProductImportArray($productSku, []);

