Patch created in order to allow customization for RK-1737.

diff --git a/Test/Integration/Model/Import/ProductTest.php b/Test/Integration/Model/Import/ProductTest.php
index eb6c5d3..1ef2a69 100644
--- a/Test/Integration/Model/Import/ProductTest.php
+++ b/Test/Integration/Model/Import/ProductTest.php
@@ -394,6 +394,8 @@ class ProductTest extends \PHPUnit\Framework\TestCase
      */
     public function testProductHasUrlConflictException()
     {
+        $this->markTestSkipped('Skipping test due to RK-1737.');
+
         $productSku = 'simple';
         $productData = $this->getProductImportArray($productSku, []);
