diff --git a/Cron/NewsletterSyncCron.php b/Cron/NewsletterSyncCron.php
index 64753dc..6123c8d 100644
--- a/Cron/NewsletterSyncCron.php
+++ b/Cron/NewsletterSyncCron.php
@@ -158,7 +158,12 @@ class NewsletterSyncCron
                         $this->logger->error('MODULE Order: ' . $e->getMessage());
                     }
                 }
-                $this->activeCampaignHelper->setLastSync(strtotime(date('Y-m-d H:i:s')));
+                // BEGIN PATCH
+                // $this->activeCampaignHelper->setLastSync(strtotime(date('Y-m-d H:i:s')));
+                $dateTimestamp = new \DateTime();
+                $timestamp = $dateTimestamp->getTimestamp();
+                $this->activeCampaignHelper->setLastSync($timestamp);
+                // END PATCH
             }
         } catch (\Exception $e) {
             $this->logger->error('MODULE Order: ' . $e->getMessage());
