diff --git a/Model/Category/Attribute/Backend/Image.php b/Model/Category/Attribute/Backend/Image.php
index 6a035a4..cd450e2 100644
--- a/Model/Category/Attribute/Backend/Image.php
+++ b/Model/Category/Attribute/Backend/Image.php
@@ -6,7 +6,6 @@
 namespace Magento\Catalog\Model\Category\Attribute\Backend;

 use Magento\Framework\App\Filesystem\DirectoryList;
-use Magento\Framework\File\Uploader;

 /**
  * Catalog category image attribute backend model
@@ -71,8 +70,7 @@ class Image extends \Magento\Eav\Model\Entity\Attribute\Backend\AbstractBackend

     /**
      * Gets image name from $value array.
-     *
-     * Will return empty string in a case when $value is not an array.
+     * Will return empty string in a case when $value is not an array
      *
      * @param array $value Attribute value
      * @return string
@@ -87,28 +85,8 @@ class Image extends \Magento\Eav\Model\Entity\Attribute\Backend\AbstractBackend
     }

     /**
-     * Check that image name exists in catalog/category directory and return new image name if it already exists.
-     *
-     * @param string $imageName
-     * @return string
-     */
-    private function checkUniqueImageName(string $imageName): string
-    {
-        $imageUploader = $this->getImageUploader();
-        $mediaDirectory = $this->_filesystem->getDirectoryWrite(DirectoryList::MEDIA);
-        $imageAbsolutePath = $mediaDirectory->getAbsolutePath(
-            $imageUploader->getBasePath() . DIRECTORY_SEPARATOR . $imageName
-        );
-
-        $imageName = Uploader::getNewFilename($imageAbsolutePath);
-
-        return $imageName;
-    }
-
-    /**
-     * Avoiding saving potential upload data to DB.
-     *
-     * Will set empty image attribute value if image was not uploaded.
+     * Avoiding saving potential upload data to DB
+     * Will set empty image attribute value if image was not uploaded
      *
      * @param \Magento\Framework\DataObject $object
      * @return $this
@@ -125,7 +103,6 @@ class Image extends \Magento\Eav\Model\Entity\Attribute\Backend\AbstractBackend
         }

         if ($imageName = $this->getUploadedImageName($value)) {
-            $imageName = $this->checkUniqueImageName($imageName);
             $object->setData($this->additionalData . $attributeName, $value);
             $object->setData($attributeName, $imageName);
         } elseif (!is_string($value)) {
@@ -136,8 +113,6 @@ class Image extends \Magento\Eav\Model\Entity\Attribute\Backend\AbstractBackend
     }

     /**
-     * Get Instance of Category Image Uploader.
-     *
      * @return \Magento\Catalog\Model\ImageUploader
      *
      * @deprecated 101.0.0
@@ -178,11 +153,9 @@ class Image extends \Magento\Eav\Model\Entity\Attribute\Backend\AbstractBackend
         $fileUrl = ltrim($value[0]['url'], '/');
         $baseMediaDir = $this->_filesystem->getUri(DirectoryList::MEDIA);

-        if (!$baseMediaDir) {
-            return false;
-        }
+        $usingPathRelativeToBase = strpos($fileUrl, $baseMediaDir) === 0;

-        return strpos($fileUrl, $baseMediaDir) === 0;
+        return $usingPathRelativeToBase;
     }

     /**
