diff --git a/Helper/Progress.php b/Helper/Progress.php
index 75e8257..f3fbd19 100644
--- a/Helper/Progress.php
+++ b/Helper/Progress.php
@@ -304,7 +304,10 @@ class Progress extends \Magento\Framework\App\Helper\AbstractHelper
             $status = $exploded[0];
             $message = $exploded[1];
 
-            $percent = floor($exploded[2] / 2);
+            // BEGIN PATCH
+            // $percent = floor($exploded[2] / 2);
+            $percent = floor((float) $exploded[2] / 2);
+            // END PATCH
             if ($status == self::SUCCEEDED) {
                 $status = "\e[92m" . $status . "\e[39m";
                 fwrite(STDOUT, sprintf("\r %s %-20s [%-" . $percent . "s>%-" . (50 - $percent) . "s] %s%% : %-100s", $time, $status, str_pad("", $percent, "~"), str_pad("", 50 - $percent, " "), $percent * 2, $message));
