diff --git a/src/module-elasticsuite-virtual-category/Model/ResourceModel/Product/Indexer/Fulltext/Datasource/CategoryData.php b/src/module-elasticsuite-virtual-category/Model/ResourceModel/Product/Indexer/Fulltext/Datasource/CategoryData.php
index 48aeadb..b79a236 100644
--- a/src/module-elasticsuite-virtual-category/Model/ResourceModel/Product/Indexer/Fulltext/Datasource/CategoryData.php
+++ b/src/module-elasticsuite-virtual-category/Model/ResourceModel/Product/Indexer/Fulltext/Datasource/CategoryData.php
@@ -77,7 +77,10 @@ class CategoryData extends \Smile\ElasticsuiteCatalog\Model\ResourceModel\Produc
                 'product_id'     => 'cpi.product_id',
                 'is_parent'      => 'cpi.is_parent',
                 'is_virtual'     => new \Zend_Db_Expr('"false"'),
-                'position'       => 'p.position',
+                // BEGIN PATCH
+                //'position'       => 'p.position',
+                'position'       => 'cpi.position',
+                // END PATCH
                 'is_blacklisted' => 'p.is_blacklisted',
             ]);
 
@@ -110,7 +113,10 @@ class CategoryData extends \Smile\ElasticsuiteCatalog\Model\ResourceModel\Produc
                 'product_id'     => 'cpi.product_id',
                 'is_parent'      => 'cpi.is_parent',
                 'is_virtual'     => new \Zend_Db_Expr('"false"'),
-                'position'       => 'p.position',
+                // BEGIN PATCH
+                //'position'       => 'p.position',
+                'position'       => 'cpi.position',
+                // END PATCH
                 'is_blacklisted' => 'p.is_blacklisted',
             ]);
 
