diff --git a/src/module-elasticsuite-catalog/Model/Eav/Indexer/Fulltext/Datasource/AbstractAttributeData.php b/src/module-elasticsuite-catalog/Model/Eav/Indexer/Fulltext/Datasource/AbstractAttributeData.php
index 824e461..1b002f0 100644
--- a/src/module-elasticsuite-catalog/Model/Eav/Indexer/Fulltext/Datasource/AbstractAttributeData.php
+++ b/src/module-elasticsuite-catalog/Model/Eav/Indexer/Fulltext/Datasource/AbstractAttributeData.php
@@ -70,6 +70,7 @@ abstract class AbstractAttributeData
         \Magento\Eav\Model\Entity\Attribute\Backend\DefaultBackend::class,
         \Magento\Catalog\Model\Product\Attribute\Backend\Weight::class,
         \Magento\Catalog\Model\Product\Attribute\Backend\Price::class,
+        \Magento\Catalog\Model\Product\Attribute\Backend\Url::class,
     ];

     /**
