diff --git a/view/base/requirejs-config.js b/view/base/requirejs-config.js
index 52e9270..cd23597 100644
--- a/view/base/requirejs-config.js
+++ b/view/base/requirejs-config.js
@@ -46,7 +46,8 @@ var config = {
         'tinycolor': 'jquery/spectrum/tinycolor'
     },
     'deps': [
-        'jquery/jquery-migrate'
+        'jquery/jquery-migrate',
+        'jquery/jquery.cookie'
     ],
     'config': {
         'mixins': {
