diff --git a/Gateway/Request/RefundDataBuilder.php b/Gateway/Request/RefundDataBuilder.php
index d35c234..76716c4 100644
--- a/Gateway/Request/RefundDataBuilder.php
+++ b/Gateway/Request/RefundDataBuilder.php
@@ -35,6 +35,7 @@ class RefundDataBuilder implements BuilderInterface
     private Config $configHelper;
     private PaymentCollectionFactory $orderPaymentCollectionFactory;
     private ChargedCurrency $chargedCurrency;
+    private OpenInvoice $openInvoiceHelper;
 
     public function __construct(
         Data $adyenHelper,
