diff --git a/Setup/Patch/DeclarativeSchemaApplyBefore/DropIndexTable.php b/Setup/Patch/DeclarativeSchemaApplyBefore/DropIndexTable.php
index 4bd10a3..69b7cc6 100644
--- a/Setup/Patch/DeclarativeSchemaApplyBefore/DropIndexTable.php
+++ b/Setup/Patch/DeclarativeSchemaApplyBefore/DropIndexTable.php
@@ -49,9 +49,11 @@ class DropIndexTable implements SchemaPatchInterface
         $connection = $this->schemaSetup->getConnection();
         $indexTable = $this->schemaSetup->getTable(Uninstall::AMASTY_LABEL_INDEX_TABLE);
 
-        if ($connection->isTableExists($indexTable)) {
-            $connection->dropTable($indexTable);
-        }
+        // PATCH START
+        // if ($connection->isTableExists($indexTable)) {
+        //    $connection->dropTable($indexTable);
+        // }
+        // PATCH END
 
         return $this;
     }
