diff --git a/Model/SendinblueSib.php b/Model/SendinblueSib.php
index 4a1e57f..339b95c 100644
--- a/Model/SendinblueSib.php
+++ b/Model/SendinblueSib.php
@@ -51,7 +51,7 @@ class SendinblueSib extends \Magento\Framework\Model\AbstractModel
         $this->_orderCollectionFactory = $salesOrderCollectionFactory;
         $this->_resourceConfig = $resourceConfig;
         $this->_getValueDefault = $scopeDefaultType;
-        $this->_scopeTypeDefault = $scopeDefaultType::SCOPE_TYPE_DEFAULT;
+        $this->_scopeTypeDefault = "stores";
         $this->_storeId = $defaultStoreId::DEFAULT_STORE_ID;
         $this->_dir = $dir;
         $this->_customers = $customers;

diff --git a/Plugin/CustomTransportPlugin.php b/Plugin/CustomTransportPlugin.php
index 89dd3c0..b44eda5 100644
--- a/Plugin/CustomTransportPlugin.php
+++ b/Plugin/CustomTransportPlugin.php
@@ -32,7 +32,7 @@ class CustomTransportPlugin
                 return null;
             }

-            $smtpHost = "smtp-relay.brevo.com";
+            $smtpHost = $this->scopeConfig->getValue('sendinblue/smtp_host', \Magento\Store\Model\ScopeInterface::SCOPE_STORE);
             $smtpAuthentication = $this->scopeConfig->getValue('sendinblue/smtp_authentication', \Magento\Store\Model\ScopeInterface::SCOPE_STORE);
             $smtpTls = $this->scopeConfig->getValue('sendinblue/smtp_tls', \Magento\Store\Model\ScopeInterface::SCOPE_STORE);
             $smtpPort = $this->scopeConfig->getValue('sendinblue/smtp_port', \Magento\Store\Model\ScopeInterface::SCOPE_STORE);
