diff --git a/src/module-elasticsuite-catalog/Model/Layer/Filter/DecimalFilterTrait.php b/src/module-elasticsuite-catalog/Model/Layer/Filter/DecimalFilterTrait.php
index d908319..734c19f 100644
--- a/src/module-elasticsuite-catalog/Model/Layer/Filter/DecimalFilterTrait.php
+++ b/src/module-elasticsuite-catalog/Model/Layer/Filter/DecimalFilterTrait.php
@@ -50,11 +50,8 @@ trait DecimalFilterTrait

                 list($fromValue, $toValue) = $filter;
                 $this->setCurrentValue(['from' => $fromValue, 'to' => $toValue]);
-
-                $bounds = array_filter(['gte' => $fromValue, 'lt' => $toValue]);
-                if ($fromValue === $toValue) {
-                    $bounds = array_filter(['gte' => $fromValue, 'lte' => $toValue]);
-                }
+
+                $bounds = array_filter(['gte' => $fromValue, 'lte' => $toValue]);

                 $this->getLayer()->getProductCollection()->addFieldToFilter(
                     $this->getFilterField(),
