diff --git a/src/ReportApi/Service/SelectService.php b/src/ReportApi/Service/SelectService.php
index f083d5d..4e078a8 100644
--- a/src/ReportApi/Service/SelectService.php
+++ b/src/ReportApi/Service/SelectService.php
@@ -242,15 +242,18 @@ class SelectService implements SelectServiceInterface
         $select->where($baseTable->getPkField()->toDbExpr() . '>0');
 
         foreach ($request->getFilters() as $filter) {
-            // with some additional filters not all products might de present in tmp tables.
-            // so we apply only primary filters (toolbar) and internal filters (report builder)
-            if (
-                $report
-                && !in_array($filter->getColumn(), $report->getPrimaryFilters())
-                && !in_array($filter->getColumn(), $report->getInternalFilters())
-            ) {
-                continue;
-            }
+            // BEGIN PATCH
+            // Revert changes introduced in versions 1.0.59 and 1.0.60 because of wrong result of reports used by iPet
+            // // with some additional filters not all products might de present in tmp tables.
+            // // so we apply only primary filters (toolbar) and internal filters (report builder)
+            // if (
+            //     $report
+            //     && !in_array($filter->getColumn(), $report->getPrimaryFilters())
+            //     && !in_array($filter->getColumn(), $report->getInternalFilters())
+            // ) {
+            //     continue;
+            // }
+            // END PATCH
 
             $col = $this->schema->getColumn($filter->getColumn());
 
