diff --git a/src/ReportApi/Service/SelectService.php b/src/ReportApi/Service/SelectService.php
index ac3ce60..35016e2 100644
--- a/src/ReportApi/Service/SelectService.php
+++ b/src/ReportApi/Service/SelectService.php
@@ -236,14 +236,17 @@ class SelectService implements SelectServiceInterface
         $select->where($baseTable->getPkField()->toDbExpr() . '>0');
 
         foreach ($request->getFilters() as $filter) {
-            if (
-                strpos($filter->getColumn(), 'created_at') === false
-                && strpos($filter->getColumn(), 'updated_at') === false
-            ) {
-                // some filters in queries for TMP tables can result in not all needed products present in TMP tables
-                // we apply only date filters to TMP tables
-                continue;
-            }
+            // BEGIN PATCH
+            // Revert changes introduced in versions 1.0.59 and 1.0.60 because of wrong result of reports used by iPet
+            // if (
+            //     strpos($filter->getColumn(), 'created_at') === false
+            //     && strpos($filter->getColumn(), 'updated_at') === false
+            // ) {
+            //     // some filters in queries for TMP tables can result in not all needed products present in TMP tables
+            //     // we apply only date filters to TMP tables
+            //     continue;
+            // }
+            // END PATCH
 
             $col = $this->schema->getColumn($filter->getColumn());
 
