diff --git a/Api/Data/GiftRuleInterface.php b/Api/Data/GiftRuleInterface.php
index c1c7c40..510af92 100644
--- a/Api/Data/GiftRuleInterface.php
+++ b/Api/Data/GiftRuleInterface.php
@@ -32,8 +32,6 @@ interface GiftRuleInterface extends ExtensibleDataInterface
     const SALESRULE_ID = 'salesrule_id';
     const SKU = 'sku';
     const TYPE = 'type';
-    const TOP_BANNER_SHOW_GIFT_IMAGES = 'top_banner_show_gift_images';
-    const AFTER_PRODUCT_BANNER_SHOW_GIFT_IMAGES = 'after_product_banner_show_gift_images';
     const ITEMS_DISCOUNT = 'items_discount';
     const MINIMAL_ITEMS_PRICE = 'minimal_items_price';
     const APPLY_TAX = 'apply_tax';
@@ -64,30 +62,6 @@ interface GiftRuleInterface extends ExtensibleDataInterface
      */
     public function setType($type);

-    /**
-     * @return int
-     */
-    public function getAfterProductBannerShowGiftImages();
-
-    /**
-     * @param int $afterProductBannerShowGiftImages
-     *
-     * @return \Amasty\Promo\Api\Data\GiftRuleInterface
-     */
-    public function setAfterProductBannerShowGiftImages($afterProductBannerShowGiftImages);
-
-    /**
-     * @return int
-     */
-    public function getTopBannerShowGiftImages();
-
-    /**
-     * @param int $topBannerShowGiftImages
-     *
-     * @return \Amasty\Promo\Api\Data\GiftRuleInterface
-     */
-    public function setTopBannerShowGiftImages($topBannerShowGiftImages);
-
     /**
      * @return string|null
      */
diff --git a/Model/ImageMigrate.php b/Model/ImageMigrate.php
index 7306d82..868af95 100644
--- a/Model/ImageMigrate.php
+++ b/Model/ImageMigrate.php
@@ -8,13 +8,6 @@

 namespace Amasty\Promo\Model;

-use Amasty\BannersLite\Api\BannerRepositoryInterface;
-use Amasty\BannersLite\Api\BannerRuleRepositoryInterface;
-use Amasty\BannersLite\Api\Data\BannerInterface;
-use Amasty\BannersLite\Model\Banner;
-use Amasty\BannersLite\Model\BannerImageUpload;
-use Amasty\BannersLite\Model\BannerRule;
-use Amasty\BannersLite\Model\ImageProcessor;
 use Amasty\Base\Model\Serializer;
 use Exception;
 use Magento\Framework\App\Filesystem\DirectoryList;
@@ -39,22 +32,12 @@ class ImageMigrate
      */
     private $mediaPath = '';

-    /**
-     * Loaded banners model grouped by sales rule id.
-     *
-     * @var Banner[]
-     */
-    private $loadedBanners = [];

     /**
      * @var LoggerInterface
      */
     private $logger;

-    /**
-     * @var BannerImageUpload
-     */
-    private $bannerImageUpload;

     /**
      * @var Serializer
@@ -71,38 +54,25 @@ class ImageMigrate
      */
     private $storeManager;

-    /**
-     * @var BannerRepositoryInterface
-     */
-    private $bannerRepository;

     /**
      * @var ResourceModel\Rule\CollectionFactory
      */
     private $collectionFactory;

-    /**
-     * @var BannerRuleRepositoryInterface
-     */
-    private $bannerRuleRepository;

     public function __construct(
         LoggerInterface $logger,
-        BannerImageUpload $bannerImageUpload,
         Serializer $serializer,
         Filesystem $filesystem,
         StoreManagerInterface $storeManager,
-        BannerRepositoryInterface $bannerRepository,
-        BannerRuleRepositoryInterface $bannerRuleRepository,
         ResourceModel\Rule\CollectionFactory $collectionFactory
-    ) {
+    )
+    {
         $this->logger = $logger;
-        $this->bannerImageUpload = $bannerImageUpload;
         $this->serializer = $serializer;
         $this->filesystem = $filesystem;
         $this->storeManager = $storeManager;
-        $this->bannerRepository = $bannerRepository;
-        $this->bannerRuleRepository = $bannerRuleRepository;
         $this->collectionFactory = $collectionFactory;
     }

@@ -110,189 +80,12 @@ class ImageMigrate
     {
         /** @var Rule[] $rules */
         $rules = $this->collectionFactory->create()->getItems();
-        $this->bannerImageUpload->setBaseTmpPath(static::BASE_TMP_DIRECTORY);
         $this->mediaPath = $this->storeManager->getStore()
                 ->getBaseUrl(UrlInterface::URL_TYPE_MEDIA) . $this->bannerImageUpload->getBasePath();

-        foreach ($rules as $rule) {
-            $this->saveBannerData(
-                BannerInterface::TOP_BANNER,
-                $rule->getData('top_banner_image'),
-                $rule->getData('top_banner_alt'),
-                $rule->getData('top_banner_on_hover_text'),
-                $rule->getData('top_banner_link'),
-                $rule->getSalesruleId()
-            );
-            $this->saveBannerData(
-                BannerInterface::AFTER_BANNER,
-                $rule->getData('after_product_banner_image'),
-                $rule->getData('after_product_banner_alt'),
-                $rule->getData('after_product_banner_on_hover_text'),
-                $rule->getData('after_product_banner_link'),
-                $rule->getSalesruleId()
-            );
-            $this->saveBannerData(
-                BannerInterface::PRODUCT_LABEL,
-                $rule->getData('label_image'),
-                $rule->getData('label_image_alt'),
-                null,
-                null,
-                $rule->getSalesruleId()
-            );
-        }
-    }
-
-    /**
-     * @param int $type
-     * @param string $image
-     * @param string $alt
-     * @param string $hover
-     * @param string $link
-     * @param int $salesRuleId
-     *
-     * @return bool
-     * @throws LocalizedException
-     */
-    private function saveBannerData($type, $image, $alt, $hover, $link, $salesRuleId)
-    {
-        if (!($image || $alt || $hover || $link)) {
-            return false;
-        }
-
-        if (!($bannerImage = $this->moveImage($image))) {
-            return false;
-        }
-
-        try {
-            /** @var BannerRule $bannerRuleModel */
-            $bannerRuleModel = $this->bannerRuleRepository->getBySalesruleId($salesRuleId);
-        } catch (NoSuchEntityException $exception) {
-            /** @var BannerRule $bannerRuleModel */
-            $bannerRuleModel = $this->bannerRuleRepository->getEmptyModel();
-        }
-
-        $bannerRuleModel->setSalesruleId($salesRuleId);
-        $bannerRuleModel->setShowBannerFor(0);
-
-        $bannerModel = $this->getBannerBySalesRule($salesRuleId, $type);
-
-        $bannerModel
-            ->setBannerType($type)
-            ->setBannerImage($bannerImage)
-            ->setBannerAlt($alt)
-            ->setBannerHoverText($hover)
-            ->setBannerLink($link)
-            ->setSalesruleId($salesRuleId);
-
-        try {
-            $this->bannerRuleRepository->save($bannerRuleModel);
-            $this->bannerRepository->save($bannerModel);
-        } catch (CouldNotSaveException $exception) {
-            $this->logger->critical($exception);
-
-            throw new LocalizedException(
-                __(
-                    'Could not migrate banners data. Exception message: %1. For details see exception log.',
-                    $exception->getMessage()
-                )
-            );
-        }
-
-        return true;
     }
+}

-    /**
-     * @param int $ruleId
-     * @param int $bannerType
-     *
-     * @return Banner
-     */
-    private function getBannerBySalesRule($ruleId, $bannerType)
-    {
-        $bannerModel = false;
-
-        try {
-            if (!isset($this->loadedBanners[$ruleId])) {
-                $this->loadedBanners[$ruleId] = $this->bannerRepository->getBySalesruleId($ruleId);
-            }
-
-            foreach ($this->loadedBanners[$ruleId] as $promoBanner) {
-                if ($promoBanner->getBannerType() == $bannerType) {
-                    $bannerModel = $promoBanner;
-                    break;
-                }
-            }
-        } catch (NoSuchEntityException $exception) {
-            $bannerModel = false;
-        }
-        if (!$bannerModel) {
-            $bannerModel = $this->bannerRepository->getEmptyModel();
-            $this->loadedBanners[$ruleId][] = $bannerModel;
-        }
-
-        return $bannerModel;
-    }
-
-    /**
-     * Move image to Banners Lite folder.
-     * If exception is caught continue without image data.
-     *
-     * @param string $imageSerialized
-     *
-     * @return string|false
-     */
-    private function moveImage($imageSerialized)
-    {
-        $imageData = $this->serializer->unserialize($imageSerialized);
-
-        if (!$imageData) {
-            return false;
-        }
-
-        if (is_array(current($imageData))) {
-            $imageData = current($imageData);
-        }
-
-        try {
-            if ($this->whetherToMoveFile($imageData['name'])) {
-                $this->bannerImageUpload->moveFileFromTmp($imageData['name']);
-            }
-            $imageData['cookie']['upload'] = true;
-            $imageData['url'] = $this->mediaPath . DIRECTORY_SEPARATOR . $imageData['name'];
-        } catch (Exception $exception) {
-            $this->logger->critical($exception);
-
-            return false;
-        }

-        $imageSerialized = $this->serializer->serialize([$imageData]);

-        return $imageSerialized;
-    }

-    /**
-     * @param string $filename
-     *
-     * @return bool
-     *
-     * @throws FileNotFoundException
-     */
-    private function whetherToMoveFile($filename)
-    {
-        //check file exist in Amasty_Promo banners folder
-        $isExist = $this->filesystem->getDirectoryRead(DirectoryList::MEDIA)->isExist(
-            $this->bannerImageUpload->getFilePath(self::BASE_TMP_DIRECTORY, $filename)
-        );
-
-        //check file exists in Amasty_BannersLite folder
-        $haveSameFile = $this->filesystem->getDirectoryRead(DirectoryList::MEDIA)->isExist(
-            $this->bannerImageUpload->getFilePath(ImageProcessor::BANNERS_MEDIA_PATH, $filename)
-        );
-
-        if (!($isExist || $haveSameFile)) {
-            throw new FileNotFoundException('Could not find original banner file to migrate');
-        }
-
-        return $isExist && !$haveSameFile;
-    }
-}
diff --git a/Model/Rule.php b/Model/Rule.php
index 8a8aa3b..9b6472d 100644
--- a/Model/Rule.php
+++ b/Model/Rule.php
@@ -105,42 +105,6 @@ class Rule extends \Magento\Framework\Model\AbstractModel implements GiftRuleInt
         return $this;
     }

-    /**
-     * @inheritdoc
-     */
-    public function getAfterProductBannerShowGiftImages()
-    {
-        return $this->_getData(GiftRuleInterface::AFTER_PRODUCT_BANNER_SHOW_GIFT_IMAGES);
-    }
-
-    /**
-     * @inheritdoc
-     */
-    public function setAfterProductBannerShowGiftImages($afterProductBannerShowGiftImages)
-    {
-        $this->setData(GiftRuleInterface::AFTER_PRODUCT_BANNER_SHOW_GIFT_IMAGES, $afterProductBannerShowGiftImages);
-
-        return $this;
-    }
-
-    /**
-     * @inheritdoc
-     */
-    public function getTopBannerShowGiftImages()
-    {
-        return $this->_getData(GiftRuleInterface::TOP_BANNER_SHOW_GIFT_IMAGES);
-    }
-
-    /**
-     * @inheritdoc
-     */
-    public function setTopBannerShowGiftImages($topBannerShowGiftImages)
-    {
-        $this->setData(GiftRuleInterface::TOP_BANNER_SHOW_GIFT_IMAGES, $topBannerShowGiftImages);
-
-        return $this;
-    }
-
     /**
      * @inheritdoc
      */
diff --git a/composer.json.rej b/composer.json.rej
new file mode 100644
index 0000000..ad20513
--- /dev/null
+++ b/composer.json.rej
@@ -0,0 +1,9 @@
+diff a/composer.json b/composer.json	(rejected hunks)
+@@ -5,7 +5,6 @@
+         "php": "7.0.2|7.0.4|~7.0.6|^7.1.0",
+         "amasty/base": ">=1.4.5",
+         "amasty/module-advanced-conditions": ">=1.3.6",
+-        "amasty/module-banners-lite": ">=1.0.5",
+         "amasty/module-salesrulewizard": "*",
+         "amasty/rgrid": "*"
+     },
diff --git a/etc/di.xml b/etc/di.xml
index 3967db6..61a32a7 100644
--- a/etc/di.xml
+++ b/etc/di.xml
@@ -98,11 +98,6 @@
         <plugin name="AmastyRules::PreviewQuotaFix" type="Amasty\Promo\Plugin\CheckoutStaging\Model\ResourceModel\PreviewQuotaPlugin"/>
     </type>
     <!--finish magneto ee bug fixes-->
-    <type name="Amasty\Promo\Block\Banner">
-        <arguments>
-            <argument name="bannerProvider" xsi:type="object">Amasty\BannersLite\Model\ProductBannerProvider\Proxy</argument>
-        </arguments>
-    </type>
     <type name="Magento\Quote\Api\ShipmentEstimationInterface">
         <plugin name="AmastyPromo::restrictAutoAddOnShippingEstimate" type="Amasty\Promo\Plugin\Quote\Model\ShippingMethodManagementPlugin"/>
     </type>
diff --git a/etc/module.xml b/etc/module.xml
index c8ae284..8eb124d 100644
--- a/etc/module.xml
+++ b/etc/module.xml
@@ -14,7 +14,6 @@
             <module name="Magento_Checkout" />
             <module name="Magento_ConfigurableProduct" />
             <module name="Amasty_Conditions"/>
-            <module name="Amasty_BannersLite"/>
         </sequence>
     </module>
 </config>
diff --git a/view/adminhtml/ui_component/sales_rule_form.xml b/view/adminhtml/ui_component/sales_rule_form.xml
index 540514f..4ebf213 100644
--- a/view/adminhtml/ui_component/sales_rule_form.xml
+++ b/view/adminhtml/ui_component/sales_rule_form.xml
@@ -128,48 +128,5 @@
             </argument>
         </field>
     </fieldset>
-    <fieldset name="promo_banners_highlights">
-        <fieldset name="top_banner">
-            <field name="top_banner_show_gift_images">
-                <argument name="data" xsi:type="array">
-                    <item name="config" xsi:type="array">
-                        <item name="dataType" xsi:type="string">boolean</item>
-                        <item name="formElement" xsi:type="string">checkbox</item>
-                        <item name="component" xsi:type="string">Amasty_Promo/js/form/element/gift-image-toggle</item>
-                        <item name="prefer" xsi:type="string">toggle</item>
-                        <item name="valueMap" xsi:type="array">
-                            <item name="true" xsi:type="number">1</item>
-                            <item name="false" xsi:type="number">0</item>
-                        </item>
-                        <item name="default" xsi:type="number">0</item>
-                        <item name="label" xsi:type="string" translate="true">Show Gift images</item>
-                        <item name="source" xsi:type="string">sales_rule</item>
-                        <item name="dataScope" xsi:type="string">extension_attributes.ampromo_rule.top_banner_show_gift_images</item>
-                        <item name="sortOrder" xsi:type="number">50</item>
-                    </item>
-                </argument>
-            </field>
-        </fieldset>
-        <fieldset name="after_description_banner">
-            <field name="after_product_banner_show_gift_images">
-                <argument name="data" xsi:type="array">
-                    <item name="config" xsi:type="array">
-                        <item name="dataType" xsi:type="string">boolean</item>
-                        <item name="formElement" xsi:type="string">checkbox</item>
-                        <item name="component" xsi:type="string">Amasty_Promo/js/form/element/gift-image-toggle</item>
-                        <item name="prefer" xsi:type="string">toggle</item>
-                        <item name="valueMap" xsi:type="array">
-                            <item name="true" xsi:type="number">1</item>
-                            <item name="false" xsi:type="number">0</item>
-                        </item>
-                        <item name="default" xsi:type="number">0</item>
-                        <item name="label" xsi:type="string" translate="true">Show Gift images</item>
-                        <item name="source" xsi:type="string">sales_rule</item>
-                        <item name="dataScope" xsi:type="string">extension_attributes.ampromo_rule.after_product_banner_show_gift_images</item>
-                        <item name="sortOrder" xsi:type="number">50</item>
-                    </item>
-                </argument>
-            </field>
-        </fieldset>
-    </fieldset>
+
 </form>
\ No newline at end of file
diff --git a/view/adminhtml/ui_component/salesrulestaging_update_form.xml b/view/adminhtml/ui_component/salesrulestaging_update_form.xml
index 09a37d1..26381e4 100644
--- a/view/adminhtml/ui_component/salesrulestaging_update_form.xml
+++ b/view/adminhtml/ui_component/salesrulestaging_update_form.xml
@@ -126,48 +126,5 @@
             </argument>
         </field>
     </fieldset>
-    <fieldset name="promo_banners_highlights">
-        <fieldset name="top_banner">
-            <field name="top_banner_show_gift_images">
-                <argument name="data" xsi:type="array">
-                    <item name="config" xsi:type="array">
-                        <item name="dataType" xsi:type="string">boolean</item>
-                        <item name="formElement" xsi:type="string">checkbox</item>
-                        <item name="component" xsi:type="string">Amasty_Promo/js/form/element/gift-image-toggle</item>
-                        <item name="prefer" xsi:type="string">toggle</item>
-                        <item name="valueMap" xsi:type="array">
-                            <item name="true" xsi:type="number">1</item>
-                            <item name="false" xsi:type="number">0</item>
-                        </item>
-                        <item name="default" xsi:type="number">0</item>
-                        <item name="label" xsi:type="string" translate="true">Show Gift images</item>
-                        <item name="source" xsi:type="string">sales_rule</item>
-                        <item name="dataScope" xsi:type="string">extension_attributes.ampromo_rule.top_banner_show_gift_images</item>
-                        <item name="sortOrder" xsi:type="number">50</item>
-                    </item>
-                </argument>
-            </field>
-        </fieldset>
-        <fieldset name="after_description_banner">
-            <field name="after_product_banner_show_gift_images">
-                <argument name="data" xsi:type="array">
-                    <item name="config" xsi:type="array">
-                        <item name="dataType" xsi:type="string">boolean</item>
-                        <item name="formElement" xsi:type="string">checkbox</item>
-                        <item name="component" xsi:type="string">Amasty_Promo/js/form/element/gift-image-toggle</item>
-                        <item name="prefer" xsi:type="string">toggle</item>
-                        <item name="valueMap" xsi:type="array">
-                            <item name="true" xsi:type="number">1</item>
-                            <item name="false" xsi:type="number">0</item>
-                        </item>
-                        <item name="default" xsi:type="number">0</item>
-                        <item name="label" xsi:type="string" translate="true">Show Gift images</item>
-                        <item name="source" xsi:type="string">sales_rule</item>
-                        <item name="dataScope" xsi:type="string">extension_attributes.ampromo_rule.after_product_banner_show_gift_images</item>
-                        <item name="sortOrder" xsi:type="number">50</item>
-                    </item>
-                </argument>
-            </field>
-        </fieldset>
-    </fieldset>
+
 </form>
\ No newline at end of file
diff --git a/view/frontend/web/css/banner.css b/view/frontend/web/css/banner.css
deleted file mode 100644
index 5b71b98..0000000
--- a/view/frontend/web/css/banner.css
+++ /dev/null
@@ -1,34 +0,0 @@
-.ampromo-gift a.product-image{
-    text-decoration: none;
-}
-
-ul.ampromo-gift {
-    padding: 0;
-    list-style-type: none;
-}
-
-.ampromo-gift li{
-    margin: 10px 10px;
-    display: inline;
-}
-
-.ampromo-banner-tooltip{
-    padding:2px 5px;
-    background:#FEC;
-    border:1px solid #FAA819;
-    color:#1F5070;
-    text-align:left;
-    width:300px;
-}
-
-.ampromo-product-name{
-    font-weight: bold;
-}
-
-.ampromo-short-description{
-}
-
-.ampromo-label {
-    position: absolute;
-    z-index: 1000;
-}
\ No newline at end of file
