diff --git a/Observer/Controller/Action/Predispatch.php b/Observer/Controller/Action/Predispatch.php
index a445f55..fc17c97 100644
--- a/Observer/Controller/Action/Predispatch.php
+++ b/Observer/Controller/Action/Predispatch.php
@@ -39,13 +39,13 @@ class Predispatch implements \Magento\Framework\Event\ObserverInterface
     {
         $coupon = $this->request->getParam('remove') ? "" : $this->request->getParam('coupon');
 
-        if ($coupon === null) {
+        if (empty($coupon)) {
             $coupon = $this->request->getParam('coupon_code') ?
                 $this->request->getParam('coupon_code') :
                 $this->checkoutSession->getCoupon();
         }
 
-        if ($coupon !== null) {
+        if (!empty($coupon)) {
             if ($quote = $this->checkoutSession->getQuote()) {
                 $this->checkoutSession->setCoupon($coupon);
                 $quote->setCouponCode($coupon)
