diff --git a/view/frontend/web/js/popup.js b/view/frontend/web/js/popup.js
index 73e3954..c04f546 100644
--- a/view/frontend/web/js/popup.js
+++ b/view/frontend/web/js/popup.js
@@ -267,7 +267,7 @@ define([
                     var event = new $.Event('reloaded');
                     widget.element.trigger(event, [itemsCount]);
 
-                    if (widget.options.autoOpenPopup && widget.isOpen && container.children().length) {
+                    if (widget.options.autoOpenPopup && container.children().length) {
                         $('a[data-role="ampromo-popup-show"]').click();
                         widget.show();
                         widget.isOpen = false;
