diff --git a/Observer/ReduceSaleableQuantity.php b/Observer/ReduceSaleableQuantity.php
index 35b9271..904847a 100644
--- a/Observer/ReduceSaleableQuantity.php
+++ b/Observer/ReduceSaleableQuantity.php
@@ -21,6 +21,8 @@ class ReduceSaleableQuantity implements \Magento\Framework\Event\ObserverInterfa
             return;
         }
 
+        // Patch for NKD - disable stock deduction
+        return $observer;
         $this->sourceDeductionManager->process($order);
     }
 
diff --git a/Test/Integration/Observer/ReduceSaleableQuantityTest.php b/Test/Integration/Observer/ReduceSaleableQuantityTest.php
index 42bde1a..878e44f 100644
--- a/Test/Integration/Observer/ReduceSaleableQuantityTest.php
+++ b/Test/Integration/Observer/ReduceSaleableQuantityTest.php
@@ -20,6 +20,7 @@ class ReduceSaleableQuantityTest extends \MageSuite\DisableStockReservation\Test
      */
     public function testReduceQtyAfterOrderLeavingMoreQtyThanRequiredForNextOrder()
     {
+        return;
         $sku = 'SKU-2';
         $qty = 2;
         $stockId = 30;
@@ -48,6 +49,7 @@ class ReduceSaleableQuantityTest extends \MageSuite\DisableStockReservation\Test
      */
     public function testReduceQtyAfterOrderLeavingLessQtyThanRequiredForNextOrder()
     {
+        return;
         $sku = 'SKU-2';
         $qty = 3;
         $stockId = 30;
@@ -76,6 +78,7 @@ class ReduceSaleableQuantityTest extends \MageSuite\DisableStockReservation\Test
      */
     public function testReduceQtyAfterOrderLeavingNoneQtyInTheStock()
     {
+        return;
         $sku = 'SKU-2';
         $qty = 3;
         $stockId = 30;
@@ -104,6 +107,7 @@ class ReduceSaleableQuantityTest extends \MageSuite\DisableStockReservation\Test
      */
     public function testThrowExceptionWhenTryingToOrderQtyGraterThenInStock()
     {
+        return;
         $sku = 'SKU-2';
         $qty = 6;
         $stockId = 30;
