diff --git a/ViewModel/FileAttribute.php b/ViewModel/FileAttribute.php
index f46533f..0e9cdde 100644
--- a/ViewModel/FileAttribute.php
+++ b/ViewModel/FileAttribute.php
@@ -135,6 +135,11 @@ class FileAttribute implements ArgumentInterface
 
                 if ($filename) {
                     $filePath = $this->entityType . $filename;
+                    // BEGINNING OF PATCH
+                    if ($attribute->getAttributeCode() == \Creativestyle\CustomizationDogmaOrder\Setup\Patch\Data\AddCustomerPetAttributes::PET_IMAGE_ATTRIBUTE) {
+                        $filePath = \Creativestyle\CustomizationDogmaOrder\Setup\Patch\Data\AddCustomerPetAttributes::PET_IMAGE_ATTRIBUTE . $filename;
+                    }
+                    // END OF PATCH
                     $fileProcessor = $this->fileProcessorFactory
                         ->create(['entityTypeCode' => $this->entityType]);
                     $fileInfo = $this->mediaDirectory->stat($filePath);
