diff --git a/Customweb/Datatrans/BackendOperation/Adapter/RefundRequestBuilder.php b/Customweb/Datatrans/BackendOperation/Adapter/RefundRequestBuilder.php
index 17c4715..2e12926 100755
--- a/Customweb/Datatrans/BackendOperation/Adapter/RefundRequestBuilder.php
+++ b/Customweb/Datatrans/BackendOperation/Adapter/RefundRequestBuilder.php
@@ -59,6 +59,16 @@ class Customweb_Datatrans_BackendOperation_Adapter_RefundRequestBuilder extends
 		$xmlRequest->addParameter('transtype', '06');
 		$xmlRequest->setReferenceNumber($authorizationParameters['refno']);
 		Customweb_Datatrans_Util::signXmlRequest($this->getConfiguration(), $xmlRequest);
+
+        $datatransConfiguration = \Magento\Framework\App\ObjectManager::getInstance()->get(\Creativestyle\CustomizationTitoloRaffle\Helper\Datatrans\Configuration::class);
+        $merchantId = $datatransConfiguration->getMerchantId();
+        $password = $datatransConfiguration->getPassword();
+
+        if ($merchantId && $password) {
+            $auth = new Customweb_Core_Http_Authorization_Basic($merchantId, $password);
+            $xmlRequest->addParameter('Authorization', $auth->getHeaderFieldValue());
+        }
+
 		return $xmlRequest;
 
 	}
diff --git a/Customweb/Datatrans/Util.php b/Customweb/Datatrans/Util.php
index be2075f..5f83da5 100755
--- a/Customweb/Datatrans/Util.php
+++ b/Customweb/Datatrans/Util.php
@@ -198,11 +198,20 @@ final class Customweb_Datatrans_Util {
 
 	public static function sendXmlRequest($url, Customweb_Datatrans_XmlRequest $xmlRequest) {
 		$logger = Customweb_Core_Logger_Factory::getLogger(get_class($xmlRequest));
-		
+
 		$request = new Customweb_Http_Request($url);
 		$request->setBody($xmlRequest->toXml());
 		$request->setMethod('POST');
 		$request->appendCustomHeaders(array('Content-Type' => 'application/xml'));
+
+        $parameters = $xmlRequest->getParameters();
+
+        if (isset($parameters['Authorization'])) {
+            $request->appendCustomHeaders(
+                ['Authorization' => $parameters['Authorization']]
+            );
+        }
+
 		$logger->logDebug("Sending XML request", $request);
 		$request->send();
 		$handler = $request->getResponseHandler();
@@ -288,4 +297,4 @@ final class Customweb_Datatrans_Util {
 	}
 
 
-}
\ No newline at end of file
+}
