diff --git a/Customweb/Datatrans/Util.php b/Customweb/Datatrans/Util.php
index 15bca27..7a3cb15 100755
--- a/Customweb/Datatrans/Util.php
+++ b/Customweb/Datatrans/Util.php
@@ -203,6 +203,18 @@ final class Customweb_Datatrans_Util {
 		$request->setBody($xmlRequest->toXml());
 		$request->setMethod('POST');
 		$request->appendCustomHeaders(array('Content-Type' => 'application/xml'));
+
+        $datatransConfiguration = \Magento\Framework\App\ObjectManager::getInstance()->get(\Creativestyle\CustomizationTitoloRaffle\Helper\Datatrans\Configuration::class);
+        $merchantId = $datatransConfiguration->getMerchantId();
+        $password = $datatransConfiguration->getPassword();
+
+        if ($merchantId && $password) {
+            $auth = new Customweb_Core_Http_Authorization_Basic($merchantId, $password);
+            $request->appendCustomHeaders(
+                ['Authorization' => $auth->getHeaderFieldValue()]
+            );
+        }
+        
 		$logger->logDebug("Sending XML request", $request);
 		$request->send();
 		$handler = $request->getResponseHandler();
