diff --git a/Model/Processor/Order.php b/Model/Processor/Order.php
index b8f940f..6c52cb1 100644
--- a/Model/Processor/Order.php
+++ b/Model/Processor/Order.php
@@ -215,9 +215,9 @@ class Order extends Common {
             2 => $config->getAddressLine3(),
         );
         $lengths = array(
-            \Postnl\Ecs\Model\System\Config\Source\Address::FIELD_STREET => 95,
+            \Postnl\Ecs\Model\System\Config\Source\Address::FIELD_STREET => 30,
             \Postnl\Ecs\Model\System\Config\Source\Address::FIELD_HOUSE => 5,
-            \Postnl\Ecs\Model\System\Config\Source\Address::FIELD_ANNEX => 35,
+            \Postnl\Ecs\Model\System\Config\Source\Address::FIELD_ANNEX => 30,
         );
         foreach ($this->_streetParts() as $part)
         {
