diff --git a/Block/Info/Basic.php b/Block/Info/Basic.php
index e2b95d5..d8d3efc 100644
--- a/Block/Info/Basic.php
+++ b/Block/Info/Basic.php
@@ -26,8 +26,34 @@
 
 namespace Payone\Core\Block\Info;
 
+use Magento\Framework\View\Element\Template\Context;
+use Payone\Core\Model\TransactionStatusRepository;
+
 class Basic extends Base
 {
+    /**
+     * @var \Magento\Store\Model\App\Emulation
+     */
+    protected $emulation;
+
+    /**
+     * Basic constructor.
+     * @param TransactionStatusRepository $transactionStatusRepository
+     * @param Context $context
+     * @param array $data
+     * @param \Magento\Store\Model\App\Emulation $emulation
+     */
+    public function __construct(
+        TransactionStatusRepository $transactionStatusRepository,
+        Context $context,
+        $data = [],
+        \Magento\Store\Model\App\Emulation $emulation
+    )
+    {
+        parent::__construct($transactionStatusRepository, $context, $data);
+        $this->emulation = $emulation;
+    }
+
     /**
      * Prepare credit card related payment info
      *
@@ -39,6 +65,7 @@ class Basic extends Base
         if (null !== $this->_paymentSpecificInformation) {
             return $this->_paymentSpecificInformation;
         }
+
         $transport = parent::_prepareSpecificInformation($transport);
         $data = [];
 
@@ -47,6 +74,7 @@ class Basic extends Base
             $data[(string)__('Payment has not been processed yet.')] = '';
         } else {
             $oOrder = $this->getInfo()->getOrder();
+            $this->emulation->startEnvironmentEmulation($oOrder->getStoreId(), 'frontend');
             if ($oOrder && $oOrder->getPayoneClearingBankcode()) {
                 $data[(string)__('Please transfer the order amount to this bank account')] = '';
                 $data[(string)__('Accountholder')] = $oOrder->getPayoneClearingBankaccountholder();
@@ -57,6 +85,7 @@ class Basic extends Base
                 $data[(string)__('Bank')] = $oOrder->getPayoneClearingBankname();
             }
             $data[(string)__('Payment reference')] = $oOrder->getPayoneTxid();
+            $this->emulation->stopEnvironmentEmulation();
         }
         return $transport->setData(array_merge($data, $transport->getData()));
     }
