diff --git a/Model/PaymentInformationManagement.php b/Model/PaymentInformationManagement.php
index 3d6b0aa..eca747f 100644
--- a/Model/PaymentInformationManagement.php
+++ b/Model/PaymentInformationManagement.php
@@ -114,9 +114,8 @@ class PaymentInformationManagement implements \Magento\Checkout\Api\PaymentInfor
             $quote->setDataChanges(true);
             $shippingAddress = $quote->getShippingAddress();
             if ($shippingAddress && $shippingAddress->getShippingMethod()) {
-                $shippingDataArray = explode('_', $shippingAddress->getShippingMethod());
-                $shippingCarrier = array_shift($shippingDataArray);
-                $shippingAddress->setLimitCarrier($shippingCarrier);
+                $shippingRate = $shippingAddress->getShippingRateByCode($shippingAddress->getShippingMethod());
+                $shippingAddress->setLimitCarrier($shippingRate->getCarrier());
             }
         }
         $this->paymentMethodManagement->set($cartId, $paymentMethod);
