diff --git a/Ui/DataProvider/Product/Form/Modifier/Data/AssociatedProducts.php b/Ui/DataProvider/Product/Form/Modifier/Data/AssociatedProducts.php
index 7815408..ea20a98 100644
--- a/Ui/DataProvider/Product/Form/Modifier/Data/AssociatedProducts.php
+++ b/Ui/DataProvider/Product/Form/Modifier/Data/AssociatedProducts.php
@@ -86,6 +86,10 @@ class AssociatedProducts
      * @var ImageHelper
      */
     protected $imageHelper;
+    /**
+     * @var \Creativestyle\CustomizationLensplazaRxConfigurator\Helper\GlassesAttributeSet
+     */
+    protected $glassesAttributeSet;

     /**
      * @var Escaper
@@ -115,6 +119,7 @@ class AssociatedProducts
         CurrencyInterface $localeCurrency,
         JsonHelper $jsonHelper,
         ImageHelper $imageHelper,
+        \Creativestyle\CustomizationLensplazaRxConfigurator\Helper\GlassesAttributeSet $glassesAttributeSet,
         Escaper $escaper = null
     ) {
         $this->locator = $locator;
@@ -127,6 +132,7 @@ class AssociatedProducts
         $this->jsonHelper = $jsonHelper;
         $this->imageHelper = $imageHelper;
         $this->escaper = $escaper ?: ObjectManager::getInstance()->get(Escaper::class);
+        $this->glassesAttributeSet = $glassesAttributeSet;
     }

     /**
@@ -291,27 +297,29 @@ class AssociatedProducts
                         $attributes[$attribute->getAttributeId()]['chosen'][$optionId] = $variationOption;
                     }

-                    $productMatrix[] = [
-                        'id' => $product->getId(),
-                        'product_link' => '<a href="' . $this->urlBuilder->getUrl(
-                            'catalog/product/edit',
-                            ['id' => $product->getId()]
-                        ) . '" target="_blank">' . $this->escaper->escapeHtml($product->getName()) . '</a>',
-                        'sku' => $product->getSku(),
-                        'name' => $this->escaper->escapeHtml($product->getName()),
-                        'qty' => $this->getProductStockQty($product),
-                        'price' => $price,
-                        'price_string' => $currency->toCurrency(sprintf("%f", $price)),
-                        'price_currency' => $this->locator->getStore()->getBaseCurrency()->getCurrencySymbol(),
-                        'configurable_attribute' => $this->getJsonConfigurableAttributes($variationOptions),
-                        'weight' => $product->getWeight(),
-                        'status' => $product->getStatus(),
-                        'variationKey' => $this->getVariationKey($variationOptions),
-                        'canEdit' => 0,
-                        'newProduct' => 0,
-                        'attributes' => $this->getTextAttributes($variationOptions),
-                        'thumbnail_image' => $this->imageHelper->init($product, 'product_thumbnail_image')->getUrl(),
-                    ];
+                    if ($this->glassesAttributeSet->isGlassesProduct($product->getId())) {
+                        $productMatrix[] = [
+                            'id' => $product->getId(),
+                            'product_link' => '<a href="' . $this->urlBuilder->getUrl(
+                                    'catalog/product/edit',
+                                    ['id' => $product->getId()]
+                                ) . '" target="_blank">' . $this->escaper->escapeHtml($product->getName()) . '</a>',
+                            'sku' => $product->getSku(),
+                            'name' => $this->escaper->escapeHtml($product->getName()),
+                            'qty' => $this->getProductStockQty($product),
+                            'price' => $price,
+                            'price_string' => $currency->toCurrency(sprintf("%f", $price)),
+                            'price_currency' => $this->locator->getStore()->getBaseCurrency()->getCurrencySymbol(),
+                            'configurable_attribute' => $this->getJsonConfigurableAttributes($variationOptions),
+                            'weight' => $product->getWeight(),
+                            'status' => $product->getStatus(),
+                            'variationKey' => $this->getVariationKey($variationOptions),
+                            'canEdit' => 0,
+                            'newProduct' => 0,
+                            'attributes' => $this->getTextAttributes($variationOptions),
+                            'thumbnail_image' => $this->imageHelper->init($product, 'product_thumbnail_image')->getUrl(),
+                        ];
+                    }
                     $productIds[] = $product->getId();
                 }
             }
