diff --git a/Pricing/Price/TierPrice.php b/Pricing/Price/TierPrice.php
index 1aa43a3..f392300 100644
--- a/Pricing/Price/TierPrice.php
+++ b/Pricing/Price/TierPrice.php
@@ -172,7 +172,7 @@ class TierPrice extends AbstractPrice implements TierPriceInterface, BasePricePr
         $qtyCache = [];
         $allCustomersGroupId = $this->groupManagement->getAllCustomersGroup()->getId();
         foreach ($priceList as $priceKey => &$price) {
-            if ($price['price'] >= $this->priceInfo->getPrice(FinalPrice::PRICE_CODE)->getValue()) {
+            if ($price['price'] > $this->priceInfo->getPrice(FinalPrice::PRICE_CODE)->getValue()) {
                 unset($priceList[$priceKey]);
                 continue;
             }
