diff --git a/Block/Product/Renderer/Configurable.php b/Block/Product/Renderer/Configurable.php
index c982938..f0e7adb 100644
--- a/Block/Product/Renderer/Configurable.php
+++ b/Block/Product/Renderer/Configurable.php
@@ -182,6 +182,10 @@ class Configurable extends \Magento\ConfigurableProduct\Block\Product\View\Type\
                     $attributeDataArray
                 );
             }
+
+            if (isset($attributeDataArray['additional_data'])) {
+                $config[$attributeId]['additional_data'] = $attributeDataArray['additional_data'];
+            }
         }

         return $this->jsonEncoder->encode($config);
