diff --git a/Model/Indexer/Product/Price/AbstractAction.php b/Model/Indexer/Product/Price/AbstractAction.php
--- a/Model/Indexer/Product/Price/AbstractAction.php	(date 1621413943168)
+++ b/Model/Indexer/Product/Price/AbstractAction.php	(date 1621413943168)
@@ -389,38 +389,46 @@
         $changedIds = array_unique(array_merge($changedIds, ...array_values($parentProductsTypes)));
         $productsTypes = array_merge_recursive($productsTypes, $parentProductsTypes);

-        if ($changedIds) {
-            $this->deleteIndexData($changedIds);
-        }
+        try {
+            $this->getConnection()->beginTransaction();
+            if ($changedIds) {
+                $this->deleteIndexData($changedIds);
+            }

-        $typeIndexers = $this->getTypeIndexers();
-        foreach ($typeIndexers as $productType => $indexer) {
-            $entityIds = $productsTypes[$productType] ?? [];
-            if (empty($entityIds)) {
-                continue;
-            }
+            $typeIndexers = $this->getTypeIndexers();
+            foreach ($typeIndexers as $productType => $indexer) {
+                $entityIds = $productsTypes[$productType] ?? [];
+                if (empty($entityIds)) {
+                    continue;
+                }

-            if ($indexer instanceof DimensionalIndexerInterface) {
-                foreach ($this->dimensionCollectionFactory->create() as $dimensions) {
-                    $this->tableMaintainer->createMainTmpTable($dimensions);
-                    $temporaryTable = $this->tableMaintainer->getMainTmpTable($dimensions);
-                    $this->_emptyTable($temporaryTable);
-                    $indexer->executeByDimensions($dimensions, \SplFixedArray::fromArray($entityIds, false));
-                    // copy to index
-                    $this->_insertFromTable(
-                        $temporaryTable,
-                        $this->tableMaintainer->getMainTableByDimensions($dimensions)
-                    );
-                }
-            } else {
-                // handle 3d-party indexers for backward compatibility
-                $this->_emptyTable($this->_defaultIndexerResource->getIdxTable());
-                $this->_copyRelationIndexData($entityIds);
-                $indexer->reindexEntity($entityIds);
-                $this->_syncData($entityIds);
-            }
-        }
+                if ($indexer instanceof DimensionalIndexerInterface) {
+                    foreach ($this->dimensionCollectionFactory->create() as $dimensions) {
+                        $this->tableMaintainer->createMainTmpTable($dimensions);
+                        $temporaryTable = $this->tableMaintainer->getMainTmpTable($dimensions);
+                        $this->_emptyTable($temporaryTable);
+                        $indexer->executeByDimensions($dimensions, \SplFixedArray::fromArray($entityIds, false));
+                        // copy to index
+                        $this->_insertFromTable(
+                            $temporaryTable,
+                            $this->tableMaintainer->getMainTableByDimensions($dimensions)
+                        );
+                    }
+                } else {
+                    // handle 3d-party indexers for backward compatibility
+                    $this->_emptyTable($this->_defaultIndexerResource->getIdxTable());
+                    $this->_copyRelationIndexData($entityIds);
+                    $indexer->reindexEntity($entityIds);
+                    $this->_syncData($entityIds);
+                }
+            }

+            $this->getConnection()->commit();
+        } catch (\Exception $e) {
+            $this->getConnection()->rollBack();
+            throw $e;
+        }
+
         return $changedIds;
     }

