diff --git a/Observer/Transactionstatus/Appointed.php b/Observer/Transactionstatus/Appointed.php
index cdcb2b1..aa2d6fc 100644
--- a/Observer/Transactionstatus/Appointed.php
+++ b/Observer/Transactionstatus/Appointed.php
@@ -137,6 +137,8 @@ class Appointed implements ObserverInterface
         $oInvoice->register();
         $oInvoice->save();
 
+        $oOrder->setTaxInvoiced($oOrder->getTaxInvoiced() + $oInvoice->getTaxAmount());
+        $oOrder->setBaseTaxInvoiced($oOrder->getBaseTaxInvoiced() + $oInvoice->getBaseTaxAmount());
         $oOrder->save();
 
         if ($this->baseHelper->getConfigParam('send_invoice_email', 'emails')) {
