diff --git a/Model/Processor/Order.php b/Model/Processor/Order.php
index 15555a0..b8f940f 100644
--- a/Model/Processor/Order.php
+++ b/Model/Processor/Order.php
@@ -354,7 +354,7 @@ class Order extends Common {
         
         $carrier = $order->getShippingCarrier();
         $node->appendChild($xml->createElementNS('http://www.toppak.nl/deliveryorder_new','shippingAgentCode', 
-            $carrier ? $carrier->getConfigData('title') : $order->getShippingMethod()
+            $order->getData('shipping_code_erp')
         ));
 		
         
