diff --git a/Model/Processor/Order.php b/Model/Processor/Order.php
index 26b46e9..6a1d66d 100644
--- a/Model/Processor/Order.php
+++ b/Model/Processor/Order.php
@@ -352,9 +352,9 @@ class Order extends Common {
             ''
         ));
 
-        $carrier = $order->getShippingCarrier();
+
         $node->appendChild($xml->createElementNS('http://www.toppak.nl/deliveryorder_new','shippingAgentCode',
-            $carrier ? $carrier->getConfigData('title') : $order->getShippingMethod()
+            $order->getShippingDescription()
         ));
 
 
