diff --git a/Model/AbstractModel.php b/Model/AbstractModel.php
index ba4aac5..5a3a91d 100644
--- a/Model/AbstractModel.php
+++ b/Model/AbstractModel.php
@@ -40,6 +40,7 @@ use TIG\PersistentShoppingCart\Helper\Data as Helper;

 abstract class AbstractModel extends FrameworkAbstractModel
 {
+    const XPATH_CART_COOKIE_LIFETIME = 'tig_persistentshoppingcart/general/guest_cookie_lifetime';
     /**
      * @var string
      */
@@ -50,6 +51,11 @@ abstract class AbstractModel extends FrameworkAbstractModel
      */
     private $sessionConfig;

+    /**
+     * @var \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
+     */
+    private $scopeConfig;
+
     /**
      * @var \Magento\Framework\Stdlib\CookieManagerInterface $cookieManager
      */
@@ -69,6 +75,7 @@ abstract class AbstractModel extends FrameworkAbstractModel
      * AbstractModel constructor.
      *
      * @param \Magento\Framework\Session\Config\ConfigInterface $sessionConfig
+     * @param \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfigInterface
      * @param \Magento\Framework\Stdlib\CookieManagerInterface $cookieManager
      * @param \Magento\Framework\Stdlib\Cookie\CookieMetadataFactory $cookieMetadata
      * @param \TIG\PersistentShoppingCart\Helper\Data $helper
@@ -79,6 +86,7 @@ abstract class AbstractModel extends FrameworkAbstractModel
      */
     public function __construct(
         ConfigInterface $sessionConfig,
+        \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
         CookieManagerInterface $cookieManager,
         CookieMetadataFactory $cookieMetadata,
         Helper $helper,
@@ -88,6 +96,7 @@ abstract class AbstractModel extends FrameworkAbstractModel
         \Magento\Framework\Data\Collection\AbstractDb $resourceCollection = null
     ) {
         $this->sessionConfig  = $sessionConfig;
+        $this->scopeConfig = $scopeConfig;
         $this->cookieManager  = $cookieManager;
         $this->cookieMetadata = $cookieMetadata;
         $this->helper         = $helper;
@@ -182,7 +191,7 @@ abstract class AbstractModel extends FrameworkAbstractModel
     private function processClientCookie($value)
     {
         $metadata = $this->cookieMetadata->createPublicCookieMetadata();
-        $metadata->setDuration($this->sessionConfig->getCookieLifetime());
+        $metadata->setDuration($this->getCookieDuration());
         $metadata->setPath($this->sessionConfig->getCookiePath());
         $metadata->setDomain($this->sessionConfig->getCookieDomain());
 
@@ -194,4 +203,21 @@ abstract class AbstractModel extends FrameworkAbstractModel
 
         $this->cookieManager->deleteCookie($this->cookieName);
     }
+
+    private function getGuestCookieLifetimeConfig()
+    {
+        return $this->scopeConfig->getValue(
+            self::XPATH_CART_COOKIE_LIFETIME,
+            \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+        );
+    }
+
+    private function getCookieDuration()
+    {
+        if($this->getGuestCookieLifetimeConfig() > 0) {
+            return intval($this->getGuestCookieLifetimeConfig());
+        }
+
+        return $this->sessionConfig->getCookieLifetime();
+    }
 }
diff --git a/Model/QuoteCookie.php b/Model/QuoteCookie.php
index db3ddc9..c8509d9 100644
--- a/Model/QuoteCookie.php
+++ b/Model/QuoteCookie.php
@@ -86,6 +86,7 @@ class QuoteCookie extends AbstractToken

         parent::__construct(
             $sessionConfig,
+            $scopeConfig,
             $cookieManager,
             $cookieMetadata,
             $helper,
diff --git a/etc/adminhtml/system.xml b/etc/adminhtml/system.xml
index c83d6fb..d908f0f 100644
--- a/etc/adminhtml/system.xml
+++ b/etc/adminhtml/system.xml
@@ -44,6 +44,10 @@
                 <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
                 <comment><![CDATA[Make sure Customers > Persistent Shopping Cart > Enable Persistence is set to "Yes" and Persist Shopping Cart is set to "Yes".]]></comment>
             </field>
+            <field id="guest_cookie_lifetime" translate="label" type="text" showInDefault="1" showInWebsite="1" showInStore="1">
+                <label>Guest Cookie Cart Lifetime</label>
+                <validate>validate-number validate-zero-or-greater</validate>
+            </field>
         </group>
     </section>
 </system>
