diff --git a/src/module-elasticsuite-tracker/Block/Variables/Page/Base.php b/src/module-elasticsuite-tracker/Block/Variables/Page/Base.php
index 9113564..eb5c774 100644
--- a/src/module-elasticsuite-tracker/Block/Variables/Page/Base.php
+++ b/src/module-elasticsuite-tracker/Block/Variables/Page/Base.php
@@ -37,16 +37,22 @@ class Base extends \Smile\ElasticsuiteTracker\Block\Variables\Page\AbstractBlock
      */
     protected $localeResolver;
 
+    /**
+     * @var \Magento\Framework\App\RequestInterface
+     */
+    protected $requestInterface;
+
     /**
      * Set the default template for page variable blocks
      *
-     * @param Template\Context                               $context        The template context
-     * @param \Magento\Framework\Json\Helper\Data            $jsonHelper     The Magento's JSON Helper
-     * @param \Smile\ElasticsuiteTracker\Helper\Data         $trackerHelper  The Smile Tracker helper
-     * @param \Magento\Framework\Registry                    $registry       Magento Core Registry
-     * @param \Magento\Framework\View\Layout\PageType\Config $pageTypeConfig The page type configuration
-     * @param \Magento\Framework\Locale\ResolverInterface    $localeResolver Locale Resolver
-     * @param array                                          $data           The block data
+     * @param Template\Context                               $context          The template context
+     * @param \Magento\Framework\Json\Helper\Data            $jsonHelper       The Magento's JSON Helper
+     * @param \Smile\ElasticsuiteTracker\Helper\Data         $trackerHelper    The Smile Tracker helper
+     * @param \Magento\Framework\Registry                    $registry         Magento Core Registry
+     * @param \Magento\Framework\View\Layout\PageType\Config $pageTypeConfig   The page type configuration
+     * @param \Magento\Framework\Locale\ResolverInterface    $localeResolver   Locale Resolver
+     * @param \Magento\Framework\App\RequestInterface        $requestInterface RequestInterface
+     * @param array                                          $data             The block data
      */
     public function __construct(
         Template\Context $context,
@@ -55,10 +61,14 @@ class Base extends \Smile\ElasticsuiteTracker\Block\Variables\Page\AbstractBlock
         \Magento\Framework\Registry $registry,
         \Magento\Framework\View\Layout\PageType\Config $pageTypeConfig,
         \Magento\Framework\Locale\ResolverInterface $localeResolver,
+        // BEGIN PATCH
+        \Magento\Framework\App\RequestInterface $requestInterface,
+        // END PATCH
         array $data = []
     ) {
         $this->pageTypeConfig = $pageTypeConfig;
         $this->localeResolver = $localeResolver;
+        $this->requestInterface = $requestInterface;
 
         return parent::__construct($context, $jsonHelper, $trackerHelper, $registry, $data);
     }
@@ -134,7 +144,9 @@ class Base extends \Smile\ElasticsuiteTracker\Block\Variables\Page\AbstractBlock
     {
         return [
             'locale' => $this->localeResolver->getLocale(),
-            'domain' => $_SERVER['SERVER_NAME'],
+            // BEGIN PATCH
+            'domain' => $this->requestInterface->getServer('SERVER_ADDR'),
+            // END PATCH
         ];
     }
 
