diff --git a/src/module-elasticsuite-virtual-category/Model/Preview.php b/src/module-elasticsuite-virtual-category/Model/Preview.php
index 32efa02..c21fb59 100644
--- a/src/module-elasticsuite-virtual-category/Model/Preview.php
+++ b/src/module-elasticsuite-virtual-category/Model/Preview.php
@@ -133,7 +133,10 @@ class Preview extends AbstractPreview
      */
     protected function getSortedProductIds() : array
     {
-        return ($this->getSortBy() === 'position') ? $this->category->getSortedProductIds() : [];
+        // PATCH START
+        // return ($this->getSortBy() === 'position') ? $this->category->getSortedProductIds() : [];
+        return str_contains($this->getSortBy(), 'position') ? $this->category->getSortedProductIds() : [];
+        // PATCH END
     }

     /**
@@ -143,7 +146,10 @@ class Preview extends AbstractPreview
     {
         $items = parent::preparePreviewItems($products);

-        if ($this->getSortBy() !== 'position') {
+        // PATCH START
+        // if ($this->getSortBy() !== 'position') {
+        if (!str_contains($this->getSortBy(), 'position')) {
+        // PATCH END
             // In order to sort the product in admin category grid, we need to set the position value
             // if the sort order is different from position because the products are sorted in js.
             // We also disable manual sorting when sort order is not position.
@@ -218,7 +224,10 @@ class Preview extends AbstractPreview
             $useConfig = $this->request->getParam('use_config', []);
             $useConfig = array_key_exists('default_sort_by', $useConfig) && $useConfig['default_sort_by'] == 'true';
             $defaultSortBy = $this->categoryConfig->getProductListDefaultSortBy();
-            $this->sortBy = $useConfig ? $defaultSortBy : $this->request->getParam('default_sort_by');
+            // PATCH START
+            // $this->sortBy = $useConfig ? $defaultSortBy : $this->request->getParam('default_sort_by');
+            $this->sortBy = $useConfig || empty($this->request->getParam('default_sort_by')) ? $defaultSortBy : $this->request->getParam('default_sort_by');
+            // PATCH END
         }

         return $this->sortBy;
