diff --git a/src/module-elasticsuite-catalog/Block/Navigation/Renderer/Slider.php b/src/module-elasticsuite-catalog/Block/Navigation/Renderer/Slider.php
index ab4c66a..4b5a983 100644
--- a/src/module-elasticsuite-catalog/Block/Navigation/Renderer/Slider.php
+++ b/src/module-elasticsuite-catalog/Block/Navigation/Renderer/Slider.php
@@ -214,7 +214,7 @@ class Slider extends AbstractRenderer
         $filter = $this->getFilter();
         $item   = current($this->getFilter()->getItems());
 
-        $regexp      = "/({$filter->getRequestVar()})=(-?[0-9]+)/";
+        $regexp      = "/({$filter->getRequestVar()})=(-?[0-9]+-?[0-9]*)/";
         $replacement = '${1}=<%- from %>-<%- to %>';
 
         return preg_replace($regexp, $replacement, $item->getUrl());
