diff --git a/Controller/Adminhtml/Shops/Login.php b/Controller/Adminhtml/Shops/Login.php
index 66b7024..899373b 100644
--- a/Controller/Adminhtml/Shops/Login.php
+++ b/Controller/Adminhtml/Shops/Login.php
@@ -123,7 +123,7 @@ class Login extends Action
         }

         $content = $response->getBody();
-        $httpCode = $response->getStatusCode();
+        $httpCode = $response->getStatus();

         if ($httpCode === 500 || $httpCode === 0) {
             return $result->setData([
diff --git a/TrustedshopsApi/ApiClient.php b/TrustedshopsApi/ApiClient.php
index 4339fac..797408b 100644
--- a/TrustedshopsApi/ApiClient.php
+++ b/TrustedshopsApi/ApiClient.php
@@ -6,8 +6,8 @@ use Exception;
 use Magento\Framework\Exception\AuthorizationException;
 use Trustedshops\Trustedshops\Helper\Data;
 use Trustedshops\Trustedshops\TrustedshopsApi\Request\RequestInterface;
-use Zend\Http\Client;
-use Zend\Http\Response;
+use \Zend_Http_Client as Client;
+use \Zend_Http_Response as Response;

 class ApiClient
 {
@@ -44,7 +44,7 @@ class ApiClient
     {
         $response = null;
         try {
-            $this->client->reset();
+            $this->client->resetParameters(true);
             $this->client->setHeaders(['Accept' => 'application/json']);
             $this->client->setEncType(Client::ENC_FORMDATA);
             $this->client->setAuth($this->getUser(), $this->getPassword());
@@ -56,9 +56,9 @@ class ApiClient
                 $this->client->setRawBody($request->getRequestData()->getRequestBody());
             }

-            $response = $this->client->send();
+            $response = $this->client->request();

-            if ((int)$response->getStatusCode() === 401) {
+            if ((int)$response->getStatus() === 401) {
                 throw new AuthorizationException(
                     __('The MyTS credentials used for authentication with the TrustedShops API are incorrect. Ensure you are logged into MyTS within the Magento 2 Backend using the correct credentials of your TrustedShops account. If your MyTS password was recently changed, try logging out and back in.')
                 );
