diff --git a/Model/Feeds.php b/Model/Feeds.php
index c41cbf1..3d504dd 100644
--- a/Model/Feeds.php
+++ b/Model/Feeds.php
@@ -1191,7 +1191,7 @@
             $this->categoriesFilterList[] = '*';
         }
         // 2. all categories available
-        $listOfCategories = $this->categoryFactory->create()->getCollection()->setStoreId($this->params['store_id'])->addAttributeToSelect(['name', 'store_id', 'is_active', 'include_in_menu']);
+        $listOfCategories = $this->categoryFactory->create()->getCollection()->setStoreId($this->params['store_id'])->addAttributeToSelect(['name', 'store_id', 'is_active', 'include_in_menu', 'include_in_mobile_navigation']);
         $this->categories = [];
         foreach ($listOfCategories as $category) {
             $this->categories[$category->getId()]['name'] = $category->getName();
@@ -1203,7 +1203,7 @@
             if ($this->_includeInMenu) {
                 $this->categories[$category->getId()]['include_in_menu'] = true;
             } else {
-                $this->categories[$category->getId()]['include_in_menu'] = $category->getIncludeInMenu();
+                $this->categories[$category->getId()]['include_in_menu'] = $category->getIncludeInMenu() || $category->getIncludeInMobileNavigation();
             }
         }
         foreach ($this->categoriesFilterList as $i => $id) {

