diff --git a/Plugin/Framework/App/Http/Context.php b/Plugin/Framework/App/Http/Context.php
index c2ed669..b7130be 100644
--- a/Plugin/Framework/App/Http/Context.php
+++ b/Plugin/Framework/App/Http/Context.php
@@ -61,9 +61,12 @@ class Context
     {
     	if ($this->helper->isActive())
     	{
-    		$subject->setValue(static::KEY, $this->helper->getCurrentTaxDisplayCacheState(), 0);
+    		// PATCH START
+		// $subject->setValue(static::KEY, $this->helper->getCurrentTaxDisplayCacheState(), 0);
+		$subject->setValue(static::KEY, $this->helper->getCurrentTaxDisplayCacheState(), $this->helper->getDefaultTaxDisplay());
+		// PATCH END
     	}
 		
     	return [];
     }
-}
\ No newline at end of file
+}
