diff --git a/Model/Options.php b/Model/Options.php
index c230353..ee75f51 100644
--- a/Model/Options.php
+++ b/Model/Options.php
@@ -100,7 +100,10 @@ class Options
         $result = [];
         $options = array_filter(explode(';', $options));
         foreach ($options as $value) {
-            $value = $this->escaper->escapeHtml(trim($value));
+            // BEGINNING OF PATCH
+//            $value = $this->escaper->escapeHtml(trim($value));
+            $value = $this->escaper->escapeHtml(trim(__($value)));
+            // END OF PATCH
             $result[$value] = $value;
         }
         if ($isOptional && trim(current($options))) {
