diff --git a/src/ReportApi/Service/StoreResolver.php b/src/ReportApi/Service/StoreResolver.php
index fbdae7c..c6682bd 100644
--- a/src/ReportApi/Service/StoreResolver.php
+++ b/src/ReportApi/Service/StoreResolver.php
@@ -33,7 +33,9 @@ class StoreResolver
     {
         foreach ($request->getFilters() as $filter) {
             if ($filter->getColumn() === 'sales_order|store_id') {
-                self::$storeId = count($filter->getValue()) > 1 ? 0 : $filter->getValue();
+                //BEGIN PATCH
+                self::$storeId = (is_array($filter->getValue()) && count($filter->getValue()) > 1) ? 0 : $filter->getValue();
+                //END PATCH
             }
         }
     }
