diff --git a/Model/Notificator.php b/Model/Notificator.php
index 3e36cd1..137c209 100644
--- a/Model/Notificator.php
+++ b/Model/Notificator.php
@@ -103,15 +103,27 @@ class Notificator implements NotificatorInterface
      */
     public function sendForgotPassword(UserInterface $user): void
     {
+        // BEGIN PATCH
+        /**
+         * Patch admin password url incorrect with setting "web/url/use_store" = 1
+         * Force option to disable store in url from url.
+         * Issue : https://github.com/magento/magento2/issues/35667
+         */
+        $store = $this->storeManager->getStore(Store::DEFAULT_STORE_ID);
+        $store->setDisableStoreInUrl(true);
+        // END PATCH
         try {
             $this->sendNotification(
                 'admin/emails/forgot_email_template',
                 [
                     'username' => $user->getFirstName().' '.$user->getLastName(),
                     'user' => $user,
-                    'store' => $this->storeManager->getStore(
-                        Store::DEFAULT_STORE_ID
-                    )
+                    // BEGIN PATCH
+                    //'store' => $this->storeManager->getStore(
+                    //    Store::DEFAULT_STORE_ID
+                    //)
+                    'store' => $store
+                    // END PATCH
                 ],
                 $user->getEmail(),
                 $user->getFirstName().' '.$user->getLastName()
