diff --git a/Model/ResourceModel/CustomerRepository.php b/Model/ResourceModel/CustomerRepository.php
index 7554219..0e4cfdb 100644
--- a/Model/ResourceModel/CustomerRepository.php
+++ b/Model/ResourceModel/CustomerRepository.php
@@ -232,28 +232,6 @@ class CustomerRepository implements CustomerRepositoryInterface
         //Model's actual ID field maybe different than "id" so "id" field from $customerData may be ignored.
         $customerModel->setId($customer->getId());
         $storeId = $customerModel->getStoreId();
-
-        if ($customer->getDefaultBilling()) {
-            $defaultBillingAddressFlag = $this->validateDefaultAddress(
-                $customer,
-                $prevCustomerData,
-                CustomerInterface::DEFAULT_BILLING
-            );
-            if (!$defaultBillingAddressFlag) {
-                $customerModel->setDefaultBilling(null);
-            }
-        }
-        if ($customer->getDefaultShipping()) {
-            $defaultShippingAddressFlag = $this->validateDefaultAddress(
-                $customer,
-                $prevCustomerData,
-                CustomerInterface::DEFAULT_SHIPPING
-            );
-            if (!$defaultShippingAddressFlag) {
-                $customerModel->setDefaultShipping(null);
-            }
-        }
-
         if ($storeId === null) {
             $customerModel->setStoreId(
                 $prevCustomerData ? $prevCustomerData->getStoreId() : $this->storeManager->getStore()->getId()
@@ -594,60 +572,7 @@ class CustomerRepository implements CustomerRepositoryInterface
         }
         return $customerData;
     }
-
-    /**
-     * To validate default address
-     *
-     * @param CustomerInterface $customer
-     * @param CustomerInterface|null $prevCustomerData
-     * @param string $defaultAddressType
-     * @return bool
-     * @throws InputException
-     */
-    private function validateDefaultAddress(
-        CustomerInterface $customer,
-        ?CustomerInterface $prevCustomerData,
-        string $defaultAddressType
-    ): bool {
-        $defaultAddressId = (int)(
-        $defaultAddressType === CustomerInterface::DEFAULT_BILLING
-            ? $customer->getDefaultBilling()
-            : $customer->getDefaultShipping()
-        );
     
-        if (!$defaultAddressId) {
-            return true;
-        }
-    
-        $customerId = (int)($customer->getId() ?? 0);
-        if (!$customerId && $prevCustomerData) {
-            $customerId = (int)($prevCustomerData->getId() ?? 0);
-        }
-        if ($prevCustomerData && $prevCustomerData->getAddresses()) {
-            foreach ($prevCustomerData->getAddresses() as $address) {
-                if ($defaultAddressId === (int)$address->getId()) {
-                    if ($customerId && (int)$address->getCustomerId() !== $customerId) {
-                        $this->throwInvalidAddressException($defaultAddressType);
-                    }
-                    return true;
-                }
-            }
-            $this->throwInvalidAddressException($defaultAddressType);
-        }
-
-        if ($defaultAddressId) {
-            try {
-                $customerAddress = $this->addressRepository->getById($defaultAddressId);
-                if ((int)$customerAddress->getCustomerId() !== $customerId) {
-                    $this->throwInvalidAddressException($defaultAddressType);
-                }
-            } catch (NoSuchEntityException $e) {
-                return false;
-            }
-        }
-        return true;
-    }
-
     /**
      * Throw invalid address exception
      *
