commit cb39c47f0edb89afa99f695a91f8497f55eedd4c
Author: Adam Karnowka <adam.karnowka@creativestyle.pl>
Date:   Mon Sep 10 14:13:52 2018 +0200

    Fix for ordering top navigation links with sortOrder directive

--- a/Navigation.php
+++ b/Navigation.php
@@ -46,6 +46,9 @@ class Navigation extends Links
      */
     private function compare(SortLinkInterface $firstLink, SortLinkInterface $secondLink)
     {
-        return ($firstLink->getSortOrder() < $secondLink->getSortOrder());
+        if ($firstLink->getSortOrder() == $secondLink->getSortOrder()) {
+            return 0;
+        }
+        return ($firstLink->getSortOrder() < $secondLink->getSortOrder()) ? -1 : 1;
     }
 }
