diff --git a/view/frontend/web/js/payment-service-mixin.js b/view/frontend/web/js/payment-service-mixin.js
index b023789..81222b4 100644
--- a/view/frontend/web/js/payment-service-mixin.js
+++ b/view/frontend/web/js/payment-service-mixin.js
@@ -22,10 +22,24 @@ define([
 
         function updateMethodList(methods) {
             if (methods.length && cachedMassages !== JSON.stringify(registry.get('paymentData'))) {
-                methodList([]);
-                methodList.push(methods);
+                // BEGIN PATCH
+                const currentMethods = methodList();
+
+                methods.forEach((newMethod, index) => {
+                    const existingMethodIndex = currentMethods.findIndex(
+                        (existingMethod) => {
+                            return existingMethod.method === newMethod.method;
+                        }
+                    );
+
+                    if (existingMethodIndex !== -1) {
+                        methods[index] = currentMethods[existingMethodIndex];
+                    }
+                });
+
                 methodList(methods);
                 cachedMassages = JSON.stringify(registry.get('paymentData'));
+                // END PATCH
             }
         }
 
