diff --git a/Service/Order/Quote/AddressHandler.php b/Service/Order/Quote/AddressHandler.php
index 34751b5..a1441e9 100644
--- a/Service/Order/Quote/AddressHandler.php
+++ b/Service/Order/Quote/AddressHandler.php
@@ -128,7 +128,9 @@ class AddressHandler
      *
      * @return string
      */
-    public function getStreet(array $address, int $storeId): string
+    // BEGIN PATCH
+    // public function getStreet(array $address, int $storeId): string
+    public function getStreet(array $address, int $storeId): array
     {
         $numberOfStreetLines = $this->configProvider->getCustomerStreetLines($storeId);
 
@@ -151,8 +153,10 @@ class AddressHandler
             $street = [$street[0], trim($street[1])];
         }
 
-        return trim(implode("\n", $street));
+        // return trim(implode("\n", $street));
+        return $street;
     }
+    // END PATCH
 
     /**
      * @param $dgOrder
