diff --git a/Model/InstantPurchaseOptionLoadingFactory.php b/Model/InstantPurchaseOptionLoadingFactory.php
index e770b9df6116e..9937b9f4a2c3c 100644
--- a/Model/InstantPurchaseOptionLoadingFactory.php
+++ b/Model/InstantPurchaseOptionLoadingFactory.php
@@ -10,2 +10,3 @@
 use Magento\Customer\Model\AddressFactory;
+use Magento\Framework\Exception\NoSuchEntityException;
 use Magento\Quote\Api\Data\ShippingMethodInterfaceFactory;
@@ -85,6 +86,11 @@ public function create(
         $paymentToken = $this->paymentTokenManagement->getByPublicHash($paymentTokenPublicHash, $customerId);
         $shippingAddress = $this->getAddress($shippingAddressId);
         $billingAddress = $this->getAddress($billingAddressId);
+        if ((int)$shippingAddress->getCustomerId() !== $customerId ||
+            (int)$billingAddress->getCustomerId() !== $customerId) {
+            throw new NoSuchEntityException(__('Address not found.'));
+        }
+
         $shippingMethod = $this->shippingMethodFactory->create()
             ->setCarrierCode($carrierCode)
             ->setMethodCode($shippingMethodCode);
