diff --git a/Block/Adminhtml/Category/Tree.php b/Block/Adminhtml/Category/Tree.php
index 824703a..78be2e6 100644
--- a/Block/Adminhtml/Category/Tree.php
+++ b/Block/Adminhtml/Category/Tree.php
@@ -333,7 +333,7 @@ class Tree extends AbstractCategory
      */
     public function getTreeJson($parenNodeCategory = null)
     {
-        $rootArray = $this->_getNodeJson($this->getRoot($parenNodeCategory));
+        $rootArray = $this->_getNodeJson($this->getRoot($parenNodeCategory, 12));
         $json = $this->_jsonEncoder->encode($rootArray['children'] ?? []);
         return $json;
     }
