diff --git a/Plugin/Session/ModifySameSite.php b/Plugin/Session/ModifySameSite.php
index 747f9cb..cb00819 100644
--- a/Plugin/Session/ModifySameSite.php
+++ b/Plugin/Session/ModifySameSite.php
@@ -51,6 +51,10 @@ class ModifySameSite
      */
     public function beforeSetCookieSameSite(ConfigInterface $subject, string $cookieSameSite = 'Lax'): array
     {
+        if (!$subject->getCookieSecure()) {
+            return [$cookieSameSite];
+        }
+
         $agent = $this->header->getHttpUserAgent();
         $sameSite = $this->validator->shouldSendSameSiteNone($agent);
         if ($sameSite === false) {
