diff --git a/Block/Adminhtml/Order/Totals.php b/Block/Adminhtml/Order/Totals.php
index fa841d2..edf2bc9 100755
--- a/Block/Adminhtml/Order/Totals.php
+++ b/Block/Adminhtml/Order/Totals.php
@@ -27,6 +27,11 @@ class Totals extends MagentoTotals
         $parent = $this->getParentBlock();
         $order = $parent->getOrder();
 
+        $quoteId = $parent->getOrder()->getQuoteId();
+        if (!$quoteId) {
+            return $this;
+        }
+
         if ($order->getTransactionFee() != 0) {
             $parent->addTotal(
                 new \Magento\Framework\DataObject(
