diff --git a/Model/Total/Fee.php b/Model/Total/Fee.php
index 17b47ce..a1473a3 100644
--- a/Model/Total/Fee.php
+++ b/Model/Total/Fee.php
@@ -80,6 +80,10 @@ class Fee extends Address\Total\AbstractTotal
         $baseTax = 0;
         $tax = 0;

+        // BEGIN PATCH
+        $rate = 0;
+        // END PATCH
+
         if ($this->helper->isEnable()) {
             $baseFee = $this->calculator->calculate($quote);
             $fee = $this->helper->getStoreFee($baseFee, $quote);
@@ -93,15 +97,27 @@ class Fee extends Address\Total\AbstractTotal
                     $baseTax = $this->taxCalculator->calcTaxAmount(
                         $baseFee,
                         $rate,
-                        false,
+                        // BEGIN PATCH
+                        // false,
+                        true,
+                        // END PATCH
                         true
                     );
                     $tax = $this->taxCalculator->calcTaxAmount(
                         $fee,
                         $rate,
-                        false,
+                        // BEGIN PATCH
+                        // false,
+                        true,
+                        // END PATCH
                         true
                     );
+
+                    // BEGIN PATCH
+                    // subtract tax value from paymentFee
+                    $baseFee -= $baseTax;
+                    $fee -= $tax;
+                    // END PATCH
                 }
             }
         }
@@ -116,6 +132,11 @@ class Fee extends Address\Total\AbstractTotal
         $total->addBaseTotalAmount('tax', $baseTax);
         $total->addTotalAmount('tax', $tax);

+        // BEGIN PATCH
+        $total->addBaseTotalAmount('grand_total', $tax);
+        $total->addTotalAmount('grand_total', $tax);
+        // END PATCH
+
         $quote->setPaymentFee($fee);
         $quote->setBasePaymentFee($baseFee);
         $quote->setPaymentFeeTax($tax);
