diff --git a/Model/Total/Fee.php b/Model/Total/Fee.php
index 17b47ce..7ca73df 100644
--- a/Model/Total/Fee.php
+++ b/Model/Total/Fee.php
@@ -102,6 +102,12 @@ class Fee extends Address\Total\AbstractTotal
                         false,
                         true
                     );
+
+                    // BEGIN PATCH
+                    // subtract tax value from paymentFee
+                    $baseFee -= $baseTax;
+                    $fee -= $tax;
+                    // END PATCH
                 }
             }
         }
@@ -121,6 +127,22 @@ class Fee extends Address\Total\AbstractTotal
         $quote->setPaymentFeeTax($tax);
         $quote->setBasePaymentFeeTax($baseTax);
 
+        // BEGIN PATCH
+        // add paymentFee tax to correct tax group
+        $appliedTaxes = $total->getAppliedTaxes();
+
+        foreach ($appliedTaxes as &$appliedTax) {
+            if ($appliedTax['percent'] != $rate) {
+                continue;
+            }
+
+            $appliedTax['base_amount'] += $baseTax;
+            $appliedTax['amount'] += $tax;
+        }
+
+        $total->setAppliedTaxes($appliedTaxes);
+        // END PATCH
+
         return $this;
     }
 
