diff --git a/Model/Country.php b/Model/Country.php
--- a/Model/Country.php
+++ b/Model/Country.php	(date 1779963449731)
@@ -172,14 +172,14 @@
      */
     public function getFormats()
     {
-        if (!isset(self::$_format[$this->getId()]) && $this->getId()) {
-            self::$_format[$this->getId()] = $this->_formatFactory->create()->getCollection()->setCountryFilter(
+        if (!isset(self::$_format[(string) $this->getId()]) && $this->getId()) {
+            self::$_format[(string) $this->getId()] = $this->_formatFactory->create()->getCollection()->setCountryFilter(
                 $this
             )->load();
         }

-        if (isset(self::$_format[$this->getId()])) {
-            return self::$_format[$this->getId()];
+        if (isset(self::$_format[(string) $this->getId()])) {
+            return self::$_format[(string) $this->getId()];
         }

         return null;
