diff --git a/Model/ResourceModel/Country/Collection.php b/Model/ResourceModel/Country/Collection.php
index 4d76215..5979f21 100644
--- a/Model/ResourceModel/Country/Collection.php
+++ b/Model/ResourceModel/Country/Collection.php
@@ -297,7 +297,9 @@ class Collection extends \Magento\Framework\Model\ResourceModel\Db\Collection\Ab
         foreach ($this->storeManager->getWebsites() as $website) {
             $defaultCountryConfig = $this->_scopeConfig->getValue(
                 \Magento\Directory\Helper\Data::XML_PATH_DEFAULT_COUNTRY,
-                ScopeInterface::SCOPE_WEBSITES,
+                //BEGIN PATCH
+                ScopeInterface::SCOPE_STORE,
+                //END PATCH
                 $website
             );
             $defaultCountry[$defaultCountryConfig][] = $website->getId();
