diff --git a/Helper/Data.php b/Helper/Data.php
index 2f52d12..b2baf12 100644
--- a/Helper/Data.php
+++ b/Helper/Data.php
@@ -40,6 +40,19 @@ class Data extends AbstractHelper
      */
     protected $_dataLayerName = 'dataLayer';
 
+    // BEGIN PATCH
+    protected \MageSuite\PerformanceProduct\Model\Container\ProductCategoryCollectionData $categoryContainer;
+
+    public function __construct(
+        \Magento\Framework\App\Helper\Context $context,
+        \MageSuite\PerformanceProduct\Model\Container\ProductCategoryCollectionData $categoryContainer
+    ) {
+        parent::__construct($context);
+
+        $this->categoryContainer = $categoryContainer;
+    }
+    // END PATCH
+
     /**
      * Whether Tag Manager is ready to use
      *
@@ -231,8 +244,10 @@ class Data extends AbstractHelper
 
         $categoryList = [];
         $index = 1;
-        $categoryCollection = $product->getCategoryCollection();
-        $categories = $categoryCollection->addAttributeToSelect('name');
+        // BEGIN PATCH
+        $categoryCollection = $this->categoryContainer->getProductCategoriesCollection((int)$product->getId()) ?? $product->getCategoryCollection()->addAttributeToSelect('name');
+        $categories = $categoryCollection->getItems();
+        // END PATCH
 
         if (array_key_exists('item_category', $viewItem)) {
             $index = 2;
