diff --git a/Model/Directories/GetDirectoryTree.php b/Model/Directories/GetDirectoryTree.php
index bff9d98..94e5acd 100644
--- a/Model/Directories/GetDirectoryTree.php
+++ b/Model/Directories/GetDirectoryTree.php
@@ -102,13 +102,28 @@ class GetDirectoryTree
                 );
                 if ($imageDirectory->isDirectory()) {
                     $directories[] = $this->getDirectoryData($imageFolderPath);
-                    foreach ($imageDirectory->readRecursively() as $path) {
+                    // PATCH BEGIN
+                    $files = $imageDirectory->readRecursively();
+
+                    $files = array_filter($files, function($path) {
+                        return empty(pathinfo($path, PATHINFO_EXTENSION));
+                    });
+
+                    foreach ($files as $path) {
                         if ($imageDirectory->isDirectory($path)) {
                             $directories[] = $this->getDirectoryData(
                                 $mediaDirectory->getRelativePath($imageDirectory->getAbsolutePath($path))
                             );
                         }
                     }
+                    // foreach ($imageDirectory->readRecursively() as $path) {
+                    //     if ($imageDirectory->isDirectory($path)) {
+                    //         $directories[] = $this->getDirectoryData(
+                    //             $mediaDirectory->getRelativePath($imageDirectory->getAbsolutePath($path))
+                    //         );
+                    //     }
+                    // }
+                    // PATCH END
                 }
             }
         }
