diff --git a/Block/Question/QList.php b/Block/Question/QList.php
index d245d0a..24899ca 100644
--- a/Block/Question/QList.php
+++ b/Block/Question/QList.php
@@ -498,7 +498,7 @@ class QList extends \Magento\Framework\View\Element\Template implements \Magento
      */
     public function formatDate(
         $date = null,
-        $format = IntlDateFormatter::SHORT,
+        $format = \IntlDateFormatter::SHORT,
         $showTime = false,
         $timezone = null
     ) {
