diff --git a/Controller/Index/Index.php b/Controller/Index/Index.php
index f805863..153f1fb 100644
--- a/Controller/Index/Index.php
+++ b/Controller/Index/Index.php
@@ -40,7 +40,7 @@ class Index extends \Magento\Framework\App\Action\Action
     public function dubleoptinProcess($userEmail)
     {
         $nlStatus = $this->_model->checkNlStatus($userEmail);
-        if (!empty($userEmail) && $nlStatus = 1) {
+        if (!empty($userEmail) && $nlStatus == 1) {
             $optinListId = $this->_model->getDbData('optin_list_id');
             $listId = $this->_model->getDbData('selected_list_data');

@@ -66,11 +66,11 @@ class Index extends \Magento\Framework\App\Action\Action
         $optinUrlCheck = $this->_model->getDbData('optin_url_check');
         if ($optinUrlCheck === 'yes' && !empty($doubleoptinRedirect)) {
             header("Location: ".$doubleoptinRedirect);
-            ob_flush_end();
+            ob_end_flush();
         } else {
             $shopName = $this->_model->_getValueDefault->getValue('web/unsecure/base_url', \Magento\Store\Model\ScopeInterface::SCOPE_STORE);
             header("Location: ".$shopName);
-            ob_flush_end();
+            ob_end_flush();
         }
     }
     
