diff --git a/Model/ResourceModel/Indexer/Stock/DefaultStock.php b/Model/ResourceModel/Indexer/Stock/DefaultStock.php
index 81942ff..c151e58 100644
--- a/Model/ResourceModel/Indexer/Stock/DefaultStock.php
+++ b/Model/ResourceModel/Indexer/Stock/DefaultStock.php
@@ -292,6 +292,7 @@ class DefaultStock extends AbstractIndexer implements StockInterface
      */
     protected function _updateIndex($entityIds)
     {
+        $this->deleteOldRecords($entityIds);
         $connection = $this->getConnection();
         $select = $this->_getStockStatusSelect($entityIds, true);
         $select = $this->getQueryProcessorComposite()->processQuery($select, $entityIds, true);
@@ -314,7 +315,6 @@ class DefaultStock extends AbstractIndexer implements StockInterface
             }
         }
 
-        $this->deleteOldRecords($entityIds);
         $this->_updateIndexTable($data);
 
         return $this;
