diff --git a/Model/Import/Product/Type/AbstractType.php b/Model/Import/Product/Type/AbstractType.php
index 17d0840..e7a5fdf 100644
--- a/Model/Import/Product/Type/AbstractType.php
+++ b/Model/Import/Product/Type/AbstractType.php
@@ -534,7 +534,7 @@ abstract class AbstractType
     public function clearEmptyData(array $rowData)
     {
         foreach ($this->_getProductAttributes($rowData) as $attrCode => $attrParams) {
-            if (!$attrParams['is_static'] && !isset($rowData[$attrCode])) {
+            if (!$attrParams['is_static'] && (!isset($rowData[$attrCode]) || $rowData[$attrCode] === '')) {
                 unset($rowData[$attrCode]);
             }
         }