diff --git a/Model/Order/Address.php b/Model/Order/Address.php
index 0fd4555..5f7c89d 100644
--- a/Model/Order/Address.php
+++ b/Model/Order/Address.php
@@ -142,7 +142,10 @@ class Address extends AbstractModel implements OrderAddressInterface, AddressMod
     {
         $name = '';
         if ($this->getPrefix()) {
-            $name .= $this->getPrefix() . ' ';
+            // BEGINNING OF PATCH
+//            $name .= $this->getPrefix() . ' ';
+            $name .= __($this->getPrefix()) . ' ';
+            // END OF PATCH
         }
         $name .= $this->getFirstname();
         if ($this->getMiddlename()) {
@@ -150,7 +153,10 @@ class Address extends AbstractModel implements OrderAddressInterface, AddressMod
         }
         $name .= ' ' . $this->getLastname();
         if ($this->getSuffix()) {
-            $name .= ' ' . $this->getSuffix();
+            // BEGINNING OF PATCH
+//            $name .= ' ' . $this->getSuffix();
+            $name .= ' ' . __($this->getSuffix());
+            // END OF PATCH
         }
         return $name;
     }
