diff --git a/Setup/FilePermissions.php b/Setup/FilePermissions.php
index 1ca91b5..4c5dfff 100644
--- a/Setup/FilePermissions.php
+++ b/Setup/FilePermissions.php
@@ -76,18 +76,7 @@ class FilePermissions
      */
     public function getInstallationWritableDirectories()
     {
-        if (!$this->installationWritableDirectories) {
-            $data = [
-                DirectoryList::CONFIG,
-                DirectoryList::VAR_DIR,
-                DirectoryList::MEDIA,
-                DirectoryList::STATIC_VIEW,
-            ];
-            foreach ($data as $code) {
-                $this->installationWritableDirectories[$code] = $this->directoryList->getPath($code);
-            }
-        }
-        return array_values($this->installationWritableDirectories);
+        return [];
     }

     /**