diff --git a/Controller/Article/Index.php b/Controller/Article/Index.php
index 7e2855c..4639867 100644
--- a/Controller/Article/Index.php
+++ b/Controller/Article/Index.php
@@ -110,34 +110,38 @@ class Index extends AbstractAction
         $resultPage = $this->resultPageFactory->create();
         $resultPage->getConfig()->getTitle()->set($metaTitle);
         $resultPage->getConfig()->setDescription($article->getMetaDescription());
-        $resultPage->getLayout()->getBlock('breadcrumbs')
-            ->addCrumb(
-                'home',
-                [
-                    'label' => 'Home',
-                    'title'=>__('Go to store homepage'),
-                    'link'=> $this->url->getBaseUrl()
-                ]
-            )->addCrumb(
-                'faq',
-                [
-                    'label' => $this->config->getFaqName(),
-                    'title'=>__('Go to %1', $this->config->getFaqName()),
-                    'link'=> $this->url->getFaqHomeUrl()
-                ]
-            )->addCrumb(
-                'category',
-                [
-                    'label' => $category->getName(),
-                    'title'=>__('Go to %1', $category->getName()),
-                    'link'=> $this->url->getCategoryUrl($category)
-                ]
-            )->addCrumb(
-                'article',
-                [
-                    'label' => $article->getTitle()
-                ]
-            );
+
+        $breadcrumbsBlock = $resultPage->getLayout()->getBlock('breadcrumbs');
+        if ($breadcrumbsBlock) {
+            $breadcrumbsBlock
+                ->addCrumb(
+                    'home',
+                    [
+                        'label' => 'Home',
+                        'title'=>__('Go to store homepage'),
+                        'link'=> $this->url->getBaseUrl()
+                    ]
+                )->addCrumb(
+                    'faq',
+                    [
+                        'label' => $this->config->getFaqName(),
+                        'title'=>__('Go to %1', $this->config->getFaqName()),
+                        'link'=> $this->url->getFaqHomeUrl()
+                    ]
+                )->addCrumb(
+                    'category',
+                    [
+                        'label' => $category->getName(),
+                        'title'=>__('Go to %1', $category->getName()),
+                        'link'=> $this->url->getCategoryUrl($category)
+                    ]
+                )->addCrumb(
+                    'article',
+                    [
+                        'label' => $article->getTitle()
+                    ]
+                );
+        }

         return $resultPage;
     }
diff --git a/Controller/Category/Index.php b/Controller/Category/Index.php
index ab9c6f9..ae7dda7 100644
--- a/Controller/Category/Index.php
+++ b/Controller/Category/Index.php
@@ -100,29 +100,33 @@ class Index extends AbstractAction
         $resultPage = $this->resultPageFactory->create();
         $resultPage->getConfig()->getTitle()->set($metaTitle);
         $resultPage->getConfig()->setDescription($category->getMetaDescription());
-        $resultPage->getLayout()->getBlock('breadcrumbs')
-            ->addCrumb(
-                'home',
-                [
-                    'label' => 'Home',
-                    'title'=>__('Go to store homepage'),
-                    'link'=> $this->url->getBaseUrl()
-                ]
-            )
-            ->addCrumb(
-                'faq',
-                [
-                    'label' => $this->config->getFaqName(),
-                    'title'=>__('Go to %1', $this->config->getFaqName()),
-                    'link'=> $this->url->getFaqHomeUrl()
-                ]
-            )->addCrumb(
-                'category',
-                [
-                    'label' => $category->getName(),
-                ]
-            );

+        $breadcrumbsBlock = $resultPage->getLayout()->getBlock('breadcrumbs');
+        if ($breadcrumbsBlock) {
+            $breadcrumbsBlock
+                ->addCrumb(
+                    'home',
+                    [
+                        'label' => 'Home',
+                        'title'=>__('Go to store homepage'),
+                        'link'=> $this->url->getBaseUrl()
+                    ]
+                )
+                ->addCrumb(
+                    'faq',
+                    [
+                        'label' => $this->config->getFaqName(),
+                        'title'=>__('Go to %1', $this->config->getFaqName()),
+                        'link'=> $this->url->getFaqHomeUrl()
+                    ]
+                )->addCrumb(
+                    'category',
+                    [
+                        'label' => $category->getName(),
+                    ]
+                );
+        }
+
         return $resultPage;
     }
 }
diff --git a/Controller/Index/Index.php b/Controller/Index/Index.php
index ea5e530..b6bb172 100644
--- a/Controller/Index/Index.php
+++ b/Controller/Index/Index.php
@@ -80,20 +80,25 @@ class Index extends AbstractAction
         $pageConfig->getTitle()->set($this->config->getFaqMetaTitle());
         $pageConfig->setDescription($this->config->getFaqMetaDescription());
         $pageConfig->addBodyClass('aw-columns-' . $this->config->getDefaultNumberOfColumnsToDisplay());
-        $resultPage->getLayout()->getBlock('breadcrumbs')
-            ->addCrumb(
-                'home',
-                [
-                    'label' => 'Home',
-                    'title'=>__('Go to store homepage'),
-                    'link'=> $this->url->getBaseUrl()
-                ]
-            )->addCrumb(
-                'faq',
-                [
-                    'label' => $this->config->getFaqName()
-                ]
-            );
+
+        $breadcrumbsBlock = $resultPage->getLayout()->getBlock('breadcrumbs');
+        if ($breadcrumbsBlock) {
+            $breadcrumbsBlock
+                ->addCrumb(
+                    'home',
+                    [
+                        'label' => 'Home',
+                        'title'=>__('Go to store homepage'),
+                        'link'=> $this->url->getBaseUrl()
+                    ]
+                )->addCrumb(
+                    'faq',
+                    [
+                        'label' => $this->config->getFaqName()
+                    ]
+                );
+        }
+
         return $resultPage;
     }
 }
diff --git a/Controller/Search/Index.php b/Controller/Search/Index.php
index b578e13..cc43549 100644
--- a/Controller/Search/Index.php
+++ b/Controller/Search/Index.php
@@ -111,27 +111,31 @@ class Index extends AbstractAction

         /** @var Page $resultPage */
         $resultPage = $this->resultPageFactory->create();
-        $resultPage->getLayout()->getBlock('breadcrumbs')
-            ->addCrumb(
-                'home',
-                [
-                    'label' => 'Home',
-                    'title'=>__('Go to store homepage'),
-                    'link'=> $this->url->getBaseUrl()
-                ]
-            )->addCrumb(
-                'faq',
-                [
-                    'label' => $this->config->getFaqName(),
-                    'title'=>__('Go to %1', $this->config->getFaqName()),
-                    'link'=> $this->url->getFaqHomeUrl()
-                ]
-            )->addCrumb(
-                'search',
-                [
-                    'label' => __('Search results for: "%1"', $searchQuery)
-                ]
-            );
+
+        $breadcrumbsBlock = $resultPage->getLayout()->getBlock('breadcrumbs');
+        if ($breadcrumbsBlock) {
+            $breadcrumbsBlock
+                ->addCrumb(
+                    'home',
+                    [
+                        'label' => 'Home',
+                        'title'=>__('Go to store homepage'),
+                        'link'=> $this->url->getBaseUrl()
+                    ]
+                )->addCrumb(
+                    'faq',
+                    [
+                        'label' => $this->config->getFaqName(),
+                        'title'=>__('Go to %1', $this->config->getFaqName()),
+                        'link'=> $this->url->getFaqHomeUrl()
+                    ]
+                )->addCrumb(
+                    'search',
+                    [
+                        'label' => __('Search results for: "%1"', $searchQuery)
+                    ]
+                );
+        }

         $resultPage->getLayout()->getBlock('aw_faq.search_results')->setData('data', $searchResults);
         $pageConfig = $resultPage->getConfig();