diff --git a/Model/ResourceModel/Quote/Item/Collection.php b/Model/ResourceModel/Quote/Item/Collection.php
index 0487d7e..8139edb 100644
--- a/Model/ResourceModel/Quote/Item/Collection.php
+++ b/Model/ResourceModel/Quote/Item/Collection.php
@@ -309,6 +309,9 @@ class Collection extends \Magento\Framework\Model\ResourceModel\Db\VersionContro
      */
     private function removeItemsWithAbsentProducts()
     {
+        if (empty($this->_productIds)) {
+            return;
+        }
         $productCollection = $this->_productCollectionFactory->create()->addIdFilter($this->_productIds);
         $existingProductsIds = $productCollection->getAllIds();
         $absentProductsIds = array_diff($this->_productIds, $existingProductsIds);
