--- a/Controller/Product/ListAjax.php
+++ b/Controller/Product/ListAjax.php
@@ -3,9 +3,12 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+
 namespace Magento\Review\Controller\Product;
 
-use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\App\ResponseInterface;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Framework\View\Result\Layout;
 use Magento\Review\Controller\Product as ProductController;
 use Magento\Framework\Controller\ResultFactory;
 
@@ -14,17 +17,16 @@ class ListAjax extends ProductController
     /**
      * Show list of product's reviews
      *
-     * @return \Magento\Framework\View\Result\Layout
+     * @return ResponseInterface|ResultInterface|Layout
      */
     public function execute()
     {
         if (!$this->initProduct()) {
-            throw new LocalizedException(__('Cannot initialize product'));
-        } else {
-            /** @var \Magento\Framework\View\Result\Layout $resultLayout */
-            $resultLayout = $this->resultFactory->create(ResultFactory::TYPE_LAYOUT);
+            /** @var \Magento\Framework\Controller\Result\Forward $resultForward */
+            $resultForward = $this->resultFactory->create(ResultFactory::TYPE_FORWARD);
+            return $resultForward->forward('noroute');
         }
 
-        return $resultLayout;
+        return $this->resultFactory->create(ResultFactory::TYPE_LAYOUT);
     }
-}
+}
\ No newline at end of file
