--- a/Controller/Adminhtml/Product/Attribute/Save.php
+++ b/Controller/Adminhtml/Product/Attribute/Save.php
@@ -195,7 +195,7 @@ class Save extends \Magento\Catalog\Controller\Adminhtml\Product\Attribute
 
             $data += ['is_filterable' => 0, 'is_filterable_in_search' => 0];
 
-            if (is_null($model->getIsUserDefined()) || $model->getIsUserDefined() != 0) {
+            if (($model->getIsUserDefined() === null || $model->getIsUserDefined() != 0) && !($model->getBackendType() === 'int' && $model->getFrontendInput() === 'text')) {
                 $data['backend_type'] = $model->getBackendTypeByInput($data['frontend_input']);
             }
 
