diff --git a/lib/web/requirejs/domReady.js b/lib/web/requirejs/domReady.js
index 31bd0d7..099c6f7 100644
--- a/lib/web/requirejs/domReady.js
+++ b/lib/web/requirejs/domReady.js
@@ -78,7 +78,7 @@ define(function () {
             }
         }
 
-        //Check if document already complete, and if so, just trigger page load
+        //Check if document is no longer loading, and if so, just trigger page load
         //listeners. Latest webkit browsers also use "interactive", and
         //will fire the onDOMContentLoaded before "interactive" but not after
         //entering "interactive" or "complete". More details:
@@ -89,8 +89,9 @@ define(function () {
         //so removing the || document.readyState === "interactive" test.
         //There is still a window.onload binding that should get fired if
         //DOMContentLoaded is missed.
-        if (document.readyState === "complete") {
-            pageLoaded();
+        if (document.readyState !== "loading") {
+            // Handle it asynchronously to allow scripts the opportunity to delay ready
+            setTimeout(pageLoaded);
         }
     }
 
@@ -126,4 +127,4 @@ define(function () {
     /** END OF PUBLIC API **/
 
     return domReady;
-});
\ No newline at end of file
+});
