diff --git a/src/module-elasticsuite-catalog/Model/Layer/Filter/DecimalFilterTrait.php b/src/module-elasticsuite-catalog/Model/Layer/Filter/DecimalFilterTrait.php
index 5f2bdfb..b22e1ed 100644
--- a/src/module-elasticsuite-catalog/Model/Layer/Filter/DecimalFilterTrait.php
+++ b/src/module-elasticsuite-catalog/Model/Layer/Filter/DecimalFilterTrait.php
@@ -50,7 +50,7 @@ trait DecimalFilterTrait
                 list($fromValue, $toValue) = $filter;
                 $this->setCurrentValue(['from' => $fromValue, 'to' => $toValue]);
 
-                $bounds = array_filter(['gte' => $fromValue, 'lt' => $toValue]);
+                $bounds = array_filter(['gte' => $fromValue, 'lte' => $toValue]);
 
                 $this->getLayer()->getProductCollection()->addFieldToFilter(
                     $this->getAttributeModel()->getAttributeCode(),
diff --git a/src/module-elasticsuite-catalog/view/frontend/web/js/range-slider-widget.js b/src/module-elasticsuite-catalog/view/frontend/web/js/range-slider-widget.js
index 7063672..ca661e7 100644
--- a/src/module-elasticsuite-catalog/view/frontend/web/js/range-slider-widget.js
+++ b/src/module-elasticsuite-catalog/view/frontend/web/js/range-slider-widget.js
@@ -110,7 +110,7 @@ define(["jquery", 'Magento_Catalog/js/price-utils', 'mage/template', "jquery/ui"
 
         _getItemCount : function() {
             var from = this.from, to = this.to, intervals = this.intervals;
-            var count = intervals.map(function(item) {return item.value >= from && item.value < to ? item.count : 0;})
+            var count = intervals.map(function(item) {return item.value >= from && item.value <= to ? item.count : 0;})
                                  .reduce(function(a,b) {return a + b;});
             return count;
         },
