diff --git a/Helper/Data.php b/Helper/Data.php
index 9bea197..3be22fc 100644
--- a/Helper/Data.php
+++ b/Helper/Data.php
@@ -261,14 +261,13 @@ class Data extends \Magento\Framework\App\Helper\AbstractHelper
         $groupRelations = [];
 
         foreach ($this->getPaymentMethods() as $code => $data) {
-            if (!empty($data['active'])) {
-                $storedTitle = $this->getMethodInstance($code)->getConfigData('title', $store);
-                if (isset($storedTitle)) {
-                    $methods[$code] = $storedTitle;
-                } elseif (isset($data['title'])) {
-                    $methods[$code] = $data['title'];
-                }
+
+            $storeId = $store ? (int)$store->getId() : null;
+            $storedTitle = $this->getMethodStoreTitle($code, $storeId);
+            if (!empty($storedTitle)) {
+                $methods[$code] = $storedTitle;
             }
+
             if ($asLabelValue && $withGroups && isset($data['group'])) {
                 $groupRelations[$code] = $data['group'];
             }
@@ -350,4 +349,21 @@ class Data extends \Magento\Framework\App\Helper\AbstractHelper
             $store
         );
     }
+
+    /**
+     * Get config title of payment method
+     *
+     * @param string $code
+     * @param int|null $storeId
+     * @return string
+     */
+    private function getMethodStoreTitle(string $code, ?int $storeId = null): string
+    {
+        $configPath = sprintf('%s/%s/title', self::XML_PATH_PAYMENT_METHODS, $code);
+        return (string) $this->scopeConfig->getValue(
+            $configPath,
+            \Magento\Store\Model\ScopeInterface::SCOPE_STORE,
+            $storeId
+        );
+    }
 }
