diff --git a/src/module-elasticsuite-catalog/Model/Layer/Filter/DecimalFilterTrait.php b/src/module-elasticsuite-catalog/Model/Layer/Filter/DecimalFilterTrait.php
index 315ec14..542be4a 100644
--- a/src/module-elasticsuite-catalog/Model/Layer/Filter/DecimalFilterTrait.php
+++ b/src/module-elasticsuite-catalog/Model/Layer/Filter/DecimalFilterTrait.php
@@ -51,10 +51,7 @@ trait DecimalFilterTrait
                 list($fromValue, $toValue) = $filter;
                 $this->setCurrentValue(['from' => $fromValue, 'to' => $toValue]);
 
-                $bounds = array_filter(['gte' => $fromValue, 'lt' => $toValue]);
-                if ($fromValue === $toValue) {
-                    $bounds = array_filter(['gte' => $fromValue, 'lte' => $toValue]);
-                }
+                $bounds = array_filter(['gte' => $fromValue, 'lte' => $toValue]);
 
                 $this->getLayer()->getProductCollection()->addFieldToFilter(
                     $this->getFilterField(),
diff --git a/src/module-elasticsuite-catalog/view/frontend/web/js/range-slider-widget.js b/src/module-elasticsuite-catalog/view/frontend/web/js/range-slider-widget.js
index e6c91ec..2d7f63b 100644
--- a/src/module-elasticsuite-catalog/view/frontend/web/js/range-slider-widget.js
+++ b/src/module-elasticsuite-catalog/view/frontend/web/js/range-slider-widget.js
@@ -188,13 +188,9 @@ define(['jquery', 'Magento_Catalog/js/price-utils', 'mage/template', 'Magento_Ui
 
         _getItemCount : function () {
             var from = this.from, to = this.to, intervals = this.intervals;
+            var count = intervals.map(function(item) {return item.value >= from && item.value <= to ? item.count : 0;})
+                .reduce(function(a,b) {return a + b;});
 
-            var count = intervals.map(function (item) {
-                return (item.value >= from && (item.value < to || ((from === to) && item.value <= to))) ? item.count : 0;
-            })
-            .reduce(function (a,b) {
-                return a + b;
-            });
             return count;
         },
 
