diff --git a/lib/web/magnifier/magnify.js b/lib/web/magnifier/magnify.js
index 9d67309..559e778 100644
--- a/lib/web/magnifier/magnify.js
+++ b/lib/web/magnifier/magnify.js
@@ -680,8 +680,6 @@ define([
                         $image.removeClass(imageDraggableClass);
                     }
                 } else if (gallery.fullScreen && (!transitionEnabled || !transitionActive)) {
-                    e.preventDefault();
-
                     imagePosY = getTop($image);
                     imagePosX = $image.offset().left;
 
