diff --git a/view/frontend/web/js/view/minicart.js b/view/frontend/web/js/view/minicart.js
index d152f94..5e29fa2 100644
--- a/view/frontend/web/js/view/minicart.js
+++ b/view/frontend/web/js/view/minicart.js
@@ -102,9 +102,8 @@ define([
                 self.isLoading(true);
             });
 
-            if (
-                cartData().website_id !== window.checkout.websiteId &&
-                cartData().website_id !== undefined
+            if (cartData().website_id !== window.checkout.websiteId ||
+                cartData().store_id !== window.checkout.storeId
             ) {
                 customerData.reload(['cart'], false);
             }
