diff --git a/Autoload/AutoloaderRegistry.php b/Autoload/AutoloaderRegistry.php
index f051d21..037d8c2 100644
--- a/Autoload/AutoloaderRegistry.php
+++ b/Autoload/AutoloaderRegistry.php
@@ -5,6 +5,7 @@
  */
 namespace Magento\Framework\Autoload;
 
+use InvalidArgumentException;
 use Magento\Framework\Autoload\AutoloaderInterface;
 
 /**
@@ -23,7 +24,7 @@ class AutoloaderRegistry
      * @param AutoloaderInterface $newAutoloader
      * @return void
      */
-    public static function registerAutoloader(AutoloaderInterface $newAutoloader)
+    public static function registerAutoloader(AutoloaderInterface $newAutoloader): void
     {
         self::$autoloader = $newAutoloader;
     }
@@ -31,15 +32,15 @@ class AutoloaderRegistry
     /**
      * Returns the registered autoloader
      *
-     * @throws \Exception
+     * @throws InvalidArgumentException
      * @return AutoloaderInterface
      */
-    public static function getAutoloader()
+    public static function getAutoloader(): AutoloaderInterface
     {
-        if (self::$autoloader !== null) {
-            return self::$autoloader;
-        } else {
-            throw new \Exception('Autoloader is not registered, cannot be retrieved.');
+        if (!self::$autoloader instanceof AutoloaderInterface) {
+            throw new InvalidArgumentException('Autoloader is not registered, cannot be retrieved.');
         }
+
+        return self::$autoloader;
     }
 }
