diff --git a/Plugin/Catalog/Model/Product/Attribute/RepositoryPlugin.php b/Plugin/Catalog/Model/Product/Attribute/RepositoryPlugin.php
--- a/Plugin/Catalog/Model/Product/Attribute/RepositoryPlugin.php
+++ b/Plugin/Catalog/Model/Product/Attribute/RepositoryPlugin.php
@@ -36,8 +36,9 @@
         \Magento\Catalog\Model\Product\Attribute\Repository $subject,
         \Magento\Catalog\Api\Data\ProductAttributeInterface $attribute
     ) {
-        $isPageBuilderEnabled = $attribute->getData('extension_attributes')
-            ? $attribute->getData('extension_attributes')->getIsPagebuilderEnabled()
+        $extensionAttributes = $attribute->getData('extension_attributes');
+        $isPageBuilderEnabled = $extensionAttributes && method_exists($extensionAttributes, 'getIsPagebuilderEnabled')
+            ? $extensionAttributes->getIsPagebuilderEnabled()
             : 0;
         $attribute->setData('is_pagebuilder_enabled', $isPageBuilderEnabled);
     }
@@ -58,7 +59,10 @@
         $extensionAttribute = $result->getExtensionAttributes()
             ? $result->getExtensionAttributes()
             : $this->extensionAttributesFactory->create();
-        $extensionAttribute->setIsPagebuilderEnabled($isPageBuilderEnabled);
+
+        if (method_exists($extensionAttribute, 'setIsPagebuilderEnabled')) {
+            $extensionAttribute->setIsPagebuilderEnabled($isPageBuilderEnabled);
+        }
         $result->setExtensionAttributes($extensionAttribute);
         return $result;
     }
