diff --git a/Model/ResourceModel/Archive.php b/Model/ResourceModel/Archive.php
index 31d3a9f..7413c30 100644
--- a/Model/ResourceModel/Archive.php
+++ b/Model/ResourceModel/Archive.php
@@ -185,11 +185,18 @@ class Archive extends EntityAbstract
             $ids = [$ids];
         }
 
+        // BEGINNING OF PATCH
+        $conditionalField = ($archiveEntity === ArchivalList::ORDER) ? 'entity_id' : 'order_id';
+        // END OF PATCH
+
         $select = $this->getConnection()->select()->from(
             $this->getArchiveEntityTable($archiveEntity),
             'entity_id'
         )->where(
-            'entity_id IN(?)',
+        // BEGINNING OF PATCH
+        // 'entity_id IN(?)',
+            $this->getConnection()->quoteIdentifier($conditionalField) . ' IN(?)',
+            // END OF PATCH
             $ids
         );
 
