diff --git a/view/base/web/js/block-loader.js b/view/base/web/js/block-loader.js
index 531591b..bf4ef3e 100644
--- a/view/base/web/js/block-loader.js
+++ b/view/base/web/js/block-loader.js
@@ -15,14 +15,18 @@ define([
         blockContentLoadingClass = '_block-content-loading',
         blockLoader,
         blockLoaderClass,
-        loaderImageHref;
+        blockLoaderElement = $.Deferred(),
+        loaderImageHref = $.Deferred();
 
     templateLoader.loadTemplate(blockLoaderTemplatePath).done(function (blockLoaderTemplate) {
-        blockLoader = template($.trim(blockLoaderTemplate), {
-            loaderImageHref: loaderImageHref
+        loaderImageHref.done(function (loaderHref) {
+            blockLoader = template($.trim(blockLoaderTemplate), {
+                loaderImageHref: loaderHref
+            });
+            blockLoader = $(blockLoader);
+            blockLoaderClass = '.' + blockLoader.attr('class');
+            blockLoaderElement.resolve();
         });
-        blockLoader = $(blockLoader);
-        blockLoaderClass = '.' + blockLoader.attr('class');
     });
 
     /**
@@ -70,7 +74,7 @@ define([
     }
 
     return function (loaderHref) {
-        loaderImageHref = loaderHref;
+        loaderImageHref.resolve(loaderHref);
         ko.bindingHandlers.blockLoader = {
             /**
              * Process loader for block
@@ -81,11 +85,11 @@ define([
                 element = $(element);
 
                 if (ko.unwrap(displayBlockLoader())) {
-                    addBlockLoader(element);
+                    blockLoaderElement.done(addBlockLoader(element));
                 } else {
-                    removeBlockLoader(element);
+                    blockLoaderElement.done(removeBlockLoader(element));
                 }
             }
         };
     };
-});
+});
\ No newline at end of file
