diff --git a/Controller/Product.php b/Controller/Product.php
index 80f8e93..4cdfef5 100644
--- a/Controller/Product.php
+++ b/Controller/Product.php
@@ -186,8 +186,12 @@ abstract class Product extends \Magento\Framework\App\Action\Action
         }
 
         if ($categoryId) {
-            $category = $this->categoryRepository->get($categoryId);
-            $this->coreRegistry->register('current_category', $category);
+            try {
+                $category = $this->categoryRepository->get($categoryId);
+                $this->coreRegistry->register('current_category', $category);
+            } catch (\Magento\Framework\Exception\NoSuchEntityException $e) {
+                $this->logger->critical($e);
+            }
         }
 
         try {
