diff --git a/Plugin/Checkout/LayoutProcessorPlugin.php b/Plugin/Checkout/LayoutProcessorPlugin.php
index 468e514..b986f04 100644
--- a/Plugin/Checkout/LayoutProcessorPlugin.php
+++ b/Plugin/Checkout/LayoutProcessorPlugin.php
@@ -69,7 +69,7 @@ class LayoutProcessorPlugin
             'formElement' => 'container',
             'component' => 'Magento_Ui/js/form/components/html',
             'additionalClasses' => 'repertus_packstation_search_button',
-            'content' => '<div><button data-bind="afterRender: setModalElement, blockLoader: isLoading" id="repertus_find_packstation_button" href="#" data-modal-search-url="'. $this->urlBuilder->getUrl('repertus_packstation/packstation/search') .'">' . __('Search for Packstation') . '</button></div>',
+            'content' => '<div><button type="button" data-bind="afterRender: setModalElement, blockLoader: isLoading" id="repertus_find_packstation_button" href="#" data-modal-search-url="'. $this->urlBuilder->getUrl('repertus_packstation/packstation/search') .'">' . __('Search for Packstation') . '</button></div>',
             'sortOrder' => 1
         ];
 
