diff --git a/view/frontend/web/js/repertus-packstation.js b/view/frontend/web/js/repertus-packstation.js
index f0491f8..8e8c675 100644
--- a/view/frontend/web/js/repertus-packstation.js
+++ b/view/frontend/web/js/repertus-packstation.js
@@ -236,7 +236,7 @@ require([
                     } else {
                         streetInput.first().val(addressTypeText + ' ' + numberValue);
                         streetInput.first().trigger('change');
-                        streetInput.last().val('');
+                        streetInput.not(":first").val('');
                     }
                 }
             }
