diff --git a/different-billing-address-for-packstation.patch b/different-billing-address-for-packstation.patch
new file mode 100644
index 0000000..e69de29
diff --git a/view/frontend/web/js/view/payment/default-mixin.js b/view/frontend/web/js/view/payment/default-mixin.js
index 55cfbe4..e2e8c62 100644
--- a/view/frontend/web/js/view/payment/default-mixin.js
+++ b/view/frontend/web/js/view/payment/default-mixin.js
@@ -3,8 +3,19 @@ define(
         'ko',
         'Magento_Checkout/js/model/quote',
         'Magento_Ui/js/model/messageList',
-        'mage/translate'
-    ], function (ko, quote, messageList, $t) {
+        'Magento_Checkout/js/model/payment-service',
+        'Magento_Checkout/js/action/select-payment-method',
+        'Magento_Checkout/js/action/create-billing-address',
+        'mage/translate',
+    ], function (
+        ko,
+        quote,
+        messageList,
+        paymentService,
+        selectPaymentMethod,
+        createBillingAddress,
+        $t
+    ) {
         'use strict';
 
         function isAddressPackstationOrPostOffice(address) {
@@ -42,6 +53,15 @@ define(
                         messageList.addErrorMessage({
                             message: $t('A Packstation or Post Office can not be selected as billing address.')
                         });
+
+                        if (!quote.paymentMethod()) {
+                            var paymentMethods = paymentService.getAvailablePaymentMethods();
+                            if (paymentMethods.length) {
+                                var separateBillingAddress = createBillingAddress({});
+                                quote.billingAddress(separateBillingAddress);
+                                selectPaymentMethod(paymentMethods[0]);
+                            }
+                        }
                     }
                 }, me);
 
