diff --git a/view/base/web/lib/require-css/css.js b/view/base/web/lib/require-css/css.js
index 795b782..2f9ba40 100644
--- a/view/base/web/lib/require-css/css.js
+++ b/view/base/web/lib/require-css/css.js
@@ -31,7 +31,7 @@
  */
 
 define(function() {
-//>>excludeStart('excludeRequireCss', pragmas.excludeRequireCss)
+//>>excludeStart("excludeRequireCss", pragmas.excludeRequireCss)
     if (typeof window == 'undefined')
         return { load: function(n, r, load){ load() } };
 
@@ -55,11 +55,11 @@ define(function() {
     else if (engine[4])
         useImportLoad = parseInt(engine[4]) < 18;
 
-//>>excludeEnd('excludeRequireCss')
+//>>excludeEnd("excludeRequireCss")
     //main api object
     var cssAPI = {};
 
-//>>excludeStart('excludeRequireCss', pragmas.excludeRequireCss)
+//>>excludeStart("excludeRequireCss", pragmas.excludeRequireCss)
     cssAPI.pluginBuilder = './css-builder';
 
     // <style> @import load method
@@ -149,7 +149,7 @@ define(function() {
         head.appendChild(link);
     }
 
-//>>excludeEnd('excludeRequireCss')
+//>>excludeEnd("excludeRequireCss")
     cssAPI.normalize = function(name, normalize) {
         if (name.substr(name.length - 4, 4) == '.css')
             name = name.substr(0, name.length - 4);
@@ -157,13 +157,13 @@ define(function() {
         return normalize(name);
     }
 
-//>>excludeStart('excludeRequireCss', pragmas.excludeRequireCss)
+//>>excludeStart("excludeRequireCss", pragmas.excludeRequireCss)
     cssAPI.load = function(cssId, req, load, config) {
 
         (useImportLoad ? importLoad : linkLoad)(req.toUrl(cssId + '.css'), load);
 
     }
 
-//>>excludeEnd('excludeRequireCss')
+//>>excludeEnd("excludeRequireCss")
     return cssAPI;
 });
\ No newline at end of file
