diff --git a/Model/Storage/DbStorage.php b/Model/Storage/DbStorage.php
index b095607..cfbed7a 100644
--- a/Model/Storage/DbStorage.php
+++ b/Model/Storage/DbStorage.php
@@ -235,6 +235,14 @@ class DbStorage extends AbstractStorage
                     . ' = ' . $this->connection->quote($entityType)
                     . $requestPathFilter
                 );
+                // PATCH BEGIN
+                $oldUrlsSelect->orWhere(
+                    $this->connection->quoteIdentifier(UrlRewrite::STORE_ID)
+                    . ' = ' . $this->connection->quote($storeId, 'INTEGER')
+                    . ' AND ' . $this->connection->quoteIdentifier(UrlRewrite::REQUEST_PATH)
+                    . ' IN (' . $this->connection->quote($requestPaths) . ')'
+                );
+                // PATCH END
             }
         }
         // prevent query locking in a case when nothing to delete
