Original issue: https://github.com/magento/magento2/issues/39324

--- a/dev/tests/integration/framework/Magento/TestFramework/Store/StoreManager.php	2025-03-21 08:44:48
+++ b/dev/tests/integration/framework/Magento/TestFramework/Store/StoreManager.php	2025-03-21 08:45:16
@@ -128,6 +128,12 @@
         //In order to restore configFixture values
         $testAppConfig = ObjectManager::getInstance()->get(Config::class);
         $reflection = new \ReflectionClass($testAppConfig);
+        // PATCH START
+        if ($reflection->implementsInterface(\Magento\Framework\Interception\InterceptorInterface::class)) {
+            $reflection = $reflection->getParentClass();
+        }
+        // PATCH END
+
         $dataProperty = $reflection->getProperty('data');
         $dataProperty->setAccessible(true);
         $savedConfig = $dataProperty->getValue($testAppConfig);
