diff --git a/setup/src/Magento/Setup/Module.php b/setup/src/Magento/Setup/Module.php
index ceb1733..4ad4e8e 100644
--- a/setup/src/Magento/Setup/Module.php
+++ b/setup/src/Magento/Setup/Module.php
@@ -76,7 +76,7 @@ class Module implements
         $result = array_merge_recursive(
             include __DIR__ . '/../../../config/module.config.php',
             include __DIR__ . '/../../../config/router.config.php',
-            include __DIR__ . '/../../../config/di.config.php',
+            include __DIR__ . '/../../../config/di.config.php'
         );
         // phpcs:enable
         return $result;
