diff --git a/Model/Client/Deepl.php b/Model/Client/Deepl.php
index a0fdc1c..ae53f33 100644
--- a/Model/Client/Deepl.php
+++ b/Model/Client/Deepl.php
@@ -106,7 +106,10 @@ class Deepl implements TranslatorInterface
             $this->_handleError($result);
         }
 
-        $translate = json_decode($result->getContent(), true);
+//        $translate = json_decode($result->getContent(), true);
+        // BEGIN PATCH
+        $translate = json_decode($result->getBody(), true);
+        // END PATCH
 
         if (!isset($translate['translations'][0]['text'])) {
             throw new LocalizedException(__('Translation is empty.'));
@@ -137,7 +140,10 @@ class Deepl implements TranslatorInterface
             $this->_handleError($result);
         }
 
-        $usage = json_decode($result->getContent(), true);
+//        $usage = json_decode($result->getContent(), true);
+        // BEGIN PATCH
+        $usage = json_decode($result->getBody(), true);
+        // END PATCH
 
         if (!isset($usage['character_count'])) {
             throw new LocalizedException(__('Usage is empty.'));
