Change GTM domain script kratki only

diff --git a/view/adminhtml/templates/gtm.phtml b/view/adminhtml/templates/gtm.phtml
index 3b05f9f..e77779b 100644
--- a/view/adminhtml/templates/gtm.phtml
+++ b/view/adminhtml/templates/gtm.phtml
@@ -17,7 +17,7 @@ $containerCode = $block->getEmbeddedCode();
 <script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
             new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
         j=d.createElement(s),dl=l!='<?= /* @noEscape */ $dataLayerName ?>'?'&l='+l:'';j.async=true;j.src=
-        'https://www.googletagmanager.com/gtm.js?id='+i+dl<?= $containerCode ? "+'{$containerCode}'" : '' ?>;f.parentNode.insertBefore(j,f);
+        'https://msr.kratki.com.com/gtm.js?id='+i+dl<?= $containerCode ? "+'{$containerCode}'" : '' ?>;f.parentNode.insertBefore(j,f);
     })(window,document,'script','<?= /* @noEscape */ $dataLayerName ?>','<?= /* @noEscape */ $accountId ?>');</script>
 <?php else: ?>
     <?= /* @noEscape */ $block->getAdvancedSettingsJsCode() ?>
