diff --git a/Controller/ApplePay/ShippingMethods.php b/Controller/ApplePay/ShippingMethods.php
index 50be298..0387525 100644
--- a/Controller/ApplePay/ShippingMethods.php
+++ b/Controller/ApplePay/ShippingMethods.php
@@ -119,7 +119,7 @@ class ShippingMethods extends Action
                 return [
                     'type' => 'final',
                     'label' => $total->getData('title'),
-                    'amount' => number_format($total->getData('value'), 2, '.', ''),
+                    'amount' => number_format((float)$total->getData('value'), 2, '.', ''),
                 ];
             }, array_values($cart->getTotals()))
         ]);
