diff --git a/view/frontend/web/js/view/payment/method-renderer.js b/view/frontend/web/js/view/payment/method-renderer.js
index 3e0d47a..5303271 100644
--- a/view/frontend/web/js/view/payment/method-renderer.js
+++ b/view/frontend/web/js/view/payment/method-renderer.js
@@ -82,7 +82,7 @@ define(
          */
         if (canUseApplePay()) {
             var applePayComponent = defaultComponent;
-            if (checkoutConfig.applepay.integration_type === 'direct') {
+            if (checkoutConfig && checkoutConfig.applepay.integration_type === 'direct') {
                 applePayComponent = 'Mollie_Payment/js/view/payment/method-renderer/applepay-direct';
             }

