diff --git a/Service/Quote/CartContainsRecurringProduct.php b/Service/Quote/CartContainsRecurringProduct.php
index ac46507..b169bf0 100644
--- a/Service/Quote/CartContainsRecurringProduct.php
+++ b/Service/Quote/CartContainsRecurringProduct.php
@@ -27,6 +27,13 @@ class CartContainsRecurringProduct
         $items = $cart->getItemsCollection()->getItems();
         foreach ($items as $item) {
             $buyRequest = $item->getOptionByCode('info_buyRequest');
+
+            // BEGIN PATCH
+            if (empty($buyRequest)) {
+                continue;
+            }
+            // END PATCH
+
             if ($buyRequest && strstr($buyRequest->getValue(), 'is_recurring') !== false &&
                 $this->jsonContainsRecurringValue($buyRequest->getValue())) {
                 return true;
