diff --git a/Cron/RemoveSentPaymentReminders.php b/Cron/RemoveSentPaymentReminders.php
index 1f4ebe5..6c772e9 100644
--- a/Cron/RemoveSentPaymentReminders.php
+++ b/Cron/RemoveSentPaymentReminders.php
@@ -61,7 +61,10 @@ class RemoveSentPaymentReminders
     protected function deletePaymentReminders(): void
     {
         do {
-            $date = (new \DateTimeImmutable($this->dateTime->gmtDate()))->sub(new \DateInterval('P1W'));
+            // PATCH BEGIN
+            // $date = (new \DateTimeImmutable($this->dateTime->gmtDate()))->sub(new \DateInterval('P1W'));
+            $date = (new \DateTimeImmutable($this->dateTime->gmtDate()))->sub(new \DateInterval('P2W'));
+            // PATCH END
             $this->builder->addFilter(Order::CREATED_AT, $date, 'lt');
             $this->builder->setPageSize(10);

diff --git a/Model/Adminhtml/Source/SecondChanceEmailDelay.php b/Model/Adminhtml/Source/SecondChanceEmailDelay.php
index 2905471..fecedcd 100644
--- a/Model/Adminhtml/Source/SecondChanceEmailDelay.php
+++ b/Model/Adminhtml/Source/SecondChanceEmailDelay.php
@@ -13,18 +13,55 @@ class SecondChanceEmailDelay implements OptionSourceInterface
     public function toOptionArray()
     {
         $output = [];
+        // PATCH BEGIN
+        //$output[] = [
+        //    'label' => __('1 hours'),
+        //    'value' => 1,
+        //];
+        //
+        //for ($i = 2; $i <= 8; $i++) {
+        //    $output[] = [
+        //        'label' => __('%1 hours', $i),
+        //        'value' => $i,
+        //    ];
+        //}
         $output[] = [
-            'label' => __('1 hours'),
+            'label' => __('No delay (instant email) - only for testing purposes'),
+            'value' => 0,
+        ];
+
+        $output[] = [
+            'label' => __('1 hour'),
             'value' => 1,
         ];

-        for ($i = 2; $i <= 8; $i++) {
+        for ($i = 2; $i < 8; $i++) {
+            $output[] = [
+                'label' => __('%1 hours', $i),
+                'value' => $i,
+            ];
+        }
+
+        for ($i = 8; $i < 24; $i += 4) {
             $output[] = [
                 'label' => __('%1 hours', $i),
                 'value' => $i,
             ];
         }

+        $output[] = [
+            'label' => __('1 day'),
+            'value' => 24,
+        ];
+
+        for ($i = 48; $i <= 336; $i += 24) {
+            $output[] = [
+                'label' => __('%1 days', $i / 24),
+                'value' => $i,
+            ];
+        }
+        // PATCH END
+
         return $output;
     }
-}
\ No newline at end of file
+}
diff --git a/Observer/MollieStartTransaction/SavePendingOrder.php b/Observer/MollieStartTransaction/SavePendingOrder.php
index adba860..c0a83e9 100644
--- a/Observer/MollieStartTransaction/SavePendingOrder.php
+++ b/Observer/MollieStartTransaction/SavePendingOrder.php
@@ -55,10 +55,15 @@ class SavePendingOrder implements ObserverInterface
         /** @var OrderInterface $order */
         $order = $observer->getData('order');

-        if (!$this->config->automaticallySendSecondChanceEmails($order->getStoreId()) ||
-            $order->getPayment()->getMethod() == Banktransfer::CODE) {
+        // PATCH BEGIN
+        //if (!$this->config->automaticallySendSecondChanceEmails($order->getStoreId()) ||
+        //    $order->getPayment()->getMethod() == Banktransfer::CODE) {
+        //    return;
+        //}
+        if (!$this->config->automaticallySendSecondChanceEmails($order->getStoreId())) {
             return;
         }
+        // PATCH END

         try {
             // If this succeeds there already exists a reminder.
