diff --git a/Model/Feeds.php b/Model/Feeds.php
index a76ddca..fde0bcc 100644
--- a/Model/Feeds.php
+++ b/Model/Feeds.php
@@ -1220,6 +1220,24 @@ class Feeds extends \Magento\Framework\Model\AbstractModel
             $this->categoriesFilterList = [0 => '*'];
         }
         $this->categoriesFilterList = array_values($this->categoriesFilterList);
+
+        //BEGIN PATCH - add all virtual categories:
+        /** @var \Magento\Eav\Model\Config $eavConfig */
+        $eavConfig = $this->objectManager->get(\Magento\Eav\Model\Config::class);
+        $attribute = $eavConfig->getAttribute(\Magento\Catalog\Model\Category::ENTITY, 'is_virtual_category');
+
+        /** @var \Magento\Framework\App\ResourceConnection $resourceConnection */
+        $resourceConnection = $this->objectManager->get(\Magento\Framework\App\ResourceConnection::class);
+        $connection = $resourceConnection->getConnection();
+        $tableName = $connection->getTableName('catalog_category_entity_int');
+        $select = $connection->select()
+            ->from($tableName, ['entity_id'])
+            ->where('attribute_id = ?', $attribute->getAttributeId())
+            ->where('value = 1');
+        $categoriesIds = $connection->fetchAssoc($select);
+
+        $this->categoriesFilterList = array_merge($this->categoriesFilterList, array_keys($categoriesIds));
+        //END PATCH
     }
     /**
      * Analyse product template, then check what attributes are required
diff --git a/Model/ResourceModel/Product/Collection.php b/Model/ResourceModel/Product/Collection.php
index 7c86d39..afc8207 100644
--- a/Model/ResourceModel/Product/Collection.php
+++ b/Model/ResourceModel/Product/Collection.php
@@ -390,10 +390,14 @@ class Collection extends \Magento\Catalog\Model\ResourceModel\Product\Collection
             //BEGIN PATCH
             $in = $params['category_filter'] ? 'IN' : 'NOT IN';
             $where .= empty($where) ? '' : ' AND';
-            $where .= " e.entity_id ". $in ." (SELECT product_id FROM catalog_category_product WHERE category_id IN (" . $filter . ")) ";
+            $where .= " e.entity_id ". $in ." (SELECT product_id FROM catalog_category_product WHERE category_id IN (" . $filter . ")) AND categories_1.product_id IS NOT NULL AND categories_2.product_id IS NOT NULL";
+
+
+            $this->getSelect()->joinLeft(['categories_1' => $tableCcpi], 'categories_1.product_id = e.entity_id', ["category_id", "categories_ids" => "GROUP_CONCAT( DISTINCT categories_1.category_id)"]);
+            $this->getSelect()->joinLeft(['categories_2' => $tableCcpi], 'categories_2.product_id = cpslcpe.entity_id', ["category_id", "categories_ids" => "GROUP_CONCAT( DISTINCT categories_2.category_id)"]);
+            $this->getSelect()->joinLeft(['cce_1' => $tableCce], 'categories_1.category_id = cce_1.entity_id AND cce_1.path LIKE \''. $categoryRootId . '/' . $storeRootId . '/%\'', ['parent_id', 'created_at', 'updated_at', 'path', 'position', 'level','children_count']);
+            $this->getSelect()->joinLeft(['cce_2' => $tableCce], 'categories_2.category_id = cce_2.entity_id AND cce_2.path LIKE \''. $categoryRootId . '/' . $storeRootId . '/%\'', ['parent_id', 'created_at', 'updated_at', 'path', 'position', 'level','children_count']);
             //END PATCH
-            $this->getSelect()->joinInner(['categories' => $tableCcpi], $ct, ["categories_ids" => "GROUP_CONCAT( DISTINCT categories.category_id)"]);
-            $this->getSelect()->joinInner(['cce' => $tableCce], "categories.category_id=cce.entity_id AND cce.path LIKE '" . $categoryRootId . "/" . $storeRootId . "/%'", []);
         } else {
             $this->getSelect()->joinLeft(['categories' => $tableCcpi], "categories.product_id=e.entity_id", ["categories_ids" => "GROUP_CONCAT( DISTINCT categories.category_id)"]);
             $this->getSelect()->joinLeft(['cce' => $tableCce], "categories.category_id=cce.entity_id AND cce.path LIKE '" . $categoryRootId . "/" . $storeRootId . "/%'", []);
