diff --git a/Helper/AttributesCategories.php b/Helper/AttributesCategories.php
index 0fdbdbf..80cf3df 100644
--- a/Helper/AttributesCategories.php
+++ b/Helper/AttributesCategories.php
@@ -80,7 +80,10 @@ class AttributesCategories extends \Magento\Framework\App\Helper\AbstractHelper
         $path = 0;
         $categoriesList = [];

-        foreach (explode(',', (string)$item->getCategoriesIds()) as $key => $category) {
+        //BEGIN PATCH
+        $itemCategories = $this->getItemCategories($item);
+        foreach (explode(',', (string)$itemCategories) as $key => $category) {
+        //END PATCH
             $isInCategoryFilter = $model->params['category_filter'] && isset($model->categories[$category]) && isset($model->categories[$category]['path']);
             $isOutCategoryFilter = !$model->params['category_filter'] && isset($model->categories[$category]) && isset($model->categories[$category]['path']);

@@ -146,6 +149,47 @@ class AttributesCategories extends \Magento\Framework\App\Helper\AbstractHelper
         return implode($separator, array_slice($category, $from, $length));
     }

+    //BEGIN PATCH
+    protected static array $virtualCategoriesIds = [];
+
+    protected function getItemCategories($item)
+    {
+        if ($item instanceof \Magento\Catalog\Model\Product) {
+            $virtualCategoriesIds = $this->getVirtualCategoriesIds();
+            $productCategoriesIds = explode(',', $item->getCategoriesIds());
+
+            return implode(',', array_diff($productCategoriesIds, $virtualCategoriesIds));
+        }
+
+        return $item->getCategoriesIds();
+    }
+
+    protected function getVirtualCategoriesIds(): array
+    {
+        if (!empty(self::$virtualCategoriesIds)) {
+            return self::$virtualCategoriesIds;
+        }
+
+        $objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+
+        $eavConfig = $objectManager->get(\Magento\Eav\Model\Config::class);
+        $attribute = $eavConfig->getAttribute(\Magento\Catalog\Model\Category::ENTITY, 'is_virtual_category');
+
+        /** @var \Magento\Framework\App\ResourceConnection $resourceConnection */
+        $resourceConnection = $objectManager->get(\Magento\Framework\App\ResourceConnection::class);
+        $connection = $resourceConnection->getConnection();
+        $tableName = $connection->getTableName('catalog_category_entity_int');
+        $select = $connection->select()
+            ->from($tableName, ['entity_id'])
+            ->where('attribute_id = ?', $attribute->getAttributeId())
+            ->where('value = 1');
+        $categoriesIds = $connection->fetchAssoc($select);
+
+        self::$virtualCategoriesIds = array_keys($categoriesIds);
+        return self::$virtualCategoriesIds;
+    }
+    //END PATCH
+
     /**
      * {category_mapping} attribute processing
      * @param \Wyomind\DataFeedManager\Model\Feeds $model
