diff --git a/Observer/AddBlockTranslation.php b/Observer/AddBlockTranslation.php
index 675f8a5..47ac356 100755
--- a/Observer/AddBlockTranslation.php
+++ b/Observer/AddBlockTranslation.php
@@ -119,10 +119,14 @@ class AddBlockTranslation implements ObserverInterface
             }
         } else {
             // Save Translation Date When Save Block
-            if ($postData['block_id']) {
+            // if ($postData['block_id']) {
+            if (!empty($postData['block_id'])) {
+            // END PATCH
                 $this->deeplAddonConfig->updateBlockTranslationDate($postData['block_id'], $dateTime);
             } else {
-                if ($postData['identifier']) {
+                // if ($postData['identifier']) {
+                if (!empty($postData['identifier']) && !empty($postData['store_id'])) {
+                // END PATCH
                     $blockData = $this->deeplAddonConfig->checkBlockExist(
                         $postData['identifier'],
                         $postData['store_id'][0]
diff --git a/Observer/AddPageTranslation.php b/Observer/AddPageTranslation.php
index 2e3f107..6608505 100755
--- a/Observer/AddPageTranslation.php
+++ b/Observer/AddPageTranslation.php
@@ -144,10 +144,15 @@ class AddPageTranslation implements ObserverInterface
                 }
             }
         } else {
-            if ($postData['page_id']) {
+            // BEGIN PATCH
+            // if ($postData['page_id']) {
+            if (!empty($postData['page_id']) ) {
+            // END PATCH
                 $this->deeplAddonConfig->updatePageTranslationDate($postData['page_id'], $dateTime);
             } else {
-                if ($postData['unique_identifier']) {
+                // if ($postData['unique_identifier']) {
+                if (!empty($postData['unique_identifier'])) {
+                // END PATCH
                     $destinationPage = $this->_pageInterfaceFactory->create();
                     $existingPage = $destinationPage->setStoreId($postData['store_id'][0])
                     ->load(
