diff --git a/Console/Command/Translate.php b/Console/Command/Translate.php
index 37d1d1a..fa45ba7 100755
--- a/Console/Command/Translate.php
+++ b/Console/Command/Translate.php
@@ -16,6 +16,23 @@ use Symfony\Component\Console\Output\OutputInterface;
 
 class Translate extends AddonTranslateGeneral
 {
+    // BEGIN PATCH
+    protected \Extendware\DeeplTranslator\Model\Translator\ProductFactory $productTranslatorFactory;
+
+    public function __construct(
+        \Magento\Store\Model\StoreManagerInterface $storeManager,
+        \Extendware\DeeplTranslator\Helper\Data $translateHelper,
+        \Extendware\DeeplTranslator\Helper\MessageQueue $messageQueue,
+        \Magento\Framework\App\State $state,
+        \Extendware\DeeplTranslator\Model\Translator\ProductFactory $productTranslatorFactory,
+        array $data = []
+    ) {
+        $this->productTranslatorFactory = $productTranslatorFactory;
+
+        parent::__construct($storeManager, $translateHelper, $messageQueue, $state, $data);
+    }
+    // END PATCH
+
     /**
      * Start the command for translating products
      *
@@ -28,6 +45,9 @@ class Translate extends AddonTranslateGeneral
         InputInterface $input,
         OutputInterface $output
     ) {
+        // BEGIN PATCH
+        $this->data['translateModel'] = $this->productTranslatorFactory->create();
+        // END PATCH
         return $this->commandProcess($input, $output);
     }
 
diff --git a/etc/di.xml b/etc/di.xml
index c82c7b8..d8ad91b 100755
--- a/etc/di.xml
+++ b/etc/di.xml
@@ -51,7 +51,9 @@
                 <item name="queueAttribute" xsi:type="string">product_action_attribute.translate</item>
                 <item name="processKey" xsi:type="string">product</item>
                 <item name="translateMethod" xsi:type="string">translateProduct</item>
-                <item name="translateModel" xsi:type="object">Extendware\DeeplTranslator\Model\Translator\Product</item>
+                <!--BEGIN PATCH-->
+                <!--<item name="translateModel" xsi:type="object">Extendware\DeeplTranslator\Model\Translator\Product</item>-->
+                <!--END PATCH-->
             </argument>
         </arguments>
     </type>
