diff --git a/view/frontend/templates/js/calendar.phtml b/view/frontend/templates/js/calendar.phtml
index 5579816..65e5cff 100644
--- a/view/frontend/templates/js/calendar.phtml
+++ b/view/frontend/templates/js/calendar.phtml
@@ -34,7 +34,7 @@ require([
             timeText: "<?= $block->escapeJs(__('Time')) ?>",
             hourText: "<?= $block->escapeJs(__('Hour')) ?>",
             minuteText: "<?= $block->escapeJs(__('Minute')) ?>",
-            dateFormat: $.datepicker.RFC_2822,
+            dateFormat: "D, d M yy", // $.datepicker.RFC_2822
             showOn: "button",
             showAnim: "",
             changeMonth: true,
