diff --git a/Ui/Component/Listing/Column/Status.php b/Ui/Component/Listing/Column/Status.php
index de9e9be..0bda37c 100644
--- a/Ui/Component/Listing/Column/Status.php
+++ b/Ui/Component/Listing/Column/Status.php
@@ -45,7 +45,7 @@ class Status extends \TechDivision\ImportGui\Ui\Component\Listing\Column\Status
         array $data = []
     ) {
         $this->pipelineQuery = $pipelineQuery;
-        parent::__construct($context, $uiComponentFactory, $urlBuilder, $components, $data);
+        parent::__construct($context, $uiComponentFactory, $components, $data);
     }
 
     /**
