diff --git a/Model/AppendReservations.php b/Model/AppendReservations.php
index 3b78818..3302a30 100644
--- a/Model/AppendReservations.php
+++ b/Model/AppendReservations.php
@@ -98,7 +98,17 @@ class AppendReservations
         $websiteCode = $this->websiteRepository->getById($websiteId)->getCode();
         $stockId = (int)$this->stockByWebsiteIdResolver->execute((int)$websiteId)->getStockId();
 
-        $this->checkItemsQuantity->execute($itemsBySku, $stockId);
+        // PATCH START
+        //$this->checkItemsQuantity->execute($itemsBySku, $stockId);
+        //
+        $itemsToCheck = [];
+        foreach ($itemsToSell as $item) {
+            if ($item->getCheckQty()) {
+                $itemsToCheck[$item->getSku()] = $item->getQuantity();
+            }
+        }
+        $this->checkItemsQuantity->execute($itemsToCheck, $stockId);
+        // PATCH END
 
         /** @var SalesEventExtensionInterface */
         $salesEventExtension = $this->salesEventExtensionFactory->create([
diff --git a/Plugin/Sales/OrderManagement/AppendReservationsAfterOrderPlacementPlugin.php b/Plugin/Sales/OrderManagement/AppendReservationsAfterOrderPlacementPlugin.php
index dc84616..a0b192b 100644
--- a/Plugin/Sales/OrderManagement/AppendReservationsAfterOrderPlacementPlugin.php
+++ b/Plugin/Sales/OrderManagement/AppendReservationsAfterOrderPlacementPlugin.php
@@ -113,12 +113,20 @@ class AppendReservationsAfterOrderPlacementPlugin
         OrderInterface $order
     ): OrderInterface {
         if ($this->reservationExecution->isDeferred()) {
-            $itemsById = $itemsBySku = $itemsToSell = [];
+            // PATCH START
+            //$itemsById = $itemsBySku = $itemsToSell = [];
+            $itemsById = $itemsBySku = $itemsToSell = $childrenOfBundle = [];
+            // PATCH END
             foreach ($order->getItems() as $item) {
                 if (!isset($itemsById[$item->getProductId()])) {
                     $itemsById[$item->getProductId()] = 0;
                 }
                 $itemsById[$item->getProductId()] += $item->getQtyOrdered();
+                // PATCH START
+                if ($item->getParentItem() && $item->getParentItem()->getProductType() == \Magento\Catalog\Model\Product\Type::TYPE_BUNDLE) {
+                    $childrenOfBundle[] = $item->getProductId();
+                }
+                // PATCH END
             }
             $productSkus = $this->getSkusByProductIds->execute(array_keys($itemsById));
             $productTypes = $this->getProductTypesBySkus->execute($productSkus);
@@ -129,10 +137,18 @@ class AppendReservationsAfterOrderPlacementPlugin
                 }
 
                 $itemsBySku[$sku] = (float)$itemsById[$productId];
-                $itemsToSell[] = $this->itemsToSellFactory->create([
+                // PATCH START
+                //$itemsToSell[] = $this->itemsToSellFactory->create([
+                //    'sku' => $sku,
+                //    'qty' => -(float)$itemsById[$productId],
+                //]);
+                $item = $this->itemsToSellFactory->create([
                     'sku' => $sku,
                     'qty' => -(float)$itemsById[$productId]
                 ]);
+                $item->setCheckQty(!in_array($productId, $childrenOfBundle));
+                $itemsToSell[] = $item;
+                // PATCH END
             }
 
             $websiteId = (int)$order->getStore()->getWebsiteId();
