diff --git a/App/Config/Initial.php b/App/Config/Initial.php
index a8ab5f2..45f01d8 100644
--- a/App/Config/Initial.php
+++ b/App/Config/Initial.php
@@ -58,8 +58,8 @@ class Initial
         } else {
             $data = $this->serializer->unserialize($data);
         }
-        $this->_data = $data['data'];
-        $this->_metadata = $data['metadata'];
+        $this->_data = $data['data'] ?? [];
+        $this->_metadata = $data['metadata'] ?? [];
     }
 
     /**
