diff --git a/View/Layout/etc/elements.xsd b/View/Layout/etc/elements.xsd
index 51f1931..d722702 100644
--- a/View/Layout/etc/elements.xsd
+++ b/View/Layout/etc/elements.xsd
@@ -328,6 +328,9 @@
         <xs:attribute type="elementNameType" name="name" use="required"/>
         <xs:attribute type="xs:string" name="template" use="optional"/>
         <xs:attribute type="xs:string" name="class" use="optional"/>
+        <!-- PATCH BEGIN: Make group attribute available to use: https://github.com/magento/magento2/issues/24656 -->
+        <xs:attribute type="xs:string" name="group" use="optional"/>
+        <!-- PATCH END -->
         <xs:attribute type="xs:boolean" name="display" default="true" use="optional"/>
         <xs:attribute type="xs:boolean" name="remove" use="optional"/>
     </xs:complexType>
