diff --git a/Interception/Code/Generator/Interceptor.php b/Interception/Code/Generator/Interceptor.php
index c363f80..4b8685f 100644
--- a/Interception/Code/Generator/Interceptor.php
+++ b/Interception/Code/Generator/Interceptor.php
@@ -212,7 +212,7 @@ METHOD_BODY
         $returnTypeValue = null;
         $returnType = $method->getReturnType();
         if ($returnType) {
-            $returnTypeValue = ($returnType->allowsNull() ? '?' : '');
+            $returnTypeValue = ($returnType->allowsNull() && $returnType->getName() !== 'mixed' ? '?' : '');
             $returnTypeValue .= ($returnType->getName() === 'self')
                 ? $this->_getFullyQualifiedClassName($method->getDeclaringClass()->getName())
                 : $returnType->getName();