commit 7218560abd0fc7879af691f59da296ac13ade6ee
Author: Adam Karnowka <adamkarnowka@Adams-MacBook-Pro-2.local>
Date:   Tue May 8 15:18:03 2018 +0200

    Patch for limited roles in admin panel

--- a/Controller/Adminhtml/Index/Render.php
+++ b/Controller/Adminhtml/Index/Render.php
@@ -54,7 +54,9 @@ class Render extends AbstractAction
     {
         if (isset($dataProviderConfigData['aclResource'])) {
             if (!$this->_authorization->isAllowed($dataProviderConfigData['aclResource'])) {
-                $this->_redirect('admin/denied');
+                if (!$this->_request->isAjax()) {
+                    $this->_redirect('admin/denied');
+                }
                 return false;
             }
         }
--- a/Test/Unit/Controller/Adminhtml/Index/RenderTest.php
+++ b/Test/Unit/Controller/Adminhtml/Index/RenderTest.php
@@ -192,6 +192,13 @@ class RenderTest extends \PHPUnit\Framework\TestCase
         $this->requestMock->expects($this->any())
             ->method('getParams')
             ->willReturn([]);
+        if ($isAllowed === false) {
+            $this->requestMock->expects($this->once())
+                ->method('isAjax')
+                ->willReturn(true);
+        }
+        $this->responseMock->expects($this->never())
+                    ->method('setRedirect');
         $this->responseMock->expects($this->any())
             ->method('appendBody')
             ->with($renderedData);
@@ -243,4 +250,4 @@ class RenderTest extends \PHPUnit\Framework\TestCase
             ],
         ];
     }
-}
+}

