Compatibility with PHP 8.1

diff --git a/Model/Import/Category.php b/Model/Import/Category.php
index fa8c23d..a521c5d 100644
--- a/Model/Import/Category.php
+++ b/Model/Import/Category.php
@@ -1063,7 +1063,7 @@ class Category extends \Magento\ImportExport\Model\Import\AbstractEntity
      * @param string $string
      * @return array
      */
-    protected function explodeEscaped($delimiter = '/', $string)
+    protected function explodeEscaped($delimiter, $string)
     {
         $exploded = explode($delimiter, $string);
         $fixed = [];
