diff --git a/view/frontend/templates/form.phtml b/view/frontend/templates/form.phtml
index eee9f74..6a8836e 100644
--- a/view/frontend/templates/form.phtml
+++ b/view/frontend/templates/form.phtml
@@ -69,8 +69,7 @@ $viewModel = $block->getViewModel();
                           class="input-text"
                           cols="5"
                           rows="3"
-                          data-validate="{required:true}"><?= $block->escapeHtml($viewModel->getUserComment()) ?>
-                </textarea>
+                          data-validate="{required:true}"><?= $block->escapeHtml($viewModel->getUserComment()) ?></textarea>
             </div>
         </div>
         <?= $block->getChildHtml('form.additional.info') ?>
