diff --git a/magento-2/Helper/LineItem.php b/magento-2/Helper/LineItem.php
index 3711637..a1b8fed 100644
--- a/magento-2/Helper/LineItem.php
+++ b/magento-2/Helper/LineItem.php
@@ -95,7 +95,12 @@ class LineItem extends AbstractHelper
         array $taxInfo = [])
     {
         $expectedAmount = $this->helper->roundAmount($expectedAmount, $currencyCode);
-        $effectiveAmount = $this->helper->roundAmount($this->getTotalAmountIncludingTax($items), $currencyCode);
+        // MOV PATCH START
+        // $effectiveAmount = $this->helper->roundAmount($this->getTotalAmountIncludingTax($items), $currencyCode);
+        $gcAmount = (int)\Magento\Framework\App\ObjectManager::getInstance()->get(\Magento\Checkout\Model\Session::class)->getQuote()?->getAwGiftcardAmount();
+        $effectiveAmount = $this->helper->roundAmount($this->getTotalAmountIncludingTax($items), $currencyCode) - $gcAmount;
+        $ensureConsistency = false;
+        // MOV PATCH END
         $difference = $expectedAmount - $effectiveAmount;
         if ($difference != 0) {
             if ($ensureConsistency) {